---
title: "Mouse DRG new data 8-28"
author: "Nick Bartelo"
date: "2022-08-29"
output: html_document
---

First we import the packages we may use.

```{r, warning=FALSE, message = FALSE}
library(SingleCellExperiment)
library(ggplot2)
library(magrittr)
library(DESeq2)
library(dplyr)
#install.packages("factoextra")
library(factoextra)
#BiocManager::install("clusterProfiler")
library(enrichplot)
library(org.Hs.eg.db) # Gene Ontology
library(clusterProfiler)
#BiocManager::install('M3C')
library(M3C) # TSNE
library(umap) # umap
library(cluster)
#install.packages('amap')
library(amap) # Kmeans
library(tidyverse)  # data manipulation
library(dendextend) # for comparing two dendrograms
# load library
library(ggrepel) # Volcano plot
library(tidyverse)
library(readxl)
#install.packages('xlsx')
library(xlsx)
library(data.table)
```

Below we import the data.

```{r}
# Ligand-coding genes from mDRG bulk RNA-sequencing results
mdrg_to_ra_specific_neurons <- read.csv('mouseDRG_ligands_to_RA_pain_receptors_interaction_interactome_all_v2_bw_specific_neurons.csv', row.names = 1)
mdrg_to_ra_all_types <- read.csv('mouseDRG_ligands_to_RA_pain_receptors_interaction_interactome_all_v2_neurons_vs_all_others_v2.csv', row.names = 1)
mdrg_to_ra_all_neurons <- read.csv('mouseDRG_ligands_to_RA_pain_receptors_interaction_interactome_all_v2_neurons_vs_other_neurons.csv', row.names = 1)

# Ligand-coding genes from the RA single-cell RNA-sequencing results
ra_to_mdrg_specific_neurons <- read.csv('RA_pain_ligands_to_mouseDRG_receptors_interaction_interactome_all_v2_bw_specific_neurons.csv', row.names = 1)
ra_to_mdrg_all_types <- read.csv('RA_pain_ligands_to_mouseDRG_receptors_interaction_interactome_all_v2_neurons_vs_all_others_v2.csv', row.names = 1)
ra_to_mdrg_all_neurons <- read.csv('RA_pain_ligands_to_mouseDRG_receptors_interaction_interactome_all_v2_neurons_vs_other_neurons.csv', row.names = 1)
```

Below we add a variable to each dataframe in order to identify if the genes are from the mDRG or RA single-cell results.

```{r}
mdrg_to_ra_specific_neurons$location <- 'mDRG'
ra_to_mdrg_specific_neurons$location <- 'RA sc'
mdrg_to_ra_all_types$location <- 'mDRG'
ra_to_mdrg_all_types$location <- 'RA sc'
mdrg_to_ra_all_neurons$location <- 'mDRG'
ra_to_mdrg_all_neurons$location <- 'RA sc'
```

Below we combine the data.

```{r}
specific_neuron_data <- rbind(mdrg_to_ra_specific_neurons, ra_to_mdrg_specific_neurons)
all_types_data <- rbind(mdrg_to_ra_all_types, ra_to_mdrg_all_types)
all_neurons_data <- rbind(mdrg_to_ra_all_neurons, ra_to_mdrg_all_neurons)
```

Below we show the number of unique genes in each file for the ligands and the receptors.

```{r}
length(unique(specific_neuron_data$ligand))
length(unique(all_types_data$ligand))
length(unique(all_neurons_data$ligand))

length(unique(specific_neuron_data$receptor))
length(unique(all_types_data$receptor))
length(unique(all_neurons_data$receptor))
```

Below we split each dataframe by fibroblast subtype.

```{r}
print('-----------------For specific neuron data file:--------------')
specific_neuron_data_f1 <- specific_neuron_data[(specific_neuron_data$ligand_type == 'SC-F1') | (specific_neuron_data$receptor_type == 'SC-F1'),]
print(nrow(specific_neuron_data_f1))
specific_neuron_data_f2 <- specific_neuron_data[(specific_neuron_data$ligand_type == 'SC-F2') | (specific_neuron_data$receptor_type == 'SC-F2'),]
print(nrow(specific_neuron_data_f2))
specific_neuron_data_f3 <- specific_neuron_data[(specific_neuron_data$ligand_type == 'SC-F3') | (specific_neuron_data$receptor_type == 'SC-F3'),]
print(nrow(specific_neuron_data_f3))
specific_neuron_data_f4 <- specific_neuron_data[(specific_neuron_data$ligand_type == 'SC-F4') | (specific_neuron_data$receptor_type == 'SC-F4'),]
print(nrow(specific_neuron_data_f4))

print('-----------------For all types data file:--------------')

all_types_data_f1 <- all_types_data[(all_types_data$ligand_type == 'SC-F1') | (all_types_data$receptor_type == 'SC-F1'),]
print(nrow(all_types_data_f1))
all_types_data_f2 <- all_types_data[(all_types_data$ligand_type == 'SC-F2') | (all_types_data$receptor_type == 'SC-F2'),]
print(nrow(all_types_data_f2))
all_types_data_f3 <- all_types_data[(all_types_data$ligand_type == 'SC-F3') | (all_types_data$receptor_type == 'SC-F3'),]
print(nrow(all_types_data_f3))
all_types_data_f4 <- all_types_data[(all_types_data$ligand_type == 'SC-F4') | (all_types_data$receptor_type == 'SC-F4'),]
print(nrow(all_types_data_f4))

print('-----------------For all neurons data file:--------------')
all_neuron_data_f1 <- all_neurons_data[(all_neurons_data$ligand_type == 'SC-F1') | (all_neurons_data$receptor_type == 'SC-F1'),]
print(nrow(all_neuron_data_f1))
all_neuron_data_f2 <- all_neurons_data[(all_neurons_data$ligand_type == 'SC-F2') | (all_neurons_data$receptor_type == 'SC-F2'),]
print(nrow(all_neuron_data_f2))
all_neuron_data_f3 <- all_neurons_data[(all_neurons_data$ligand_type == 'SC-F3') | (all_neurons_data$receptor_type == 'SC-F3'),]
print(nrow(all_neuron_data_f3))
all_neuron_data_f4 <- all_neurons_data[(all_neurons_data$ligand_type == 'SC-F4') | (all_neurons_data$receptor_type == 'SC-F4'),]
print(nrow(all_neuron_data_f4))
```

Below we get statistics for a multitude of things for specific neurons.

```{r}
print('Order: Total Unique Ligands')
print('Total Unique Receptors')
print('mDRG Ligand to Synovial Receptor Connections')
print('mDRG Ligands')
print('mDRG Receptors')
print('Synovial Ligand to mDRG Receptor Connections')
print('Synovial Ligands')
print('Synovial Receptors')
print('-------------SC-F1------------------')
length(unique(specific_neuron_data_f1$ligand))
length(unique(specific_neuron_data_f1$receptor))
specific_neuron_data_f1_mdrg = specific_neuron_data_f1[specific_neuron_data_f1$location == 'mDRG',]
nrow(specific_neuron_data_f1_mdrg)
length(unique(specific_neuron_data_f1_mdrg$ligand))
length(unique(specific_neuron_data_f1_mdrg$receptor))
specific_neuron_data_f1_ra = specific_neuron_data_f1[specific_neuron_data_f1$location == 'RA sc',]
nrow(specific_neuron_data_f1_ra)
length(unique(specific_neuron_data_f1_ra$ligand))
length(unique(specific_neuron_data_f1_ra$receptor))

print('-------------SC-F2------------------')
length(unique(specific_neuron_data_f2$ligand))
length(unique(specific_neuron_data_f2$receptor))
specific_neuron_data_f2_mdrg = specific_neuron_data_f2[specific_neuron_data_f2$location == 'mDRG',]
nrow(specific_neuron_data_f2_mdrg)
length(unique(specific_neuron_data_f2_mdrg$ligand))
length(unique(specific_neuron_data_f2_mdrg$receptor))
specific_neuron_data_f2_ra = specific_neuron_data_f2[specific_neuron_data_f2$location == 'RA sc',]
nrow(specific_neuron_data_f2_ra)
length(unique(specific_neuron_data_f2_ra$ligand))
length(unique(specific_neuron_data_f2_ra$receptor))

print('-------------SC-F3------------------')
length(unique(specific_neuron_data_f3$ligand))
length(unique(specific_neuron_data_f3$receptor))
specific_neuron_data_f3_mdrg = specific_neuron_data_f3[specific_neuron_data_f3$location == 'mDRG',]
nrow(specific_neuron_data_f3_mdrg)
length(unique(specific_neuron_data_f3_mdrg$ligand))
length(unique(specific_neuron_data_f3_mdrg$receptor))
specific_neuron_data_f3_ra = specific_neuron_data_f3[specific_neuron_data_f3$location == 'RA sc',]
nrow(specific_neuron_data_f3_ra)
length(unique(specific_neuron_data_f3_ra$ligand))
length(unique(specific_neuron_data_f3_ra$receptor))

print('-------------SC-F4------------------')
length(unique(specific_neuron_data_f4$ligand))
length(unique(specific_neuron_data_f4$receptor))
specific_neuron_data_f4_mdrg = specific_neuron_data_f4[specific_neuron_data_f4$location == 'mDRG',]
nrow(specific_neuron_data_f4_mdrg)
length(unique(specific_neuron_data_f4_mdrg$ligand))
length(unique(specific_neuron_data_f4_mdrg$receptor))
specific_neuron_data_f4_ra = specific_neuron_data_f4[specific_neuron_data_f4$location == 'RA sc',]
nrow(specific_neuron_data_f4_ra)
length(unique(specific_neuron_data_f4_ra$ligand))
length(unique(specific_neuron_data_f4_ra$receptor))
```

Below we get statistics for a multitude of things for all types.

```{r}
print('Order: Total Unique Ligands')
print('Total Unique Receptors')
print('mDRG Ligand to Synovial Receptor Connections')
print('mDRG Ligands')
print('mDRG Receptors')
print('Synovial Ligand to mDRG Receptor Connections')
print('Synovial Ligands')
print('Synovial Receptors')
print('-------------SC-F1------------------')
length(unique(all_types_data_f1$ligand))
length(unique(all_types_data_f1$receptor))
all_types_data_f1_mdrg = all_types_data_f1[all_types_data_f1$location == 'mDRG',]
nrow(all_types_data_f1_mdrg)
length(unique(all_types_data_f1_mdrg$ligand))
length(unique(all_types_data_f1_mdrg$receptor))
all_types_data_f1_ra = all_types_data_f1[all_types_data_f1$location == 'RA sc',]
nrow(all_types_data_f1_ra)
length(unique(all_types_data_f1_ra$ligand))
length(unique(all_types_data_f1_ra$receptor))

print('-------------SC-F2------------------')
length(unique(all_types_data_f2$ligand))
length(unique(all_types_data_f2$receptor))
all_types_data_f2_mdrg = all_types_data_f2[all_types_data_f2$location == 'mDRG',]
nrow(all_types_data_f2_mdrg)
length(unique(all_types_data_f2_mdrg$ligand))
length(unique(all_types_data_f2_mdrg$receptor))
all_types_data_f2_ra = all_types_data_f2[all_types_data_f2$location == 'RA sc',]
nrow(all_types_data_f2_ra)
length(unique(all_types_data_f2_ra$ligand))
length(unique(all_types_data_f2_ra$receptor))

print('-------------SC-F3------------------')
length(unique(all_types_data_f3$ligand))
length(unique(all_types_data_f3$receptor))
all_types_data_f3_mdrg = all_types_data_f3[all_types_data_f3$location == 'mDRG',]
nrow(all_types_data_f3_mdrg)
length(unique(all_types_data_f3_mdrg$ligand))
length(unique(all_types_data_f3_mdrg$receptor))
all_types_data_f3_ra = all_types_data_f3[all_types_data_f3$location == 'RA sc',]
nrow(all_types_data_f3_ra)
length(unique(all_types_data_f3_ra$ligand))
length(unique(all_types_data_f3_ra$receptor))

print('-------------SC-F4------------------')
length(unique(all_types_data_f4$ligand))
length(unique(all_types_data_f4$receptor))
all_types_data_f4_mdrg = all_types_data_f4[all_types_data_f4$location == 'mDRG',]
nrow(all_types_data_f4_mdrg)
length(unique(all_types_data_f4_mdrg$ligand))
length(unique(all_types_data_f4_mdrg$receptor))
all_types_data_f4_ra = all_types_data_f4[all_types_data_f4$location == 'RA sc',]
nrow(all_types_data_f4_ra)
length(unique(all_types_data_f4_ra$ligand))
length(unique(all_types_data_f4_ra$receptor))
```

Below we get statistics for a multitude of things for all neurons.

```{r}
print('Order: Total Unique Ligands')
print('Total Unique Receptors')
print('mDRG Ligand to Synovial Receptor Connections')
print('mDRG Ligands')
print('mDRG Receptors')
print('Synovial Ligand to mDRG Receptor Connections')
print('Synovial Ligands')
print('Synovial Receptors')
print('-------------SC-F1------------------')
length(unique(all_neuron_data_f1$ligand))
length(unique(all_neuron_data_f1$receptor))
all_neuron_data_f1_mdrg = all_neuron_data_f1[all_neuron_data_f1$location == 'mDRG',]
nrow(all_neuron_data_f1_mdrg)
length(unique(all_neuron_data_f1_mdrg$ligand))
length(unique(all_neuron_data_f1_mdrg$receptor))
all_neuron_data_f1_ra = all_neuron_data_f1[all_neuron_data_f1$location == 'RA sc',]
nrow(all_neuron_data_f1_ra)
length(unique(all_neuron_data_f1_ra$ligand))
length(unique(all_neuron_data_f1_ra$receptor))

print('-------------SC-F2------------------')
length(unique(all_neuron_data_f2$ligand))
length(unique(all_neuron_data_f2$receptor))
all_neuron_data_f2_mdrg = all_neuron_data_f2[all_neuron_data_f2$location == 'mDRG',]
nrow(all_neuron_data_f2_mdrg)
length(unique(all_neuron_data_f2_mdrg$ligand))
length(unique(all_neuron_data_f2_mdrg$receptor))
all_neuron_data_f2_ra = all_neuron_data_f2[all_neuron_data_f2$location == 'RA sc',]
nrow(all_neuron_data_f2_ra)
length(unique(all_neuron_data_f2_ra$ligand))
length(unique(all_neuron_data_f2_ra$receptor))

print('-------------SC-F3------------------')
length(unique(all_neuron_data_f3$ligand))
length(unique(all_neuron_data_f3$receptor))
all_neuron_data_f3_mdrg = all_neuron_data_f3[all_neuron_data_f3$location == 'mDRG',]
nrow(all_neuron_data_f3_mdrg)
length(unique(all_neuron_data_f3_mdrg$ligand))
length(unique(all_neuron_data_f3_mdrg$receptor))
all_neuron_data_f3_ra = all_neuron_data_f3[all_neuron_data_f3$location == 'RA sc',]
nrow(all_neuron_data_f3_ra)
length(unique(all_neuron_data_f3_ra$ligand))
length(unique(all_neuron_data_f3_ra$receptor))

print('-------------SC-F4------------------')
length(unique(all_neuron_data_f4$ligand))
length(unique(all_neuron_data_f4$receptor))
all_neuron_data_f4_mdrg = all_neuron_data_f4[all_neuron_data_f4$location == 'mDRG',]
nrow(all_neuron_data_f4_mdrg)
length(unique(all_neuron_data_f4_mdrg$ligand))
length(unique(all_neuron_data_f4_mdrg$receptor))
all_neuron_data_f4_ra = all_neuron_data_f4[all_neuron_data_f4$location == 'RA sc',]
nrow(all_neuron_data_f4_ra)
length(unique(all_neuron_data_f4_ra$ligand))
length(unique(all_neuron_data_f4_ra$receptor))
```

Below we specify by the three specific neuron cell types for all types before splitting by fibroblast subtype.

```{r}
print('------------Peripheral sensory neurofilament neurons--------------')
all_neurons_data_peripheral_sensory_neurofilament_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Peripheral sensory neurofilament neurons') | (all_neurons_data$receptor_type == 'Peripheral sensory neurofilament neurons'),]

print('------------Peripheral sensory non-peptidergic neurons--------------')
all_neurons_data_peripheral_sensory_nonpeptidergic_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Peripheral sensory non-peptidergic neurons') | (all_neurons_data$receptor_type == 'Peripheral sensory non-peptidergic neurons'),]

print('------------Peripheral sensory peptidergic neurons--------------')
all_neurons_data_peripheral_sensory_peptidergic_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Peripheral sensory peptidergic neurons') | (all_neurons_data$receptor_type == 'Peripheral sensory peptidergic neurons'),]

print('------------Cerebellum neurons--------------')
all_neurons_data_cerebellum_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Cerebellum neurons') | (all_neurons_data$receptor_type == 'Cerebellum neurons'),]

print('------------Cholinergic and monoaminergic neurons--------------')
all_neurons_data_cholinergic_and_monoaminergic_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Cholinergic and monoaminergic neurons') | (all_neurons_data$receptor_type == 'Cholinergic and monoaminergic neurons'),]

print('------------Dentate gyrus granule neurons--------------')
all_neurons_data_dentate_gyrus_granule_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Dentate gyrus granule neurons') | (all_neurons_data$receptor_type == 'Dentate gyrus granule neurons'),]

print('------------Di- and mesencephalon excitatory neurons--------------')
all_neurons_data_di_and_mesencephalon_excitatory_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Di- and mesencephalon excitatory neurons') | (all_neurons_data$receptor_type == 'Di- and mesencephalon excitatory neurons'),]

print('------------Di- and mesencephalon inhibitory neurons--------------')
all_neurons_data_di_and_mesencephalon_inhibitory_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Di- and mesencephalon inhibitory neurons') | (all_neurons_data$receptor_type == 'Di- and mesencephalon inhibitory neurons'),]

print('------------Enteric neurons--------------')
all_neurons_data_enteric_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Enteric neurons') | (all_neurons_data$receptor_type == 'Enteric neurons'),]

print('------------Hindbrain neurons--------------')
all_neurons_data_hindbrain_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Hindbrain neurons') | (all_neurons_data$receptor_type == 'Hindbrain neurons'),]

print('------------Olfactory inhibitory neurons--------------')
all_neurons_data_olfactory_inhibitory_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Olfactory inhibitory neurons') | (all_neurons_data$receptor_type == 'Olfactory inhibitory neurons'),]

print('------------Peptidergic neurons--------------')
all_neurons_data_peptidergic_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Peptidergic neurons') | (all_neurons_data$receptor_type == 'Peptidergic neurons'),]

print('------------Spinal cord inhibitory neurons--------------')
all_neurons_data_spinal_cord_inhibitory_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Spinal cord inhibitory neurons') | (all_neurons_data$receptor_type == 'Spinal cord inhibitory neurons'),]

print('------------Spinal cord excitatory neurons--------------')
all_neurons_data_spinal_cord_excitatory_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Spinal cord excitatory neurons') | (all_neurons_data$receptor_type == 'Spinal cord excitatory neurons'),]

print('------------Sympathetic cholinergic neurons--------------')
all_neurons_data_sympathetic_cholinergic_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Sympathetic cholinergic neurons') | (all_neurons_data$receptor_type == 'Sympathetic cholinergic neurons'),]

print('------------Sympathetic noradrenergic neurons--------------')
all_neurons_data_sympathetic_nonadrenergic_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Sympathetic noradrenergic neurons') | (all_neurons_data$receptor_type == 'Sympathetic noradrenergic neurons'),]
```

Below we look at some statistics for the unique cell types.

```{r}
print('Order: Total Unique Ligands')
print('Total Unique Receptors')
print('mDRG Ligand to Synovial Receptor Connections')
print('mDRG Ligands')
print('mDRG Receptors')
print('Synovial Ligand to mDRG Receptor Connections')
print('Synovial Ligands')
print('Synovial Receptors')

print('------------Cerebellum neurons--------------')
length(unique(all_neurons_data_cerebellum_neurons$ligand))
length(unique(all_neurons_data_cerebellum_neurons$receptor))
all_neurons_data_cerebellum_neurons_mdrg = all_neurons_data_cerebellum_neurons[all_neurons_data_cerebellum_neurons$location == 'mDRG',]
nrow(all_neurons_data_cerebellum_neurons_mdrg)
length(unique(all_neurons_data_cerebellum_neurons_mdrg$ligand))
length(unique(all_neurons_data_cerebellum_neurons_mdrg$receptor))
all_neurons_data_cerebellum_neurons_ra = all_neurons_data_cerebellum_neurons[all_neurons_data_cerebellum_neurons$location == 'RA sc',]
nrow(all_neurons_data_cerebellum_neurons_ra)
length(unique(all_neurons_data_cerebellum_neurons_ra$ligand))
length(unique(all_neurons_data_cerebellum_neurons_ra$receptor))
print('------------Cholinergic and monoaminergic neurons--------------')
length(unique(all_neurons_data_cholinergic_and_monoaminergic_neurons$ligand))
length(unique(all_neurons_data_cholinergic_and_monoaminergic_neurons$receptor))
all_neurons_data_cholinergic_and_monoaminergic_neurons_mdrg = all_neurons_data_cholinergic_and_monoaminergic_neurons[all_neurons_data_cholinergic_and_monoaminergic_neurons$location == 'mDRG',]
nrow(all_neurons_data_cholinergic_and_monoaminergic_neurons_mdrg)
length(unique(all_neurons_data_cholinergic_and_monoaminergic_neurons_mdrg$ligand))
length(unique(all_neurons_data_cholinergic_and_monoaminergic_neurons_mdrg$receptor))
all_neurons_data_cholinergic_and_monoaminergic_neurons_ra = all_neurons_data_cholinergic_and_monoaminergic_neurons[all_neurons_data_cholinergic_and_monoaminergic_neurons$location == 'RA sc',]
nrow(all_neurons_data_cholinergic_and_monoaminergic_neurons_ra)
length(unique(all_neurons_data_cholinergic_and_monoaminergic_neurons_ra$ligand))
length(unique(all_neurons_data_cholinergic_and_monoaminergic_neurons_ra$receptor))
print('------------Dentate gyrus granule neurons--------------')
length(unique(all_neurons_data_dentate_gyrus_granule_neurons$ligand))
length(unique(all_neurons_data_dentate_gyrus_granule_neurons$receptor))
all_neurons_data_dentate_gyrus_granule_neurons_mdrg = all_neurons_data_dentate_gyrus_granule_neurons[all_neurons_data_dentate_gyrus_granule_neurons$location == 'mDRG',]
nrow(all_neurons_data_dentate_gyrus_granule_neurons_mdrg)
length(unique(all_neurons_data_dentate_gyrus_granule_neurons_mdrg$ligand))
length(unique(all_neurons_data_dentate_gyrus_granule_neurons_mdrg$receptor))
all_neurons_data_dentate_gyrus_granule_neurons_ra = all_neurons_data_dentate_gyrus_granule_neurons[all_neurons_data_dentate_gyrus_granule_neurons$location == 'RA sc',]
nrow(all_neurons_data_dentate_gyrus_granule_neurons_ra)
length(unique(all_neurons_data_dentate_gyrus_granule_neurons_ra$ligand))
length(unique(all_neurons_data_dentate_gyrus_granule_neurons_ra$receptor))
print('------------Di- and mesencephalon excitatory neurons--------------')
length(unique(all_neurons_data_di_and_mesencephalon_excitatory_neurons$ligand))
length(unique(all_neurons_data_di_and_mesencephalon_excitatory_neurons$receptor))
all_neurons_data_di_and_mesencephalon_excitatory_neurons_mdrg = all_neurons_data_di_and_mesencephalon_excitatory_neurons[all_neurons_data_di_and_mesencephalon_excitatory_neurons$location == 'mDRG',]
nrow(all_neurons_data_di_and_mesencephalon_excitatory_neurons_mdrg)
length(unique(all_neurons_data_di_and_mesencephalon_excitatory_neurons_mdrg$ligand))
length(unique(all_neurons_data_di_and_mesencephalon_excitatory_neurons_mdrg$receptor))
all_neurons_data_di_and_mesencephalon_excitatory_neurons_ra = all_neurons_data_di_and_mesencephalon_excitatory_neurons[all_neurons_data_di_and_mesencephalon_excitatory_neurons$location == 'RA sc',]
nrow(all_neurons_data_di_and_mesencephalon_excitatory_neurons_ra)
length(unique(all_neurons_data_di_and_mesencephalon_excitatory_neurons_ra$ligand))
length(unique(all_neurons_data_di_and_mesencephalon_excitatory_neurons_ra$receptor))
print('------------Di- and mesencephalon inhibitory neurons--------------')
length(unique(all_neurons_data_di_and_mesencephalon_inhibitory_neurons$ligand))
length(unique(all_neurons_data_di_and_mesencephalon_inhibitory_neurons$receptor))
all_neurons_data_di_and_mesencephalon_inhibitory_neurons_mdrg = all_neurons_data_di_and_mesencephalon_inhibitory_neurons[all_neurons_data_di_and_mesencephalon_inhibitory_neurons$location == 'mDRG',]
nrow(all_neurons_data_di_and_mesencephalon_inhibitory_neurons_mdrg)
length(unique(all_neurons_data_di_and_mesencephalon_inhibitory_neurons_mdrg$ligand))
length(unique(all_neurons_data_di_and_mesencephalon_inhibitory_neurons_mdrg$receptor))
all_neurons_data_di_and_mesencephalon_inhibitory_neurons_ra = all_neurons_data_di_and_mesencephalon_inhibitory_neurons[all_neurons_data_di_and_mesencephalon_inhibitory_neurons$location == 'RA sc',]
nrow(all_neurons_data_di_and_mesencephalon_inhibitory_neurons_ra)
length(unique(all_neurons_data_di_and_mesencephalon_inhibitory_neurons_ra$ligand))
length(unique(all_neurons_data_di_and_mesencephalon_inhibitory_neurons_ra$receptor))
print('------------Enteric neurons--------------')
length(unique(all_neurons_data_enteric_neurons$ligand))
length(unique(all_neurons_data_enteric_neurons$receptor))
all_neurons_data_enteric_neurons_mdrg = all_neurons_data_enteric_neurons[all_neurons_data_enteric_neurons$location == 'mDRG',]
nrow(all_neurons_data_enteric_neurons_mdrg)
length(unique(all_neurons_data_enteric_neurons_mdrg$ligand))
length(unique(all_neurons_data_enteric_neurons_mdrg$receptor))
all_neurons_data_enteric_neurons_ra = all_neurons_data_enteric_neurons[all_neurons_data_enteric_neurons$location == 'RA sc',]
nrow(all_neurons_data_enteric_neurons_ra)
length(unique(all_neurons_data_enteric_neurons_ra$ligand))
length(unique(all_neurons_data_enteric_neurons_ra$receptor))

print('------------Hindbrain neurons--------------')
length(unique(all_neurons_data_hindbrain_neurons$ligand))
length(unique(all_neurons_data_hindbrain_neurons$receptor))
all_neurons_data_hindbrain_neurons_mdrg = all_neurons_data_hindbrain_neurons[all_neurons_data_hindbrain_neurons$location == 'mDRG',]
nrow(all_neurons_data_hindbrain_neurons_mdrg)
length(unique(all_neurons_data_hindbrain_neurons_mdrg$ligand))
length(unique(all_neurons_data_hindbrain_neurons_mdrg$receptor))
all_neurons_data_hindbrain_neurons_ra = all_neurons_data_hindbrain_neurons[all_neurons_data_hindbrain_neurons$location == 'RA sc',]
nrow(all_neurons_data_hindbrain_neurons_ra)
length(unique(all_neurons_data_hindbrain_neurons_ra$ligand))
length(unique(all_neurons_data_hindbrain_neurons_ra$receptor))

print('------------Olfactory inhibitory neurons--------------')
length(unique(all_neurons_data_olfactory_inhibitory_neurons$ligand))
length(unique(all_neurons_data_olfactory_inhibitory_neurons$receptor))
all_neurons_data_olfactory_inhibitory_neurons_mdrg = all_neurons_data_olfactory_inhibitory_neurons[all_neurons_data_olfactory_inhibitory_neurons$location == 'mDRG',]
nrow(all_neurons_data_olfactory_inhibitory_neurons_mdrg)
length(unique(all_neurons_data_olfactory_inhibitory_neurons_mdrg$ligand))
length(unique(all_neurons_data_olfactory_inhibitory_neurons_mdrg$receptor))
all_neurons_data_olfactory_inhibitory_neurons_ra = all_neurons_data_olfactory_inhibitory_neurons[all_neurons_data_olfactory_inhibitory_neurons$location == 'RA sc',]
nrow(all_neurons_data_olfactory_inhibitory_neurons_ra)
length(unique(all_neurons_data_olfactory_inhibitory_neurons_ra$ligand))
length(unique(all_neurons_data_olfactory_inhibitory_neurons_ra$receptor))

print('------------Peptidergic neurons--------------')
length(unique(all_neurons_data_peptidergic_neurons$ligand))
length(unique(all_neurons_data_peptidergic_neurons$receptor))
all_neurons_data_peptidergic_neurons_mdrg = all_neurons_data_peptidergic_neurons[all_neurons_data_peptidergic_neurons$location == 'mDRG',]
nrow(all_neurons_data_peptidergic_neurons_mdrg)
length(unique(all_neurons_data_peptidergic_neurons_mdrg$ligand))
length(unique(all_neurons_data_peptidergic_neurons_mdrg$receptor))
all_neurons_data_peptidergic_neurons_ra = all_neurons_data_peptidergic_neurons[all_neurons_data_peptidergic_neurons$location == 'RA sc',]
nrow(all_neurons_data_peptidergic_neurons_ra)
length(unique(all_neurons_data_peptidergic_neurons_ra$ligand))
length(unique(all_neurons_data_peptidergic_neurons_ra$receptor))

print('------------Spinal cord inhibitory neurons--------------')
length(unique(all_neurons_data_spinal_cord_inhibitory_neurons$ligand))
length(unique(all_neurons_data_spinal_cord_inhibitory_neurons$receptor))
all_neurons_data_spinal_cord_inhibitory_neurons_mdrg = all_neurons_data_spinal_cord_inhibitory_neurons[all_neurons_data_spinal_cord_inhibitory_neurons$location == 'mDRG',]
nrow(all_neurons_data_spinal_cord_inhibitory_neurons_mdrg)
length(unique(all_neurons_data_spinal_cord_inhibitory_neurons_mdrg$ligand))
length(unique(all_neurons_data_spinal_cord_inhibitory_neurons_mdrg$receptor))
all_neurons_data_spinal_cord_inhibitory_neurons_ra = all_neurons_data_spinal_cord_inhibitory_neurons[all_neurons_data_spinal_cord_inhibitory_neurons$location == 'RA sc',]
nrow(all_neurons_data_spinal_cord_inhibitory_neurons_ra)
length(unique(all_neurons_data_spinal_cord_inhibitory_neurons_ra$ligand))
length(unique(all_neurons_data_spinal_cord_inhibitory_neurons_ra$receptor))

print('------------Spinal cord excitatory neurons--------------')
length(unique(all_neurons_data_spinal_cord_excitatory_neurons$ligand))
length(unique(all_neurons_data_spinal_cord_excitatory_neurons$receptor))
all_neurons_data_spinal_cord_excitatory_neurons_mdrg = all_neurons_data_spinal_cord_excitatory_neurons[all_neurons_data_spinal_cord_excitatory_neurons$location == 'mDRG',]
nrow(all_neurons_data_spinal_cord_excitatory_neurons_mdrg)
length(unique(all_neurons_data_spinal_cord_excitatory_neurons_mdrg$ligand))
length(unique(all_neurons_data_spinal_cord_excitatory_neurons_mdrg$receptor))
all_neurons_data_spinal_cord_excitatory_neurons_ra = all_neurons_data_spinal_cord_excitatory_neurons[all_neurons_data_spinal_cord_excitatory_neurons$location == 'RA sc',]
nrow(all_neurons_data_spinal_cord_excitatory_neurons_ra)
length(unique(all_neurons_data_spinal_cord_excitatory_neurons_ra$ligand))
length(unique(all_neurons_data_spinal_cord_excitatory_neurons_ra$receptor))

print('------------Sympathetic cholinergic neurons--------------')
length(unique(all_neurons_data_sympathetic_cholinergic_neurons$ligand))
length(unique(all_neurons_data_sympathetic_cholinergic_neurons$receptor))
all_neurons_data_sympathetic_cholinergic_neurons_mdrg = all_neurons_data_sympathetic_cholinergic_neurons[all_neurons_data_sympathetic_cholinergic_neurons$location == 'mDRG',]
nrow(all_neurons_data_sympathetic_cholinergic_neurons_mdrg)
length(unique(all_neurons_data_sympathetic_cholinergic_neurons_mdrg$ligand))
length(unique(all_neurons_data_sympathetic_cholinergic_neurons_mdrg$receptor))
all_neurons_data_sympathetic_cholinergic_neurons_ra = all_neurons_data_sympathetic_cholinergic_neurons[all_neurons_data_sympathetic_cholinergic_neurons$location == 'RA sc',]
nrow(all_neurons_data_sympathetic_cholinergic_neurons_ra)
length(unique(all_neurons_data_sympathetic_cholinergic_neurons_ra$ligand))
length(unique(all_neurons_data_sympathetic_cholinergic_neurons_ra$receptor))
print('------------Sympathetic noradrenergic neurons--------------')
length(unique(all_neurons_data_sympathetic_nonadrenergic_neurons$ligand))
length(unique(all_neurons_data_sympathetic_nonadrenergic_neurons$receptor))
all_neurons_data_sympathetic_nonadrenergic_neurons_mdrg = all_neurons_data_sympathetic_nonadrenergic_neurons[all_neurons_data_sympathetic_nonadrenergic_neurons$location == 'mDRG',]
nrow(all_neurons_data_sympathetic_nonadrenergic_neurons_mdrg)
length(unique(all_neurons_data_sympathetic_nonadrenergic_neurons_mdrg$ligand))
length(unique(all_neurons_data_sympathetic_nonadrenergic_neurons_mdrg$receptor))
all_neurons_data_sympathetic_nonadrenergic_neurons_ra = all_neurons_data_sympathetic_nonadrenergic_neurons[all_neurons_data_sympathetic_nonadrenergic_neurons$location == 'RA sc',]
nrow(all_neurons_data_sympathetic_nonadrenergic_neurons_ra)
length(unique(all_neurons_data_sympathetic_nonadrenergic_neurons_ra$ligand))
length(unique(all_neurons_data_sympathetic_nonadrenergic_neurons_ra$receptor))

print('-------------Peripheral sensory neurofilament------------------')
length(unique(all_neurons_data_peripheral_sensory_neurofilament_neurons$ligand))
length(unique(all_neurons_data_peripheral_sensory_neurofilament_neurons$receptor))
all_neurons_data_peripheral_sensory_neurofilament_neurons_mdrg = all_neurons_data_peripheral_sensory_neurofilament_neurons[all_neurons_data_peripheral_sensory_neurofilament_neurons$location == 'mDRG',]
nrow(all_neurons_data_peripheral_sensory_neurofilament_neurons_mdrg)
length(unique(all_neurons_data_peripheral_sensory_neurofilament_neurons_mdrg$ligand))
length(unique(all_neurons_data_peripheral_sensory_neurofilament_neurons_mdrg$receptor))
all_neurons_data_peripheral_sensory_neurofilament_neurons_ra = all_neurons_data_peripheral_sensory_neurofilament_neurons[all_neurons_data_peripheral_sensory_neurofilament_neurons$location == 'RA sc',]
nrow(all_neurons_data_peripheral_sensory_neurofilament_neurons_ra)
length(unique(all_neurons_data_peripheral_sensory_neurofilament_neurons_ra$ligand))
length(unique(all_neurons_data_peripheral_sensory_neurofilament_neurons_ra$receptor))

print('-------------Peripheral sensory non-peptidergic neurons------------------')
length(unique(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons$ligand))
length(unique(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons$receptor))
all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_mdrg = all_neurons_data_peripheral_sensory_nonpeptidergic_neurons[all_neurons_data_peripheral_sensory_nonpeptidergic_neurons$location == 'mDRG',]
nrow(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_mdrg)
length(unique(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_mdrg$ligand))
length(unique(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_mdrg$receptor))
all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_ra = all_neurons_data_peripheral_sensory_nonpeptidergic_neurons[all_neurons_data_peripheral_sensory_nonpeptidergic_neurons$location == 'RA sc',]
nrow(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_ra)
length(unique(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_ra$ligand))
length(unique(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_ra$receptor))

print('-------------Peripheral sensory peptidergic neurons------------------')
length(unique(all_neurons_data_peripheral_sensory_peptidergic_neurons$ligand))
length(unique(all_neurons_data_peripheral_sensory_peptidergic_neurons$receptor))
all_neurons_data_peripheral_sensory_peptidergic_neurons_mdrg = all_neurons_data_peripheral_sensory_peptidergic_neurons[all_neurons_data_peripheral_sensory_peptidergic_neurons$location == 'mDRG',]
nrow(all_neurons_data_peripheral_sensory_peptidergic_neurons_mdrg)
length(unique(all_neurons_data_peripheral_sensory_peptidergic_neurons_mdrg$ligand))
length(unique(all_neurons_data_peripheral_sensory_peptidergic_neurons_mdrg$receptor))
all_neurons_data_peripheral_sensory_peptidergic_neurons_ra = all_neurons_data_peripheral_sensory_peptidergic_neurons[all_neurons_data_peripheral_sensory_peptidergic_neurons$location == 'RA sc',]
nrow(all_neurons_data_peripheral_sensory_peptidergic_neurons_ra)
length(unique(all_neurons_data_peripheral_sensory_peptidergic_neurons_ra$ligand))
length(unique(all_neurons_data_peripheral_sensory_peptidergic_neurons_ra$receptor))
```


Below we specify by the three specific neuron cell types for all groups before splitting by fibroblast subtype.

```{r}
print('------------Peripheral sensory neurofilament neurons--------------')
specific_neuron_data_peripheral_sensory_neurofilament_neurons <- specific_neuron_data[(specific_neuron_data$ligand_type == 'Peripheral sensory neurofilament neurons') | (specific_neuron_data$receptor_type == 'Peripheral sensory neurofilament neurons'),]

print('------------Peripheral sensory non-peptidergic neurons--------------')
specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons <- specific_neuron_data[(specific_neuron_data$ligand_type == 'Peripheral sensory non-peptidergic neurons') | (specific_neuron_data$receptor_type == 'Peripheral sensory non-peptidergic neurons'),]

print('------------Peripheral sensory peptidergic neurons--------------')
specific_neuron_data_peripheral_sensory_peptidergic_neurons <- specific_neuron_data[(specific_neuron_data$ligand_type == 'Peripheral sensory peptidergic neurons') | (specific_neuron_data$receptor_type == 'Peripheral sensory peptidergic neurons'),]
```

Below we look at some statistics for the unique cell types.

```{r}
print('Order: Total Unique Ligands')
print('Total Unique Receptors')
print('mDRG Ligand to Synovial Receptor Connections')
print('mDRG Ligands')
print('mDRG Receptors')
print('Synovial Ligand to mDRG Receptor Connections')
print('Synovial Ligands')
print('Synovial Receptors')
print('-------------Peripheral sensory neurofilament------------------')
length(unique(specific_neuron_data_peripheral_sensory_neurofilament_neurons$ligand))
length(unique(specific_neuron_data_peripheral_sensory_neurofilament_neurons$receptor))
specific_neuron_data_peripheral_sensory_neurofilament_neurons_mdrg = specific_neuron_data_peripheral_sensory_neurofilament_neurons[specific_neuron_data_peripheral_sensory_neurofilament_neurons$location == 'mDRG',]
nrow(specific_neuron_data_peripheral_sensory_neurofilament_neurons_mdrg)
length(unique(specific_neuron_data_peripheral_sensory_neurofilament_neurons_mdrg$ligand))
length(unique(specific_neuron_data_peripheral_sensory_neurofilament_neurons_mdrg$receptor))
specific_neuron_data_peripheral_sensory_neurofilament_neurons_ra = specific_neuron_data_peripheral_sensory_neurofilament_neurons[specific_neuron_data_peripheral_sensory_neurofilament_neurons$location == 'RA sc',]
nrow(specific_neuron_data_peripheral_sensory_neurofilament_neurons_ra)
length(unique(specific_neuron_data_peripheral_sensory_neurofilament_neurons_ra$ligand))
length(unique(specific_neuron_data_peripheral_sensory_neurofilament_neurons_ra$receptor))

print('-------------Peripheral sensory non-peptidergic neurons------------------')
length(unique(specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons$ligand))
length(unique(specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons$receptor))
specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons_mdrg = specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons[specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons$location == 'mDRG',]
nrow(specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons_mdrg)
length(unique(specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons_mdrg$ligand))
length(unique(specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons_mdrg$receptor))
specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons_ra = specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons[specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons$location == 'RA sc',]
nrow(specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons_ra)
length(unique(specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons_ra$ligand))
length(unique(specific_neuron_data_peripheral_sensory_nonpeptidergic_neurons_ra$receptor))

print('-------------Peripheral sensory peptidergic neurons------------------')
length(unique(specific_neuron_data_peripheral_sensory_peptidergic_neurons$ligand))
length(unique(specific_neuron_data_peripheral_sensory_peptidergic_neurons$receptor))
specific_neuron_data_peripheral_sensory_peptidergic_neurons_mdrg = specific_neuron_data_peripheral_sensory_peptidergic_neurons[specific_neuron_data_peripheral_sensory_peptidergic_neurons$location == 'mDRG',]
nrow(specific_neuron_data_peripheral_sensory_peptidergic_neurons_mdrg)
length(unique(specific_neuron_data_peripheral_sensory_peptidergic_neurons_mdrg$ligand))
length(unique(specific_neuron_data_peripheral_sensory_peptidergic_neurons_mdrg$receptor))
specific_neuron_data_peripheral_sensory_peptidergic_neurons_ra = specific_neuron_data_peripheral_sensory_peptidergic_neurons[specific_neuron_data_peripheral_sensory_peptidergic_neurons$location == 'RA sc',]
nrow(specific_neuron_data_peripheral_sensory_peptidergic_neurons_ra)
length(unique(specific_neuron_data_peripheral_sensory_peptidergic_neurons_ra$ligand))
length(unique(specific_neuron_data_peripheral_sensory_peptidergic_neurons_ra$receptor))
```

Below we specify by the three specific neuron cell types for all types before splitting by fibroblast subtype.

```{r}
print('------------Peripheral sensory neurofilament neurons--------------')
all_types_data_peripheral_sensory_neurofilament_neurons <- all_types_data[(all_types_data$ligand_type == 'Peripheral sensory neurofilament neurons') | (all_types_data$receptor_type == 'Peripheral sensory neurofilament neurons'),]

print('------------Peripheral sensory non-peptidergic neurons--------------')
all_types_data_peripheral_sensory_nonpeptidergic_neurons <- all_types_data[(all_types_data$ligand_type == 'Peripheral sensory non-peptidergic neurons') | (all_types_data$receptor_type == 'Peripheral sensory non-peptidergic neurons'),]

print('------------Peripheral sensory peptidergic neurons--------------')
all_types_data_peripheral_sensory_peptidergic_neurons <- all_types_data[(all_types_data$ligand_type == 'Peripheral sensory peptidergic neurons') | (all_types_data$receptor_type == 'Peripheral sensory peptidergic neurons'),]

print('------------Cerebellum neurons--------------')
all_types_data_cerebellum_neurons <- all_types_data[(all_types_data$ligand_type == 'Cerebellum neurons') | (all_types_data$receptor_type == 'Cerebellum neurons'),]

print('------------Cholinergic and monoaminergic neurons--------------')
all_types_data_cholinergic_and_monoaminergic_neurons <- all_types_data[(all_types_data$ligand_type == 'Cholinergic and monoaminergic neurons') | (all_types_data$receptor_type == 'Cholinergic and monoaminergic neurons'),]

print('------------Dentate gyrus granule neurons--------------')
all_types_data_dentate_gyrus_granule_neurons <- all_types_data[(all_types_data$ligand_type == 'Dentate gyrus granule neurons') | (all_types_data$receptor_type == 'Dentate gyrus granule neurons'),]

print('------------Di- and mesencephalon excitatory neurons--------------')
all_types_data_di_and_mesencephalon_excitatory_neurons <- all_types_data[(all_types_data$ligand_type == 'Di- and mesencephalon excitatory neurons') | (all_types_data$receptor_type == 'Di- and mesencephalon excitatory neurons'),]

print('------------Di- and mesencephalon inhibitory neurons--------------')
all_types_data_di_and_mesencephalon_inhibitory_neurons <- all_types_data[(all_types_data$ligand_type == 'Di- and mesencephalon inhibitory neurons') | (all_types_data$receptor_type == 'Di- and mesencephalon inhibitory neurons'),]

print('------------Enteric neurons--------------')
all_types_data_enteric_neurons <- all_types_data[(all_types_data$ligand_type == 'Enteric neurons') | (all_types_data$receptor_type == 'Enteric neurons'),]

print('------------Hindbrain neurons--------------')
all_types_data_hindbrain_neurons <- all_types_data[(all_types_data$ligand_type == 'Hindbrain neurons') | (all_types_data$receptor_type == 'Hindbrain neurons'),]

print('------------Olfactory inhibitory neurons--------------')
all_types_data_olfactory_inhibitory_neurons <- all_types_data[(all_types_data$ligand_type == 'Olfactory inhibitory neurons') | (all_types_data$receptor_type == 'Olfactory inhibitory neurons'),]

print('------------Peptidergic neurons--------------')
all_types_data_peptidergic_neurons <- all_types_data[(all_types_data$ligand_type == 'Peptidergic neurons') | (all_types_data$receptor_type == 'Peptidergic neurons'),]

print('------------Spinal cord inhibitory neurons--------------')
all_types_data_spinal_cord_inhibitory_neurons <- all_types_data[(all_types_data$ligand_type == 'Spinal cord inhibitory neurons') | (all_types_data$receptor_type == 'Spinal cord inhibitory neurons'),]

print('------------Spinal cord excitatory neurons--------------')
all_types_data_spinal_cord_excitatory_neurons <- all_types_data[(all_types_data$ligand_type == 'Spinal cord excitatory neurons') | (all_types_data$receptor_type == 'Spinal cord excitatory neurons'),]

print('------------Sympathetic cholinergic neurons--------------')
all_types_data_sympathetic_cholinergic_neurons <- all_types_data[(all_types_data$ligand_type == 'Sympathetic cholinergic neurons') | (all_types_data$receptor_type == 'Sympathetic cholinergic neurons'),]

print('------------Sympathetic noradrenergic neurons--------------')
all_types_data_sympathetic_nonadrenergic_neurons <- all_types_data[(all_types_data$ligand_type == 'Sympathetic noradrenergic neurons') | (all_types_data$receptor_type == 'Sympathetic noradrenergic neurons'),]
```

Below we look at some statistics for the unique cell types.

```{r}
print('Order: Total Unique Ligands')
print('Total Unique Receptors')
print('mDRG Ligand to Synovial Receptor Connections')
print('mDRG Ligands')
print('mDRG Receptors')
print('Synovial Ligand to mDRG Receptor Connections')
print('Synovial Ligands')
print('Synovial Receptors')

print('------------Cerebellum neurons--------------')
length(unique(all_types_data_cerebellum_neurons$ligand))
length(unique(all_types_data_cerebellum_neurons$receptor))
all_types_data_cerebellum_neurons_mdrg = all_types_data_cerebellum_neurons[all_types_data_cerebellum_neurons$location == 'mDRG',]
nrow(all_types_data_cerebellum_neurons_mdrg)
length(unique(all_types_data_cerebellum_neurons_mdrg$ligand))
length(unique(all_types_data_cerebellum_neurons_mdrg$receptor))
all_types_data_cerebellum_neurons_ra = all_types_data_cerebellum_neurons[all_types_data_cerebellum_neurons$location == 'RA sc',]
nrow(all_types_data_cerebellum_neurons_ra)
length(unique(all_types_data_cerebellum_neurons_ra$ligand))
length(unique(all_types_data_cerebellum_neurons_ra$receptor))
print('------------Cholinergic and monoaminergic neurons--------------')
length(unique(all_types_data_cholinergic_and_monoaminergic_neurons$ligand))
length(unique(all_types_data_cholinergic_and_monoaminergic_neurons$receptor))
all_types_data_cholinergic_and_monoaminergic_neurons_mdrg = all_types_data_cholinergic_and_monoaminergic_neurons[all_types_data_cholinergic_and_monoaminergic_neurons$location == 'mDRG',]
nrow(all_types_data_cholinergic_and_monoaminergic_neurons_mdrg)
length(unique(all_types_data_cholinergic_and_monoaminergic_neurons_mdrg$ligand))
length(unique(all_types_data_cholinergic_and_monoaminergic_neurons_mdrg$receptor))
all_types_data_cholinergic_and_monoaminergic_neurons_ra = all_types_data_cholinergic_and_monoaminergic_neurons[all_types_data_cholinergic_and_monoaminergic_neurons$location == 'RA sc',]
nrow(all_types_data_cholinergic_and_monoaminergic_neurons_ra)
length(unique(all_types_data_cholinergic_and_monoaminergic_neurons_ra$ligand))
length(unique(all_types_data_cholinergic_and_monoaminergic_neurons_ra$receptor))
print('------------Dentate gyrus granule neurons--------------')
length(unique(all_types_data_dentate_gyrus_granule_neurons$ligand))
length(unique(all_types_data_dentate_gyrus_granule_neurons$receptor))
all_types_data_dentate_gyrus_granule_neurons_mdrg = all_types_data_dentate_gyrus_granule_neurons[all_types_data_dentate_gyrus_granule_neurons$location == 'mDRG',]
nrow(all_types_data_dentate_gyrus_granule_neurons_mdrg)
length(unique(all_types_data_dentate_gyrus_granule_neurons_mdrg$ligand))
length(unique(all_types_data_dentate_gyrus_granule_neurons_mdrg$receptor))
all_types_data_dentate_gyrus_granule_neurons_ra = all_types_data_dentate_gyrus_granule_neurons[all_types_data_dentate_gyrus_granule_neurons$location == 'RA sc',]
nrow(all_types_data_dentate_gyrus_granule_neurons_ra)
length(unique(all_types_data_dentate_gyrus_granule_neurons_ra$ligand))
length(unique(all_types_data_dentate_gyrus_granule_neurons_ra$receptor))
print('------------Di- and mesencephalon excitatory neurons--------------')
length(unique(all_types_data_di_and_mesencephalon_excitatory_neurons$ligand))
length(unique(all_types_data_di_and_mesencephalon_excitatory_neurons$receptor))
all_types_data_di_and_mesencephalon_excitatory_neurons_mdrg = all_types_data_di_and_mesencephalon_excitatory_neurons[all_types_data_di_and_mesencephalon_excitatory_neurons$location == 'mDRG',]
nrow(all_types_data_di_and_mesencephalon_excitatory_neurons_mdrg)
length(unique(all_types_data_di_and_mesencephalon_excitatory_neurons_mdrg$ligand))
length(unique(all_types_data_di_and_mesencephalon_excitatory_neurons_mdrg$receptor))
all_types_data_di_and_mesencephalon_excitatory_neurons_ra = all_types_data_di_and_mesencephalon_excitatory_neurons[all_types_data_di_and_mesencephalon_excitatory_neurons$location == 'RA sc',]
nrow(all_types_data_di_and_mesencephalon_excitatory_neurons_ra)
length(unique(all_types_data_di_and_mesencephalon_excitatory_neurons_ra$ligand))
length(unique(all_types_data_di_and_mesencephalon_excitatory_neurons_ra$receptor))
print('------------Di- and mesencephalon inhibitory neurons--------------')
length(unique(all_types_data_di_and_mesencephalon_inhibitory_neurons$ligand))
length(unique(all_types_data_di_and_mesencephalon_inhibitory_neurons$receptor))
all_types_data_di_and_mesencephalon_inhibitory_neurons_mdrg = all_types_data_di_and_mesencephalon_inhibitory_neurons[all_types_data_di_and_mesencephalon_inhibitory_neurons$location == 'mDRG',]
nrow(all_types_data_di_and_mesencephalon_inhibitory_neurons_mdrg)
length(unique(all_types_data_di_and_mesencephalon_inhibitory_neurons_mdrg$ligand))
length(unique(all_types_data_di_and_mesencephalon_inhibitory_neurons_mdrg$receptor))
all_types_data_di_and_mesencephalon_inhibitory_neurons_ra = all_types_data_di_and_mesencephalon_inhibitory_neurons[all_types_data_di_and_mesencephalon_inhibitory_neurons$location == 'RA sc',]
nrow(all_types_data_di_and_mesencephalon_inhibitory_neurons_ra)
length(unique(all_types_data_di_and_mesencephalon_inhibitory_neurons_ra$ligand))
length(unique(all_types_data_di_and_mesencephalon_inhibitory_neurons_ra$receptor))
print('------------Enteric neurons--------------')
length(unique(all_types_data_enteric_neurons$ligand))
length(unique(all_types_data_enteric_neurons$receptor))
all_types_data_enteric_neurons_mdrg = all_types_data_enteric_neurons[all_types_data_enteric_neurons$location == 'mDRG',]
nrow(all_types_data_enteric_neurons_mdrg)
length(unique(all_types_data_enteric_neurons_mdrg$ligand))
length(unique(all_types_data_enteric_neurons_mdrg$receptor))
all_types_data_enteric_neurons_ra = all_types_data_enteric_neurons[all_types_data_enteric_neurons$location == 'RA sc',]
nrow(all_types_data_enteric_neurons_ra)
length(unique(all_types_data_enteric_neurons_ra$ligand))
length(unique(all_types_data_enteric_neurons_ra$receptor))

print('------------Hindbrain neurons--------------')
length(unique(all_types_data_hindbrain_neurons$ligand))
length(unique(all_types_data_hindbrain_neurons$receptor))
all_types_data_hindbrain_neurons_mdrg = all_types_data_hindbrain_neurons[all_types_data_hindbrain_neurons$location == 'mDRG',]
nrow(all_types_data_hindbrain_neurons_mdrg)
length(unique(all_types_data_hindbrain_neurons_mdrg$ligand))
length(unique(all_types_data_hindbrain_neurons_mdrg$receptor))
all_types_data_hindbrain_neurons_ra = all_types_data_hindbrain_neurons[all_types_data_hindbrain_neurons$location == 'RA sc',]
nrow(all_types_data_hindbrain_neurons_ra)
length(unique(all_types_data_hindbrain_neurons_ra$ligand))
length(unique(all_types_data_hindbrain_neurons_ra$receptor))

print('------------Olfactory inhibitory neurons--------------')
length(unique(all_types_data_olfactory_inhibitory_neurons$ligand))
length(unique(all_types_data_olfactory_inhibitory_neurons$receptor))
all_types_data_olfactory_inhibitory_neurons_mdrg = all_types_data_olfactory_inhibitory_neurons[all_types_data_olfactory_inhibitory_neurons$location == 'mDRG',]
nrow(all_types_data_olfactory_inhibitory_neurons_mdrg)
length(unique(all_types_data_olfactory_inhibitory_neurons_mdrg$ligand))
length(unique(all_types_data_olfactory_inhibitory_neurons_mdrg$receptor))
all_types_data_olfactory_inhibitory_neurons_ra = all_types_data_olfactory_inhibitory_neurons[all_types_data_olfactory_inhibitory_neurons$location == 'RA sc',]
nrow(all_types_data_olfactory_inhibitory_neurons_ra)
length(unique(all_types_data_olfactory_inhibitory_neurons_ra$ligand))
length(unique(all_types_data_olfactory_inhibitory_neurons_ra$receptor))

print('------------Peptidergic neurons--------------')
length(unique(all_types_data_peptidergic_neurons$ligand))
length(unique(all_types_data_peptidergic_neurons$receptor))
all_types_data_peptidergic_neurons_mdrg = all_types_data_peptidergic_neurons[all_types_data_peptidergic_neurons$location == 'mDRG',]
nrow(all_types_data_peptidergic_neurons_mdrg)
length(unique(all_types_data_peptidergic_neurons_mdrg$ligand))
length(unique(all_types_data_peptidergic_neurons_mdrg$receptor))
all_types_data_peptidergic_neurons_ra = all_types_data_peptidergic_neurons[all_types_data_peptidergic_neurons$location == 'RA sc',]
nrow(all_types_data_peptidergic_neurons_ra)
length(unique(all_types_data_peptidergic_neurons_ra$ligand))
length(unique(all_types_data_peptidergic_neurons_ra$receptor))

print('------------Spinal cord inhibitory neurons--------------')
length(unique(all_types_data_spinal_cord_inhibitory_neurons$ligand))
length(unique(all_types_data_spinal_cord_inhibitory_neurons$receptor))
all_types_data_spinal_cord_inhibitory_neurons_mdrg = all_types_data_spinal_cord_inhibitory_neurons[all_types_data_spinal_cord_inhibitory_neurons$location == 'mDRG',]
nrow(all_types_data_spinal_cord_inhibitory_neurons_mdrg)
length(unique(all_types_data_spinal_cord_inhibitory_neurons_mdrg$ligand))
length(unique(all_types_data_spinal_cord_inhibitory_neurons_mdrg$receptor))
all_types_data_spinal_cord_inhibitory_neurons_ra = all_types_data_spinal_cord_inhibitory_neurons[all_types_data_spinal_cord_inhibitory_neurons$location == 'RA sc',]
nrow(all_types_data_spinal_cord_inhibitory_neurons_ra)
length(unique(all_types_data_spinal_cord_inhibitory_neurons_ra$ligand))
length(unique(all_types_data_spinal_cord_inhibitory_neurons_ra$receptor))

print('------------Spinal cord excitatory neurons--------------')
length(unique(all_types_data_spinal_cord_excitatory_neurons$ligand))
length(unique(all_types_data_spinal_cord_excitatory_neurons$receptor))
all_types_data_spinal_cord_excitatory_neurons_mdrg = all_types_data_spinal_cord_excitatory_neurons[all_types_data_spinal_cord_excitatory_neurons$location == 'mDRG',]
nrow(all_types_data_spinal_cord_excitatory_neurons_mdrg)
length(unique(all_types_data_spinal_cord_excitatory_neurons_mdrg$ligand))
length(unique(all_types_data_spinal_cord_excitatory_neurons_mdrg$receptor))
all_types_data_spinal_cord_excitatory_neurons_ra = all_types_data_spinal_cord_excitatory_neurons[all_types_data_spinal_cord_excitatory_neurons$location == 'RA sc',]
nrow(all_types_data_spinal_cord_excitatory_neurons_ra)
length(unique(all_types_data_spinal_cord_excitatory_neurons_ra$ligand))
length(unique(all_types_data_spinal_cord_excitatory_neurons_ra$receptor))

print('------------Sympathetic cholinergic neurons--------------')
length(unique(all_types_data_sympathetic_cholinergic_neurons$ligand))
length(unique(all_types_data_sympathetic_cholinergic_neurons$receptor))
all_types_data_sympathetic_cholinergic_neurons_mdrg = all_types_data_sympathetic_cholinergic_neurons[all_types_data_sympathetic_cholinergic_neurons$location == 'mDRG',]
nrow(all_types_data_sympathetic_cholinergic_neurons_mdrg)
length(unique(all_types_data_sympathetic_cholinergic_neurons_mdrg$ligand))
length(unique(all_types_data_sympathetic_cholinergic_neurons_mdrg$receptor))
all_types_data_sympathetic_cholinergic_neurons_ra = all_types_data_sympathetic_cholinergic_neurons[all_types_data_sympathetic_cholinergic_neurons$location == 'RA sc',]
nrow(all_types_data_sympathetic_cholinergic_neurons_ra)
length(unique(all_types_data_sympathetic_cholinergic_neurons_ra$ligand))
length(unique(all_types_data_sympathetic_cholinergic_neurons_ra$receptor))
print('------------Sympathetic noradrenergic neurons--------------')
length(unique(all_types_data_sympathetic_nonadrenergic_neurons$ligand))
length(unique(all_types_data_sympathetic_nonadrenergic_neurons$receptor))
all_types_data_sympathetic_nonadrenergic_neurons_mdrg = all_types_data_sympathetic_nonadrenergic_neurons[all_types_data_sympathetic_nonadrenergic_neurons$location == 'mDRG',]
nrow(all_types_data_sympathetic_nonadrenergic_neurons_mdrg)
length(unique(all_types_data_sympathetic_nonadrenergic_neurons_mdrg$ligand))
length(unique(all_types_data_sympathetic_nonadrenergic_neurons_mdrg$receptor))
all_types_data_sympathetic_nonadrenergic_neurons_ra = all_types_data_sympathetic_nonadrenergic_neurons[all_types_data_sympathetic_nonadrenergic_neurons$location == 'RA sc',]
nrow(all_types_data_sympathetic_nonadrenergic_neurons_ra)
length(unique(all_types_data_sympathetic_nonadrenergic_neurons_ra$ligand))
length(unique(all_types_data_sympathetic_nonadrenergic_neurons_ra$receptor))

print('-------------Peripheral sensory neurofilament------------------')
length(unique(all_types_data_peripheral_sensory_neurofilament_neurons$ligand))
length(unique(all_types_data_peripheral_sensory_neurofilament_neurons$receptor))
all_types_data_peripheral_sensory_neurofilament_neurons_mdrg = all_types_data_peripheral_sensory_neurofilament_neurons[all_types_data_peripheral_sensory_neurofilament_neurons$location == 'mDRG',]
nrow(all_types_data_peripheral_sensory_neurofilament_neurons_mdrg)
length(unique(all_types_data_peripheral_sensory_neurofilament_neurons_mdrg$ligand))
length(unique(all_types_data_peripheral_sensory_neurofilament_neurons_mdrg$receptor))
all_types_data_peripheral_sensory_neurofilament_neurons_ra = all_types_data_peripheral_sensory_neurofilament_neurons[all_types_data_peripheral_sensory_neurofilament_neurons$location == 'RA sc',]
nrow(all_types_data_peripheral_sensory_neurofilament_neurons_ra)
length(unique(all_types_data_peripheral_sensory_neurofilament_neurons_ra$ligand))
length(unique(all_types_data_peripheral_sensory_neurofilament_neurons_ra$receptor))

print('-------------Peripheral sensory non-peptidergic neurons------------------')
length(unique(all_types_data_peripheral_sensory_nonpeptidergic_neurons$ligand))
length(unique(all_types_data_peripheral_sensory_nonpeptidergic_neurons$receptor))
all_types_data_peripheral_sensory_nonpeptidergic_neurons_mdrg = all_types_data_peripheral_sensory_nonpeptidergic_neurons[all_types_data_peripheral_sensory_nonpeptidergic_neurons$location == 'mDRG',]
nrow(all_types_data_peripheral_sensory_nonpeptidergic_neurons_mdrg)
length(unique(all_types_data_peripheral_sensory_nonpeptidergic_neurons_mdrg$ligand))
length(unique(all_types_data_peripheral_sensory_nonpeptidergic_neurons_mdrg$receptor))
all_types_data_peripheral_sensory_nonpeptidergic_neurons_ra = all_types_data_peripheral_sensory_nonpeptidergic_neurons[all_types_data_peripheral_sensory_nonpeptidergic_neurons$location == 'RA sc',]
nrow(all_types_data_peripheral_sensory_nonpeptidergic_neurons_ra)
length(unique(all_types_data_peripheral_sensory_nonpeptidergic_neurons_ra$ligand))
length(unique(all_types_data_peripheral_sensory_nonpeptidergic_neurons_ra$receptor))

print('-------------Peripheral sensory peptidergic neurons------------------')
length(unique(all_types_data_peripheral_sensory_peptidergic_neurons$ligand))
length(unique(all_types_data_peripheral_sensory_peptidergic_neurons$receptor))
all_types_data_peripheral_sensory_peptidergic_neurons_mdrg = all_types_data_peripheral_sensory_peptidergic_neurons[all_types_data_peripheral_sensory_peptidergic_neurons$location == 'mDRG',]
nrow(all_types_data_peripheral_sensory_peptidergic_neurons_mdrg)
length(unique(all_types_data_peripheral_sensory_peptidergic_neurons_mdrg$ligand))
length(unique(all_types_data_peripheral_sensory_peptidergic_neurons_mdrg$receptor))
all_types_data_peripheral_sensory_peptidergic_neurons_ra = all_types_data_peripheral_sensory_peptidergic_neurons[all_types_data_peripheral_sensory_peptidergic_neurons$location == 'RA sc',]
nrow(all_types_data_peripheral_sensory_peptidergic_neurons_ra)
length(unique(all_types_data_peripheral_sensory_peptidergic_neurons_ra$ligand))
length(unique(all_types_data_peripheral_sensory_peptidergic_neurons_ra$receptor))
```

Below we specify by the three specific neuron cell types for all neurons before splitting by fibroblast subtype.

```{r}
print('------------Peripheral sensory neurofilament neurons--------------')
all_neurons_data_peripheral_sensory_neurofilament_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Peripheral sensory neurofilament neurons') | (all_neurons_data$receptor_type == 'Peripheral sensory neurofilament neurons'),]

print('------------Peripheral sensory non-peptidergic neurons--------------')
all_neurons_data_peripheral_sensory_nonpeptidergic_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Peripheral sensory non-peptidergic neurons') | (all_neurons_data$receptor_type == 'Peripheral sensory non-peptidergic neurons'),]

print('------------Peripheral sensory peptidergic neurons--------------')
all_neurons_data_peripheral_sensory_peptidergic_neurons <- all_neurons_data[(all_neurons_data$ligand_type == 'Peripheral sensory peptidergic neurons') | (all_neurons_data$receptor_type == 'Peripheral sensory peptidergic neurons'),]
```

Below we look at some statistics for the unique cell types.

```{r}
print('Order: Total Unique Ligands')
print('Total Unique Receptors')
print('mDRG Ligand to Synovial Receptor Connections')
print('mDRG Ligands')
print('mDRG Receptors')
print('Synovial Ligand to mDRG Receptor Connections')
print('Synovial Ligands')
print('Synovial Receptors')
print('-------------Peripheral sensory neurofilament------------------')
length(unique(all_neurons_data_peripheral_sensory_neurofilament_neurons$ligand))
length(unique(all_neurons_data_peripheral_sensory_neurofilament_neurons$receptor))
all_neurons_data_peripheral_sensory_neurofilament_neurons_mdrg = all_neurons_data_peripheral_sensory_neurofilament_neurons[all_neurons_data_peripheral_sensory_neurofilament_neurons$location == 'mDRG',]
nrow(all_neurons_data_peripheral_sensory_neurofilament_neurons_mdrg)
length(unique(all_neurons_data_peripheral_sensory_neurofilament_neurons_mdrg$ligand))
length(unique(all_neurons_data_peripheral_sensory_neurofilament_neurons_mdrg$receptor))
all_neurons_data_peripheral_sensory_neurofilament_neurons_ra = all_neurons_data_peripheral_sensory_neurofilament_neurons[all_neurons_data_peripheral_sensory_neurofilament_neurons$location == 'RA sc',]
nrow(all_neurons_data_peripheral_sensory_neurofilament_neurons_ra)
length(unique(all_neurons_data_peripheral_sensory_neurofilament_neurons_ra$ligand))
length(unique(all_neurons_data_peripheral_sensory_neurofilament_neurons_ra$receptor))

print('-------------Peripheral sensory non-peptidergic neurons------------------')
length(unique(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons$ligand))
length(unique(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons$receptor))
all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_mdrg = all_neurons_data_peripheral_sensory_nonpeptidergic_neurons[all_neurons_data_peripheral_sensory_nonpeptidergic_neurons$location == 'mDRG',]
nrow(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_mdrg)
length(unique(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_mdrg$ligand))
length(unique(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_mdrg$receptor))
all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_ra = all_neurons_data_peripheral_sensory_nonpeptidergic_neurons[all_neurons_data_peripheral_sensory_nonpeptidergic_neurons$location == 'RA sc',]
nrow(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_ra)
length(unique(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_ra$ligand))
length(unique(all_neurons_data_peripheral_sensory_nonpeptidergic_neurons_ra$receptor))

print('-------------Peripheral sensory peptidergic neurons------------------')
length(unique(all_neurons_data_peripheral_sensory_peptidergic_neurons$ligand))
length(unique(all_neurons_data_peripheral_sensory_peptidergic_neurons$receptor))
all_neurons_data_peripheral_sensory_peptidergic_neurons_mdrg = all_neurons_data_peripheral_sensory_peptidergic_neurons[all_neurons_data_peripheral_sensory_peptidergic_neurons$location == 'mDRG',]
nrow(all_neurons_data_peripheral_sensory_peptidergic_neurons_mdrg)
length(unique(all_neurons_data_peripheral_sensory_peptidergic_neurons_mdrg$ligand))
length(unique(all_neurons_data_peripheral_sensory_peptidergic_neurons_mdrg$receptor))
all_neurons_data_peripheral_sensory_peptidergic_neurons_ra = all_neurons_data_peripheral_sensory_peptidergic_neurons[all_neurons_data_peripheral_sensory_peptidergic_neurons$location == 'RA sc',]
nrow(all_neurons_data_peripheral_sensory_peptidergic_neurons_ra)
length(unique(all_neurons_data_peripheral_sensory_peptidergic_neurons_ra$ligand))
length(unique(all_neurons_data_peripheral_sensory_peptidergic_neurons_ra$receptor))
```


Below we specify by the three specific neuron cell types for all groups after splitting into fibroblast subtypes.

```{r}
print('------------Peripheral sensory neurofilament neurons--------------')
specific_neuron_data_f1_peripheral_sensory_neurofilament_neurons <- specific_neuron_data_f1[(specific_neuron_data_f1$ligand_type == 'Peripheral sensory neurofilament neurons') | (specific_neuron_data_f1$receptor_type == 'Peripheral sensory neurofilament neurons'),]
specific_neuron_data_f2_peripheral_sensory_neurofilament_neurons <- specific_neuron_data_f2[(specific_neuron_data_f2$ligand_type == 'Peripheral sensory neurofilament neurons') | (specific_neuron_data_f2$receptor_type == 'Peripheral sensory neurofilament neurons'),]
specific_neuron_data_f3_peripheral_sensory_neurofilament_neurons <- specific_neuron_data_f3[(specific_neuron_data_f3$ligand_type == 'Peripheral sensory neurofilament neurons') | (specific_neuron_data_f3$receptor_type == 'Peripheral sensory neurofilament neurons'),]
specific_neuron_data_f4_peripheral_sensory_neurofilament_neurons <- specific_neuron_data_f4[(specific_neuron_data_f4$ligand_type == 'Peripheral sensory neurofilament neurons') | (specific_neuron_data_f4$receptor_type == 'Peripheral sensory neurofilament neurons'),]


print('------------Peripheral sensory non-peptidergic neurons--------------')
specific_neuron_data_f1_peripheral_sensory_nonpeptidergic_neurons <- specific_neuron_data_f1[(specific_neuron_data_f1$ligand_type == 'Peripheral sensory non-peptidergic neurons') | (specific_neuron_data_f1$receptor_type == 'Peripheral sensory non-peptidergic neurons'),]
specific_neuron_data_f2_peripheral_sensory_nonpeptidergic_neurons <- specific_neuron_data_f2[(specific_neuron_data_f2$ligand_type == 'Peripheral sensory non-peptidergic neurons') | (specific_neuron_data_f2$receptor_type == 'Peripheral sensory non-peptidergic neurons'),]
specific_neuron_data_f3_peripheral_sensory_nonpeptidergic_neurons <- specific_neuron_data_f3[(specific_neuron_data_f3$ligand_type == 'Peripheral sensory non-peptidergic neurons') | (specific_neuron_data_f3$receptor_type == 'Peripheral sensory non-peptidergic neurons'),]
specific_neuron_data_f4_peripheral_sensory_nonpeptidergic_neurons <- specific_neuron_data_f4[(specific_neuron_data_f4$ligand_type == 'Peripheral sensory non-peptidergic neurons') | (specific_neuron_data_f4$receptor_type == 'Peripheral sensory non-peptidergic neurons'),]

print('------------Peripheral sensory peptidergic neurons--------------')
specific_neuron_data_f1_peripheral_sensory_peptidergic_neurons <- specific_neuron_data_f1[(specific_neuron_data_f1$ligand_type == 'Peripheral sensory peptidergic neurons') | (specific_neuron_data_f1$receptor_type == 'Peripheral sensory peptidergic neurons'),]
specific_neuron_data_f2_peripheral_sensory_peptidergic_neurons <- specific_neuron_data_f2[(specific_neuron_data_f2$ligand_type == 'Peripheral sensory peptidergic neurons') | (specific_neuron_data_f2$receptor_type == 'Peripheral sensory peptidergic neurons'),]
specific_neuron_data_f3_peripheral_sensory_peptidergic_neurons <- specific_neuron_data_f3[(specific_neuron_data_f3$ligand_type == 'Peripheral sensory peptidergic neurons') | (specific_neuron_data_f3$receptor_type == 'Peripheral sensory peptidergic neurons'),]
specific_neuron_data_f4_peripheral_sensory_peptidergic_neurons <- specific_neuron_data_f4[(specific_neuron_data_f4$ligand_type == 'Peripheral sensory peptidergic neurons') | (specific_neuron_data_f4$receptor_type == 'Peripheral sensory peptidergic neurons'),]
```

Below we get statistics for a multitude of things for peripheral sensory neurofilament neurons.

```{r}
print('Order: Total Unique Ligands')
print('Total Unique Receptors')
print('mDRG Ligand to Synovial Receptor Connections')
print('mDRG Ligands')
print('mDRG Receptors')
print('Synovial Ligand to mDRG Receptor Connections')
print('Synovial Ligands')
print('Synovial Receptors')
print('-------------SC-F1------------------')
length(unique(all_neuron_data_f1$ligand))
length(unique(all_neuron_data_f1$receptor))
all_neuron_data_f1_mdrg = all_neuron_data_f1[all_neuron_data_f1$location == 'mDRG',]
nrow(all_neuron_data_f1_mdrg)
length(unique(all_neuron_data_f1_mdrg$ligand))
length(unique(all_neuron_data_f1_mdrg$receptor))
all_neuron_data_f1_ra = all_neuron_data_f1[all_neuron_data_f1$location == 'RA sc',]
nrow(all_neuron_data_f1_ra)
length(unique(all_neuron_data_f1_ra$ligand))
length(unique(all_neuron_data_f1_ra$receptor))

print('-------------SC-F2------------------')
length(unique(all_neuron_data_f2$ligand))
length(unique(all_neuron_data_f2$receptor))
all_neuron_data_f2_mdrg = all_neuron_data_f2[all_neuron_data_f2$location == 'mDRG',]
nrow(all_neuron_data_f2_mdrg)
length(unique(all_neuron_data_f2_mdrg$ligand))
length(unique(all_neuron_data_f2_mdrg$receptor))
all_neuron_data_f2_ra = all_neuron_data_f2[all_neuron_data_f2$location == 'RA sc',]
nrow(all_neuron_data_f2_ra)
length(unique(all_neuron_data_f2_ra$ligand))
length(unique(all_neuron_data_f2_ra$receptor))

print('-------------SC-F3------------------')
length(unique(all_neuron_data_f3$ligand))
length(unique(all_neuron_data_f3$receptor))
all_neuron_data_f3_mdrg = all_neuron_data_f3[all_neuron_data_f3$location == 'mDRG',]
nrow(all_neuron_data_f3_mdrg)
length(unique(all_neuron_data_f3_mdrg$ligand))
length(unique(all_neuron_data_f3_mdrg$receptor))
all_neuron_data_f3_ra = all_neuron_data_f3[all_neuron_data_f3$location == 'RA sc',]
nrow(all_neuron_data_f3_ra)
length(unique(all_neuron_data_f3_ra$ligand))
length(unique(all_neuron_data_f3_ra$receptor))

print('-------------SC-F4------------------')
length(unique(all_neuron_data_f4$ligand))
length(unique(all_neuron_data_f4$receptor))
all_neuron_data_f4_mdrg = all_neuron_data_f4[all_neuron_data_f4$location == 'mDRG',]
nrow(all_neuron_data_f4_mdrg)
length(unique(all_neuron_data_f4_mdrg$ligand))
length(unique(all_neuron_data_f4_mdrg$receptor))
all_neuron_data_f4_ra = all_neuron_data_f4[all_neuron_data_f4$location == 'RA sc',]
nrow(all_neuron_data_f4_ra)
length(unique(all_neuron_data_f4_ra$ligand))
length(unique(all_neuron_data_f4_ra$receptor))
```

# Circos Plots

## Specific Neuron Interactome

Below we create the interaction plot for F1 with the mDRG and RA scRNA for the specific neuron interactome.

```{r, fig.height=10,fig.width=10}
library(BioCircos)

# Chromosomes on which the mDRG points should be displayed
points_chromosomes_mdrg = c('mDRG', 'mDRG')
# Coordinates on which the mDRG points should be displayed (length.out = 8 because there are 8 unique genes in the mDRG from the ligand (left column) and receptor data (right column) in f1_data)
points_coordinates_mdrg = seq(3, 258, length.out = 8)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique ligands in F1 here
points_values_mdrg = 1:length(points_chromosomes_mdrg)

# Size of the outside of the circle for mdrg and ra
# Since there are uniquely 8 mdrg and 3 RA genes, we make the slices of the circle 360 - (8/11*360) = 261 for the mdrg and 360 - 261 for the receptor
myGenome = list("mDRG" = 261,
        "RA" = 99)

# Create circle for mDRG and RA identification to lie in
tracklist = BioCircosBackgroundTrack("myBackgroundTrack", minRadius = 0.99, maxRadius = 1.09,
  borderColors = "#AAAAAA", borderSize = 0.6, fillColors = "White") 

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_mdrg, points_coordinates_mdrg[1:7], 
  points_values_mdrg, colors = c("orange", "orange", "orange", "orange", "orange", "orange", "orange"), minRadius = 1.04, maxRadius = 1.04)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_mdrg, seq(3, 258, length.out = 8)[8], 
  points_values_mdrg, colors = c("blue"), size = 4, minRadius = 1.04, maxRadius = 1.04, shape = 'rect')

# Chromosomes on which the RA scRNA points should be displayed
points_chromosomes_ra = c('RA', 'RA')
# Coordinates on which the RA scRNA points should be displayed length.out = 11 because there are 3 unique genes in the RA scRNA from the ligand (left column) and receptor data (right column) in f1_data)
points_coordinates_ra = seq(3, 96, length.out = 3)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique RA scRNA genes in F1 here
points_values_ra = 1:length(points_chromosomes_ra)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ra, points_coordinates_ra[1:2], 
  points_values_ra, colors = c("blue", "blue"), minRadius = 1.04, maxRadius = 1.04, shape = 'rect', size = 4)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ra, points_coordinates_ra[3], 
  points_values_ra, colors = c("orange"), minRadius = 1.04, maxRadius = 1.04)

#tracklist = tracklist + BioCircosTextTrack("testText", 'Subtype F1 mDRG and RA Interactions', weight = "bold", size = '1em', x = -0.6, y = -1.25, displayGenomeBorder = TRUE)

links_chromosomes_1 = c('mDRG', 'mDRG', 'mDRG') # Chromosomes on which the links should start
links_chromosomes_2 = c('RA', 'RA', 'RA') # Chromosomes on which the links should end

# Coordinates to place start of links. The coordinates below correspond to the sequence given by seq(3, 258, length.out = 8) Their starting coordinate must be repeated for the number of connections between them and the number of RA genes they are connected to.
links_pos_1 = c(3, 39.43, 75.86, 112.29, 148.71, 185.14, 221.57, 258)
# Coordinates to place end of links. The coordinates below correspond to the sequence given by seq(3, 96, length.out = 3) but we have 8 values in total. Their ending coordinate must be repeated for the number of connections between them and the number of mdrg genes they are connected to.
links_pos_2 = c(3, 3, 3, 3, 3, 3, 49.5, 96)
# The labels are given at the end of the link. Therefore, we label the RA scRNA when starting at the mDRG for linking.
links_labels_12 = c('FGFR1', '', '', '', '', '', 'PTPRS', 'CTGF') # 8

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_1, links_pos_1,
  links_pos_1, links_chromosomes_2, links_pos_2, links_pos_2,
  maxRadius = 0.7, width = "0.01em", color = 'cyan', labelSize = "0.8em", labels = links_labels_12)

links_chromosomes_3 = c('RA', 'RA', 'RA') # Chromosomes on which the links should start
links_chromosomes_4 = c('mDRG', 'mDRG', 'mDRG') # Chromosomes on which the links should end

# Now we start at the RA scRNA
links_pos_3 = c(3, 3, 3, 3, 3, 3, 49.5, 96)
# Now we end at the mDRG
links_pos_4 = c(3, 39.43, 75.86, 112.29, 148.71, 185.14, 221.57, 258)
# These links are for the ligands
links_labels_34 = c("FGF1", "FGF11", "FGF9", "FGF12", "NCAM1", "PTN", "NTRK1")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_3, links_pos_3,
  links_pos_3, links_chromosomes_4, links_pos_4, links_pos_4,
  maxRadius = 0.7, width = "0.01em", color = 'cyan', labelSize = "0.8em", labels = links_labels_34, labelPadding = 3)

links_chromosomes_5 = c('mDRG', 'mDRG', 'mDRG') # Chromosomes on which the links should start
links_chromosomes_6 = c('RA', 'RA', 'RA') # Chromosomes on which the links should end
links_pos_5 = c(258)
links_pos_6 = c(96)
links_labels_56 = c("")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_5, links_pos_5,
  links_pos_5, links_chromosomes_6, links_pos_6, links_pos_6,
  maxRadius = 0.7, width = "0.01em", color = 'brown', labelSize = "0.5em", labels = links_labels_56)

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra = c('RA') # Chromosomes on which the arcs should be displayed
arcs_begin_ra = c(0)
arcs_end_ra = c(99)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra, arcs_begin_ra, arcs_end_ra, minRadius = 1.10, maxRadius = 1.15, colors = 'forestgreen') 

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_mdrg = c('mDRG') # Chromosomes on which the arcs should be displayed
arcs_begin_mdrg = c(0)
arcs_end_mdrg = c(261)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_mdrg, arcs_begin_mdrg, arcs_end_mdrg, minRadius = 1.15, maxRadius = 1.20, colors = 'deeppink3')

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'CD34+ sublining fibroblasts (SC-F1)', x = -0.6, y = -1.25, size = "1em",
 # weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'mDRG', x = 0.55, y = 1.15, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'RA', x = -0.7, y = -1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

BioCircos(tracklist, genome = myGenome, genomeFillColor = c('white', 'white'), chrPad = 0.05, displayGenomeBorder = FALSE, yChr =  FALSE,
  genomeTicksDisplay = FALSE,  genomeLabelTextSize = 0, genomeLabelDy = 0, width = 650, height = 650)
```


Below we create the interaction plot for F2 with the mDRG and RA scRNA for the specific neuron interactome.

```{r, fig.height=10,fig.width=10}
library(BioCircos)

# Chromosomes on which the mDRG points should be displayed
points_chromosomes_mdrg = c('mDRG', 'mDRG')
# Coordinates on which the mDRG points should be displayed (length.out = 11 because there are 11 unique genes in the mDRG from the ligand (left column) and receptor data (right column) in f1_data)
points_coordinates_mdrg = seq(3, 261, length.out = 11)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique ligands in F1 here
points_values_mdrg = 1:length(points_chromosomes_mdrg)

# Size of the outside of the circle for mdrg and ra
# Since there are uniquely 8 mdrg and 3 RA genes, we make the slices of the circle 360 - (11/15*360) = 264 for the mdrg and 360 - 264 for the receptor
myGenome = list("mDRG" = 264,
        "RA" = 96)

# Create circle for mDRG and RA identification to lie in
tracklist = BioCircosBackgroundTrack("myBackgroundTrack", minRadius = 0.99, maxRadius = 1.09,
  borderColors = "#AAAAAA", borderSize = 0.6, fillColors = "White") 

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_mdrg, points_coordinates_mdrg[1:8], 
  points_values_mdrg, colors = c("orange", "orange", "orange", "orange", "orange", "orange", "orange", "orange"), minRadius = 1.04, maxRadius = 1.04)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_mdrg, seq(3, 261, length.out = 11)[9:11], 
  points_values_mdrg, colors = c("blue", "blue", "blue"), size = 4, minRadius = 1.04, maxRadius = 1.04, shape = 'rect')

# Chromosomes on which the RA scRNA points should be displayed
points_chromosomes_ra = c('RA', 'RA')
# Coordinates on which the RA scRNA points should be displayed length.out = 11 because there are 3 unique genes in the RA scRNA from the ligand (left column) and receptor data (right column) in f1_data)
points_coordinates_ra = seq(3, 93, length.out = 4)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique RA scRNA genes in F1 here
points_values_ra = 1:length(points_chromosomes_ra)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ra, points_coordinates_ra[1:2], 
  points_values_ra, colors = c("blue", "blue"), minRadius = 1.04, maxRadius = 1.04, shape = 'rect', size = 4)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ra, points_coordinates_ra[3:4], 
  points_values_ra, colors = c("orange", "orange"), minRadius = 1.04, maxRadius = 1.04)

#tracklist = tracklist + BioCircosTextTrack("testText", 'Subtype F1 mDRG and RA Interactions', weight = "bold", size = '1em', x = -0.6, y = -1.25, displayGenomeBorder = TRUE)

links_chromosomes_1 = c('mDRG', 'mDRG', 'mDRG') # Chromosomes on which the links should start
links_chromosomes_2 = c('RA', 'RA', 'RA') # Chromosomes on which the links should end

# Coordinates to place start of links. The coordinates below correspond to the sequence given by seq(3, 261, length.out = 11) Their starting coordinate must be repeated for the number of connections between them and the number of RA genes they are connected to.
links_pos_1 = c(3, 28.8, 54.6, 80.4, 106.2, 132.0, 157.8, 183.6, 209.4, 235.2, 261.0)
# Coordinates to place end of links. The coordinates below correspond to the sequence given by seq(3, 93, length.out = 4) but we have 11 values in total. Their ending coordinate must be repeated for the number of connections between them and the number of mdrg genes they are connected to.
links_pos_2 = c(3, 33, 33, 33, 3, 33, 3, 33, 63, 93, 63)
# The labels are given at the end of the link. Therefore, we label the RA scRNA when starting at the mDRG for linking.
links_labels_12 = c("ITGB5", 'FGFR1', '', '', '', '', '', '', 'EFNA5', 'CTGF', '') # 11

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_1, links_pos_1,
  links_pos_1, links_chromosomes_2, links_pos_2, links_pos_2,
  maxRadius = 0.7, width = "0.01em", color = 'cyan', labelSize = "0.8em", labels = links_labels_12)

links_chromosomes_3 = c('RA', 'RA', 'RA') # Chromosomes on which the links should start
links_chromosomes_4 = c('mDRG', 'mDRG', 'mDRG') # Chromosomes on which the links should end

# Now we start at the RA scRNA
links_pos_3 = c(3, 33, 33, 33, 3, 33, 3, 33, 63, 93, 63)
# Now we end at the mDRG
links_pos_4 = c(3, 28.8, 54.6, 80.4, 106.2, 132.0, 157.8, 183.6, 209.4, 235.2, 261.0)
# These links are for the ligands
links_labels_34 = c("EDIL3", "FGF1", "FGF11", "FGF9", "LTBP3", "FGF12", "ITGB3BP", "NCAM1", "EPHA10", "NTRK1", "EPHA5")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_3, links_pos_3,
  links_pos_3, links_chromosomes_4, links_pos_4, links_pos_4,
  maxRadius = 0.7, width = "0.01em", color = 'cyan', labelSize = "0.8em", labels = links_labels_34, labelPadding = 3)

links_chromosomes_5 = c('mDRG', 'mDRG', 'mDRG') # Chromosomes on which the links should start
links_chromosomes_6 = c('RA', 'RA', 'RA') # Chromosomes on which the links should end
links_pos_5 = c(209.4, 235.2, 261.0)
links_pos_6 = c(63, 93, 63)
links_labels_56 = c("", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_5, links_pos_5,
  links_pos_5, links_chromosomes_6, links_pos_6, links_pos_6,
  maxRadius = 0.7, width = "0.01em", color = 'brown', labelSize = "0.5em", labels = links_labels_56)

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra = c('RA') # Chromosomes on which the arcs should be displayed
arcs_begin_ra = c(0)
arcs_end_ra = c(96)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra, arcs_begin_ra, arcs_end_ra, minRadius = 1.10, maxRadius = 1.15, colors = 'forestgreen') 

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_mdrg = c('mDRG') # Chromosomes on which the arcs should be displayed
arcs_begin_mdrg = c(0)
arcs_end_mdrg = c(264)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_mdrg, arcs_begin_mdrg, arcs_end_mdrg, minRadius = 1.15, maxRadius = 1.20, colors = 'deeppink3')

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'HLA-DRAhi sublining fibroblasts (SC-F2)', x = -0.6, y = -1.25, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'mDRG', x = 0.55, y = 1.15, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'RA', x = -0.7, y = -1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

BioCircos(tracklist, genome = myGenome, genomeFillColor = c('white', 'white'), chrPad = 0.05, displayGenomeBorder = FALSE, yChr =  FALSE,
  genomeTicksDisplay = FALSE,  genomeLabelTextSize = 0, genomeLabelDy = 0, width = 650, height = 650)
```

Below we create the interaction plot for F3 with the mDRG and RA scRNA for the specific neuron interactome.

```{r, fig.height=10,fig.width=10}
library(BioCircos)

# Chromosomes on which the mDRG points should be displayed
points_chromosomes_mdrg = c('mDRG', 'mDRG')
# Coordinates on which the mDRG points should be displayed (length.out = 11 because there are 11 unique genes in the mDRG from the ligand (left column) and receptor data (right column) in f1_data)
points_coordinates_mdrg = seq(3, 261, length.out = 11)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique ligands in F1 here
points_values_mdrg = 1:length(points_chromosomes_mdrg)

# Size of the outside of the circle for mdrg and ra
# Since there are uniquely 8 mdrg and 3 RA genes, we make the slices of the circle 360 - (11/15*360) = 264 for the mdrg and 360 - 264 for the receptor
myGenome = list("mDRG" = 264,
        "RA" = 96)

# Create circle for mDRG and RA identification to lie in
tracklist = BioCircosBackgroundTrack("myBackgroundTrack", minRadius = 0.99, maxRadius = 1.09,
  borderColors = "#AAAAAA", borderSize = 0.6, fillColors = "White") 

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_mdrg, points_coordinates_mdrg[1:10], 
  points_values_mdrg, colors = c("orange", "orange", "orange", "orange", "orange", "orange", "orange", "orange", "orange", "orange"), minRadius = 1.04, maxRadius = 1.04)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_mdrg, seq(3, 261, length.out = 11)[11], 
  points_values_mdrg, colors = c("blue"), size = 4, minRadius = 1.04, maxRadius = 1.04, shape = 'rect')

# Chromosomes on which the RA scRNA points should be displayed
points_chromosomes_ra = c('RA', 'RA')
# Coordinates on which the RA scRNA points should be displayed length.out = 11 because there are 3 unique genes in the RA scRNA from the ligand (left column) and receptor data (right column) in f1_data)
points_coordinates_ra = seq(3, 93, length.out = 4)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique RA scRNA genes in F1 here
points_values_ra = 1:length(points_chromosomes_ra)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ra, points_coordinates_ra[1:3], 
  points_values_ra, colors = c("blue", "blue", "blue"), minRadius = 1.04, maxRadius = 1.04, shape = 'rect', size = 4)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ra, points_coordinates_ra[4], 
  points_values_ra, colors = c("orange"), minRadius = 1.04, maxRadius = 1.04)

#tracklist = tracklist + BioCircosTextTrack("testText", 'Subtype F1 mDRG and RA Interactions', weight = "bold", size = '1em', x = -0.6, y = -1.25, displayGenomeBorder = TRUE)

links_chromosomes_1 = c('mDRG', 'mDRG', 'mDRG') # Chromosomes on which the links should start
links_chromosomes_2 = c('RA', 'RA', 'RA') # Chromosomes on which the links should end

# Coordinates to place start of links. The coordinates below correspond to the sequence given by seq(3, 261, length.out = 11) Their starting coordinate must be repeated for the number of connections between them and the number of RA genes they are connected to.
links_pos_1 = c(3, 28.8, 54.6, 80.4, 106.2, 132.0, 157.8, 183.6, 209.4, 235.2, 261.0)
# Coordinates to place end of links. The coordinates below correspond to the sequence given by seq(3, 93, length.out = 4) but we have 11 values in total. Their ending coordinate must be repeated for the number of connections between them and the number of mdrg genes they are connected to.
links_pos_2 = c(3, 33, 33, 33, 3, 63, 63, 33, 3, 33, 93)
# The labels are given at the end of the link. Therefore, we label the RA scRNA when starting at the mDRG for linking.
links_labels_12 = c("ITGB5", 'FGFR1', '', '', '', "CDON", '', '', '', '', 'CTGF') # 11

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_1, links_pos_1,
  links_pos_1, links_chromosomes_2, links_pos_2, links_pos_2,
  maxRadius = 0.7, width = "0.01em", color = 'cyan', labelSize = "0.8em", labels = links_labels_12)

links_chromosomes_3 = c('RA', 'RA', 'RA') # Chromosomes on which the links should start
links_chromosomes_4 = c('mDRG', 'mDRG', 'mDRG') # Chromosomes on which the links should end

# Now we start at the RA scRNA
links_pos_3 = c(3, 33, 33, 33, 3, 63, 63, 33, 3, 33, 93)
# Now we end at the mDRG
links_pos_4 = c(3, 28.8, 54.6, 80.4, 106.2, 132.0, 157.8, 183.6, 209.4, 235.2, 261.0)
# These links are for the ligands
links_labels_34 = c("EDIL3", "FGF1", "FGF11", "FGF9", "LTBP3", "SHH", "DHH", "FGF12", "ITGB3BP", "NCAM1", "NTRK1")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_3, links_pos_3,
  links_pos_3, links_chromosomes_4, links_pos_4, links_pos_4,
  maxRadius = 0.7, width = "0.01em", color = 'cyan', labelSize = "0.8em", labels = links_labels_34, labelPadding = 3)

links_chromosomes_5 = c('mDRG', 'mDRG', 'mDRG') # Chromosomes on which the links should start
links_chromosomes_6 = c('RA', 'RA', 'RA') # Chromosomes on which the links should end
links_pos_5 = c(261)
links_pos_6 = c(93)
links_labels_56 = c("")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_5, links_pos_5,
  links_pos_5, links_chromosomes_6, links_pos_6, links_pos_6,
  maxRadius = 0.7, width = "0.01em", color = 'brown', labelSize = "0.5em", labels = links_labels_56)

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra = c('RA') # Chromosomes on which the arcs should be displayed
arcs_begin_ra = c(0)
arcs_end_ra = c(96)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra, arcs_begin_ra, arcs_end_ra, minRadius = 1.10, maxRadius = 1.15, colors = 'forestgreen') 

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_mdrg = c('mDRG') # Chromosomes on which the arcs should be displayed
arcs_begin_mdrg = c(0)
arcs_end_mdrg = c(264)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_mdrg, arcs_begin_mdrg, arcs_end_mdrg, minRadius = 1.15, maxRadius = 1.20, colors = 'deeppink3')

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'DKK3+ sublining fibroblasts (SC-F3)', x = -0.6, y = -1.25, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'mDRG', x = 0.55, y = 1.15, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'RA', x = -0.7, y = -1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

BioCircos(tracklist, genome = myGenome, genomeFillColor = c('white', 'white'), chrPad = 0.05, displayGenomeBorder = FALSE, yChr =  FALSE,
  genomeTicksDisplay = FALSE,  genomeLabelTextSize = 0, genomeLabelDy = 0, width = 650, height = 650)
```

Below we create the interaction plot for F4 with the mDRG and RA scRNA for the specific neuron interactome.

```{r, fig.height=10,fig.width=10}
library(BioCircos)

# Chromosomes on which the mDRG points should be displayed
points_chromosomes_mdrg = c('mDRG', 'mDRG')
# Coordinates on which the mDRG points should be displayed (length.out = 11 because there are 26 unique genes in the mDRG from the ligand (left column) and receptor data (right column) in f1_data)
points_coordinates_mdrg = seq(3, 250, length.out = 26)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique ligands in F1 here
points_values_mdrg = 1:length(points_chromosomes_mdrg)

# Size of the outside of the circle for mdrg and ra
# Since there are uniquely 8 mdrg and 3 RA genes, we make the slices of the circle 360 - (26/37*360) = 253 for the mdrg and 360 - 253 for the receptor
myGenome = list("mDRG" = 253,
        "RA" = 107)

# Create circle for mDRG and RA identification to lie in
tracklist = BioCircosBackgroundTrack("myBackgroundTrack", minRadius = 0.99, maxRadius = 1.14,
  borderColors = "#AAAAAA", borderSize = 0.6, fillColors = "White") 

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_mdrg, points_coordinates_mdrg[1:14], 
  points_values_mdrg, colors = c("orange", "orange", "orange", "orange", "orange", "orange", "orange", "orange", "orange", "orange", "orange", "orange", "orange",  "orange"), minRadius = 1.04, maxRadius = 1.04)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_mdrg, seq(3, 250, length.out = 26)[15:26], 
  points_values_mdrg, colors = c("blue", "blue", "blue", "blue", "blue", "blue", "blue", "blue", "blue", "blue", "blue", "blue"), size = 4, minRadius = 1.09, maxRadius = 1.09, shape = 'rect')

# Chromosomes on which the RA scRNA points should be displayed
points_chromosomes_ra = c('RA', 'RA')
# Coordinates on which the RA scRNA points should be displayed length.out = 11 because there are 3 unique genes in the RA scRNA from the ligand (left column) and receptor data (right column) in f1_data)
points_coordinates_ra = seq(3, 104, length.out = 11)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique RA scRNA genes in F1 here
points_values_ra = 1:length(points_chromosomes_ra)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ra, points_coordinates_ra[1:6], 
  points_values_ra, colors = c("blue", "blue", "blue", "blue", "blue", "blue"), minRadius = 1.09, maxRadius = 1.09, shape = 'rect', size = 4)

# Make ligand identifier for APP gene only
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ra, points_coordinates_ra[1], 
  points_values_ra, colors = c("orange"), minRadius = 1.04, maxRadius = 1.04)
# Make ligand identifier for rest of ligands
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ra, points_coordinates_ra[7:11], 
  points_values_ra, colors = c("orange", "orange", 'orange', "orange", "orange"), minRadius = 1.04, maxRadius = 1.04)

#tracklist = tracklist + BioCircosTextTrack("testText", 'Subtype F1 mDRG and RA Interactions', weight = "bold", size = '1em', x = -0.6, y = -1.25, displayGenomeBorder = TRUE)

links_chromosomes_1 = c('mDRG', 'mDRG', 'mDRG') # Chromosomes on which the links should start
links_chromosomes_2 = c('RA', 'RA', 'RA') # Chromosomes on which the links should end

# Coordinates to place start of links. The coordinates below correspond to the sequence given by seq(3, 250, length.out = 26) Their starting coordinate must be repeated for the number of connections between them and the number of RA genes they are connected to.
links_pos_1 = c(3.00, 12.88, 22.76, 32.64, 42.52, 52.40, 62.28, 72.16, 82.04, 91.92, 101.80, 111.68, 121.56, 3, 131.44, 141.32, 151.20, 161.08, 170.96, 180.84, 190.72, 200.60, 210.48, 141.32, 220.36, 230.24, 240.12, 250.00)
# Coordinates to place end of links. The coordinates below correspond to the sequence given by seq(3, 104, length.out = 11) but we have 37 values in total. Their ending coordinate must be repeated for the number of connections between them and the number of mdrg genes they are connected to.
links_pos_2 = c(3, 3, 13.1, 23.2, 23.2, 23.2, 13.1, 33.3, 43.4, 43.4, 23.2, 13.1, 23.2, 3, 53.5, 3, 63.6, 73.7, 83.8, 3, 3, 93.9, 93.9, 3, 3, 104, 63.6, 93.9)
# The labels are given at the end of the link. Therefore, we label the RA scRNA when starting at the mDRG for linking.
links_labels_12 = c("APP", "", "ITGB5", 'FGFR1', '', '', '', "ANK3", 'CDON', '', '', '', '', '', "PTPRS", '', 'EFNA5', 'NTN4', 'SEMA6D', '', '', 'HBEGF', '', '' ,'', 'CTGF', '', '') # 28

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_1, links_pos_1,
  links_pos_1, links_chromosomes_2, links_pos_2, links_pos_2,
  maxRadius = 0.7, width = "0.01em", color = 'cyan', labelSize = "0.8em", labels = links_labels_12)

links_chromosomes_3 = c('RA', 'RA', 'RA') # Chromosomes on which the links should start
links_chromosomes_4 = c('mDRG', 'mDRG', 'mDRG') # Chromosomes on which the links should end

# Now we start at the RA scRNA
links_pos_3 = c(3, 3, 13.1, 23.2, 23.2, 23.2, 13.1, 33.3, 43.4, 43.4, 23.2, 13.1, 23.2, 3, 53.5, 3, 63.6, 73.7, 83.8, 3, 3, 93.9, 93.9, 3, 3, 104, 63.6, 93.9)
# Now we end at the mDRG
links_pos_4 = c(3.00, 12.88, 22.76, 32.64, 42.52, 52.40, 62.28, 72.16, 82.04, 91.92, 101.80, 111.68, 121.56, 3, 131.44, 141.32, 151.20, 161.08, 170.96, 180.84, 190.72, 200.60, 210.48, 141.32, 220.36, 230.24, 240.12, 250.00)
# These links are for the ligands
links_labels_34 = c("CLSTN1", "CLSTN3", "EDIL3", "FGF1", "FGF11", "FGF9", "LTBP3", "NFASC", "SHH", "DHH", "FGF12", "ITGB3BP", "NCAM1", '', 'PTN', "NGFR", "EPHA10", "UNC5A", "PLXNA1", "CAV1", "SLC45A3", "CD44", "CD82", "", "TNFRSF21", "NTRK1", "EPHA5", 'CD9')

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_3, links_pos_3,
  links_pos_3, links_chromosomes_4, links_pos_4, links_pos_4,
  maxRadius = 0.7, width = "0.01em", color = 'cyan', labelSize = "0.8em", labels = links_labels_34, labelPadding = 3)

links_chromosomes_5 = c('mDRG', 'mDRG', 'mDRG') # Chromosomes on which the links should start
links_chromosomes_6 = c('RA', 'RA', 'RA') # Chromosomes on which the links should end
links_pos_5 = c(141.32, 151.20, 161.08, 170.96, 180.84, 190.72, 200.60, 210.48, 141.32, 220.36, 230.24, 240.12, 250.00)
links_pos_6 = c(3, 63.6, 73.7, 83.8, 3, 3, 93.9, 93.9, 3, 3, 104, 63.6, 93.9)
links_labels_56 = c("", "", "", "", "", "", "", "", "", "", "", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_5, links_pos_5,
  links_pos_5, links_chromosomes_6, links_pos_6, links_pos_6,
  maxRadius = 0.7, width = "0.01em", color = 'brown', labelSize = "0.5em", labels = links_labels_56)

# Make connection for NTN4 wider
links_chromosomes_7 = c('mDRG') # Chromosomes on which the links should start
links_chromosomes_8 = c('RA') # Chromosomes on which the links should end
links_pos_7 = c(161.08)
links_pos_8 = c(73.7)
links_labels_78 = c("")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_7, links_pos_7,
  links_pos_7, links_chromosomes_8, links_pos_8, links_pos_8,
  maxRadius = 0.7, width = "0.1em", color = 'brown', labelSize = "0.5em", labels = links_labels_78)

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra = c('RA') # Chromosomes on which the arcs should be displayed
arcs_begin_ra = c(0)
arcs_end_ra = c(107)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra, arcs_begin_ra, arcs_end_ra, minRadius = 1.15, maxRadius = 1.20, colors = 'forestgreen') 

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_mdrg = c('mDRG') # Chromosomes on which the arcs should be displayed
arcs_begin_mdrg = c(0)
arcs_end_mdrg = c(253)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_mdrg, arcs_begin_mdrg, arcs_end_mdrg, minRadius = 1.20, maxRadius = 1.25, colors = 'deeppink3')

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'CD55+ lining fibroblasts (SC-F4)', x = -0.6, y = -1.30, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'mDRG', x = 0.60, y = 1.20, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'RA', x = -0.75, y = -1.05, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

BioCircos(tracklist, genome = myGenome, genomeFillColor = c('white', 'white'), chrPad = 0.05, displayGenomeBorder = FALSE, yChr =  FALSE,
  genomeTicksDisplay = FALSE,  genomeLabelTextSize = 0, genomeLabelDy = 0, width = 650, height = 650)
```

Below we create the interactome plot from the mouse DRG ligand to the human RA receptors.

```{r, fig.height=10,fig.width=10}
library(BioCircos)

# Chromosomes on which the Receptor points should be displayed
points_chromosomes_ligand = c('Ligand', 'Ligand')
# Coordinates on which the Receptor points should be displayed (length.out = 6 because there are 6 unique genes in the receptor column
points_coordinates_ligand = seq(3, 105, length.out = 6)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique ligands
points_values_ligand = 1:length(points_coordinates_ligand)

# Size of the outside of the circle for ligands and receptors
# Since there are uniquely 6 ligands and 14 receptor genes, we make the slices of the circle 6/20*360 = 108 for the receptors and 360 - 108 for the ligands
myGenome = list("Ligand" = 108,
        "Receptor" = 252)

# Create circle for cell type identification to lie in
tracklist = BioCircosBackgroundTrack("myBackgroundTrack", minRadius = 0.78, maxRadius = 1.13,
  borderColors = "#AAAAAA", borderSize = 0.6, fillColors = "White") 

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("blue", "white", "white", "white", "white", "blue"), minRadius = 1.11, maxRadius = 1.11)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("green", "green", "white", "white", "white", "white"), minRadius = 1.01, maxRadius = 1.01)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("orange", "orange", "orange", "white", "white", "white"), minRadius = 0.91, maxRadius = 0.91)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("red", "red", "red", "red", "red", "red"), minRadius = 0.81, maxRadius = 0.81)

# Chromosomes on which the ligand should be displayed
points_chromosomes_receptor = c('Receptor', 'Receptor')
# Coordinates on which the ligand should be displayed length.out = 6 
points_coordinates_receptor = seq(3, 249, length.out = 14)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique receptors
points_values_receptor = 1:length(points_chromosomes_receptor)

# Peripheral sensory peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "magenta", "white", "white", "white", "white", "white", "magenta"), minRadius = 0.80, maxRadius = 0.80)

# Peripheral sensory neurofilament neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("gold", "gold", "gold", "gold", "gold", "gold", "gold", "gold", "gold", "white", "white", 'white', "white", "white"), minRadius = 0.95, maxRadius = 0.95)

# Peripheral sensory non-peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "white", "darkgrey", "darkgrey", "darkgrey", "darkgrey", "white"), minRadius = 1.10, maxRadius = 1.10)

#tracklist = tracklist + BioCircosTextTrack("testText", 'mDRG to Synovial Fibroblast', weight = "bold", size = '1em', x = -0.46, y = -1.25, displayGenomeBorder = TRUE)

links_chromosomes_1 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should start
links_chromosomes_2 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should end

# Coordinates to place start of links. The coordinates below correspond to the sequence given by seq(3, 105, length.out = 6) Their starting coordinate must be repeated for the number of connections between them and the number of ligands they are connected to.
links_pos_1 = c(3, 3, 3, 23.4, 3, 3, 3, 23.4, 23.4, 3, 3, 3, 23.4, 43.8, 64.2, 64.2, 23.4, 3, 3, 3, 23.4, 84.6, 43.8, 3, 3, 3, 23.4, 3, 43.8, 3, 23.4, 3, 43.8, 3, 23.4, 3, 105, 64.2, 105)
# Coordinates to place end of links. The coordinates below correspond to the sequence given by seq(3, 249, length.out = 14)
links_pos_2 = c(3, 21.92, 40.85, 59.77, 3, 21.92, 40.85, 78.69, 59.77, 3, 21.92, 40.85, 78.69, 97.62, 116.54, 135.46, 59.77, 3, 21.92, 40.85, 78.69, 154.38, 97.62, 173.31, 192.23, 173.31, 211.15, 192.23, 230.08, 173.31, 211.15, 192.23, 230.08, 173.31, 211.15, 192.23, 249, 135.46, 249)
# The labels are given at the end of the link. 
links_labels_12 = c("FGF1", "FGF11", "FGF9", "EDIL3", "", "", "", "LTBP3", "", "", "", "", "", "SHH", "CLSTN1", "CLSTN3", "", "", "", "", "", "NFASC", "", "FGF12", "NCAM1", "", "ITGB3BP", "", "DHH", "", "", "", "", "", "", "", "PTN", "", "") 

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_1, links_pos_1,
  links_pos_1, links_chromosomes_2, links_pos_2, links_pos_2,
  maxRadius = 0.5, width = "0.01em", color = 'brown', labelSize = "0.75em", labels = links_labels_12)

links_chromosomes_3 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should start
links_chromosomes_4 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should end

# Now we start at the ligands
links_pos_3 = c(3, 21.92, 40.85, 59.77, 3, 21.92, 40.85, 78.69, 59.77, 3, 21.92, 40.85, 78.69, 97.62, 116.54, 135.46, 59.77, 3, 21.92, 40.85, 78.69, 154.38, 97.62, 173.31, 192.23, 173.31, 211.15, 192.23, 230.08, 173.31, 211.15, 192.23, 230.08, 173.31, 211.15, 192.23, 249, 135.46, 249)
# Now we end at the receptors
links_pos_4 = c(3, 3, 3, 23.4, 3, 3, 3, 23.4, 23.4, 3, 3, 3, 23.4, 43.8, 64.2, 64.2, 23.4, 3, 3, 3, 23.4, 84.6, 43.8, 3, 3, 3, 23.4, 3, 43.8, 3, 23.4, 3, 43.8, 3, 23.4, 3, 105, 64.2, 105)
# These links are for the receptors
links_labels_34 = c('FGFR1', "", "", "ITGB5", "", "", "", "", "", "", "", "", "", "CDON", "APP", "", "", "", "", "", "", "ANK3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "PTPRS", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_3, links_pos_3,
  links_pos_3, links_chromosomes_4, links_pos_4, links_pos_4,
  maxRadius = 0.5, width = "0.01em", color = 'brown', labelSize = "0.75em", labels = links_labels_34, labelPadding = 3)

#links_chromosomes_5 = c('ligand', 'hDRG', 'hDRG') # Chromosomes on which the links should start
#links_chromosomes_6 = c('RA', 'RA', 'RA') # Chromosomes on which the links should end
#links_pos_5 = c(180.08, 187.46, 194.84, 202.22, 209.59, 216.97, 224.35, 231.73, 239.11, 216.97, 246.49, 253.86, 261.24, 268.62, 276)
#links_pos_6 = c(48, 55.5, 55.5, 55.5, 55.5, 55.5, 55.5, 55.5, 63, 63, 63, 63, 70.5, 70.5, 78)
#links_labels_56 = c("", "", '', '', '', '', '', '', "", '', '', '', "", '', "")

#tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_5, links_pos_5,
#  links_pos_5, links_chromosomes_6, links_pos_6, links_pos_6,
#  maxRadius = 0.5, width = "0.01em", color = 'brown', labelSize = "0.5em", labels = links_labels_56)

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra = c('Ligand') # Chromosomes on which the arcs should be displayed
arcs_begin_ra = c(0)
arcs_end_ra = c(108)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra, arcs_begin_ra, arcs_end_ra, minRadius = 1.15, maxRadius = 1.20, colors = 'forestgreen') 

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_hdrg = c('Receptor') # Chromosomes on which the arcs should be displayed
arcs_begin_hdrg = c(0)
arcs_end_hdrg = c(252)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_hdrg, arcs_begin_hdrg, arcs_end_hdrg, minRadius = 1.15, maxRadius = 1.20, colors = 'deeppink3')

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Mouse DRG to Human RA - Specific Neurons', x = -0.6, y = -1.30, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Receptor', x = 0.75, y = -1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Ligand', x = -0.95, y = -1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes, arcs_begin, arcs_end, minRadius = 0.85, maxRadius = 0.99, colors = 'forestgreen') 

BioCircos(tracklist, genome = myGenome, genomeFillColor = c('white', 'white'), chrPad = 0.05, displayGenomeBorder = FALSE, yChr =  FALSE,
  genomeTicksDisplay = FALSE,  genomeLabelTextSize = 0, genomeLabelDy = 0, width = 650, height = 650)
```

Below we make a legend for the fibroblast subtypes.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (2)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (2)', 'DKK3+ sublining fibroblasts (SC-F3) (3)', 'CD55+ lining fibroblasts (SC-F4) (6)'), pch=16, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes", at=0.24, cex=2)
```

Below we plot the legend for the cell types.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Peripheral Sensory Non-peptidergic Neurons (4)', 'Peripheral Sensory Neurofilament Neurons (9)', 'Peripheral Sensory Peptidergic Neurons (2)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('darkgrey', 'gold', 'magenta'))
mtext("mDRG Neuronal Cell Types", at=0.19, cex=1.1)
```

Below we create the same interactome plot from the human RA receptors to the mouse DRG ligands.

```{r, fig.height=10,fig.width=10}
library(BioCircos)

# Chromosomes on which the Receptor points should be displayed
points_chromosomes_ligand = c('Ligand', 'Ligand')
# Coordinates on which the Receptor points should be displayed (length.out = 6 because there are 6 unique genes in the Ligand column
points_coordinates_ligand = seq(3, 117, length.out = 6)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique ligands
points_values_ligand = 1:length(points_coordinates_ligand)

# Size of the outside of the circle for ligands and receptors
# Since there are uniquely 6 ligands and 12 receptor genes, we make the slices of the circle 12/18*360 = 240 for the receptors and 360 - 240 for the ligands
myGenome = list("Ligand" = 120, "Receptor" = 240)

# Create circle for cell type identification to lie in
tracklist = BioCircosBackgroundTrack("myBackgroundTrack", minRadius = 0.78, maxRadius = 1.13,
  borderColors = "#AAAAAA", borderSize = 0.6, fillColors = "White") 

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("blue", "white", "white", "white", "white", "white"), minRadius = 1.11, maxRadius = 1.11)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("green", "green", "white", "white", "white", "white"), minRadius = 1.01, maxRadius = 1.01)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("orange", "white", "white", "white", "white", "white"), minRadius = 0.91, maxRadius = 0.91)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("red", "red", "red", "red", "red", "red"), minRadius = 0.81, maxRadius = 0.81)

# Chromosomes on which the ligand should be displayed
points_chromosomes_receptor = c('Receptor', 'Receptor')
# Coordinates on which the ligand should be displayed length.out = 12 
points_coordinates_receptor = seq(3, 237, length.out = 12)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique receptors
points_values_receptor = 1:length(points_chromosomes_receptor)

# Peripheral sensory peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("magenta", "white", "magenta", "magenta", "white", "white", "white", "white", "white", "white", "magenta", "magenta"), minRadius = 0.80, maxRadius = 0.80)

# Peripheral sensory neurofilament neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "gold", "white", "gold", "gold", "gold", "white", "white", "white", "white", "white", "white"), minRadius = 0.95, maxRadius = 0.95)

# Peripheral sensory non-peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "darkgray", "darkgray", "darkgray", "darkgray", "white", "white"), minRadius = 1.10, maxRadius = 1.10)

#tracklist = tracklist + BioCircosTextTrack("testText", 'mDRG to Synovial Fibroblast', weight = "bold", size = '1em', x = -0.46, y = -1.25, displayGenomeBorder = TRUE)

links_chromosomes_1 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should start
links_chromosomes_2 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should end

# Coordinates to place start of links. The coordinates below correspond to the sequence given by seq(3, 237, length.out = 12) Their starting coordinate must be repeated for the number of connections between them and the number of ligands they are connected to.
links_pos_1 = c(3, 24.27, 3, 45.55, 3, 66.82, 24.27, 88.09, 109.36, 130.64, 151.91, 173.18, 194.45, 66.82, 215.73, 3, 45.55, 237)
# Coordinates to place end of links. The coordinates below correspond to the sequence given by seq(3, 237, length.out = 12)
links_pos_2 = c(3, 25.8, 3, 25.8, 3, 48.6, 25.8, 71.4, 94.2, 48.6, 48.6, 117, 117, 48.6, 48.6, 3, 25.8, 117)
# The labels are given at the end of the link. 
links_labels_12 = c("CTGF", "EFNA5", "", "", "", "APP", "", "NTN4", "SEMA6D", "", "", "HBEGF", "", "", "", "", "", "") 

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_1, links_pos_1,
  links_pos_1, links_chromosomes_2, links_pos_2, links_pos_2,
  maxRadius = 0.5, width = "0.01em", color = 'cyan', labelSize = "0.75em", labels = links_labels_12)

links_chromosomes_3 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should start
links_chromosomes_4 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should end

# Now we start at the ligands
links_pos_3 = c(3, 25.8, 3, 25.8, 3, 48.6, 25.8, 71.4, 94.2, 48.6, 48.6, 117, 117, 48.6, 48.6, 3, 25.8, 117)
# Now we end at the receptors
links_pos_4 = c(3, 24.27, 3, 45.55, 3, 66.82, 24.27, 88.09, 109.36, 130.64, 151.91, 173.18, 194.45, 66.82, 215.73, 3, 45.55, 237)
# These links are for the receptors
links_labels_34 = c('NTRK1', "EPHA10", "", "EPHA5", "", "NGFR", "", "UNC5A", "PLXNA1", "CAV1", "SLC45A3", "CD44", "CD82", "", "TNFRSF21", "", "", "CD9")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_3, links_pos_3,
  links_pos_3, links_chromosomes_4, links_pos_4, links_pos_4,
  maxRadius = 0.5, width = "0.01em", color = 'cyan', labelSize = "0.75em", labels = links_labels_34, labelPadding = 3)

#links_chromosomes_5 = c('ligand', 'hDRG', 'hDRG') # Chromosomes on which the links should start
#links_chromosomes_6 = c('RA', 'RA', 'RA') # Chromosomes on which the links should end
#links_pos_5 = c(180.08, 187.46, 194.84, 202.22, 209.59, 216.97, 224.35, 231.73, 239.11, 216.97, 246.49, 253.86, 261.24, 268.62, 276)
#links_pos_6 = c(48, 55.5, 55.5, 55.5, 55.5, 55.5, 55.5, 55.5, 63, 63, 63, 63, 70.5, 70.5, 78)
#links_labels_56 = c("", "", '', '', '', '', '', '', "", '', '', '', "", '', "")

#tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_5, links_pos_5,
#  links_pos_5, links_chromosomes_6, links_pos_6, links_pos_6,
#  maxRadius = 0.5, width = "0.01em", color = 'brown', labelSize = "0.5em", labels = links_labels_56)

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra = c('Receptor') # Chromosomes on which the arcs should be displayed
arcs_begin_ra = c(0)
arcs_end_ra = c(240)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra, arcs_begin_ra, arcs_end_ra, minRadius = 1.15, maxRadius = 1.20, colors = 'forestgreen') 

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_hdrg = c('Ligand') # Chromosomes on which the arcs should be displayed
arcs_begin_hdrg = c(0)
arcs_end_hdrg = c(120)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_hdrg, arcs_begin_hdrg, arcs_end_hdrg, minRadius = 1.15, maxRadius = 1.20, colors = 'deeppink3')

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Human RA to Mouse DRG - Specific Neurons', x = -0.6, y = -1.30, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Ligand', x = 0.75, y = -1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Receptor', x = -0.95, y = -1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes, arcs_begin, arcs_end, minRadius = 0.85, maxRadius = 0.99, colors = 'forestgreen') 

BioCircos(tracklist, genome = myGenome, genomeFillColor = c('white', 'white'), chrPad = 0.05, displayGenomeBorder = FALSE, yChr =  FALSE,
  genomeTicksDisplay = FALSE,  genomeLabelTextSize = 0, genomeLabelDy = 0, width = 650, height = 650)
```

Below we make a legend for the fibroblast subtypes.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (1)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (2)', 'DKK3+ sublining fibroblasts (SC-F3) (1)', 'CD55+ lining fibroblasts (SC-F4) (6)'), pch=16, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes", at=0.24, cex=2)
```

Below we plot the legend for the cell types.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Peripheral Sensory Non-peptidergic Neurons (4)', 'Peripheral Sensory Neurofilament Neurons (4)', 'Peripheral Sensory Peptidergic Neurons (5)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('darkgrey', 'gold', 'magenta'))
mtext("mDRG Neuronal Cell Types", at=0.19, cex=1.1)
```

Below we create one plot with both mDRG to human RA and human RA to mDRG.

```{r, fig.height=10,fig.width=10}
library(BioCircos)

# Chromosomes on which the Receptor points should be displayed
points_chromosomes_ligand = c('Ligand', 'Ligand')
# Coordinates on which the Receptor points should be displayed (length.out = 11 because there are 11 unique genes in the receptor column
points_coordinates_ligand = seq(3, 101, length.out = 11)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique ligands
points_values_ligand = 1:length(points_coordinates_ligand)

# Size of the outside of the circle for ligands and receptors
# Since there are uniquely 11 ligands and 26 receptor genes, we make the slices of the circle 11/38*360 = 104 for the receptors and 360 - 104 for the ligands
myGenome = list("Ligand" = 104,
        "Receptor" = 256)

# Create circle for cell type identification to lie in
tracklist = BioCircosBackgroundTrack("myBackgroundTrack", minRadius = 0.78, maxRadius = 1.13,
  borderColors = "#AAAAAA", borderSize = 0.6, fillColors = "White") 

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("blue", "white", "white", "white", "blue", "white", "blue", "white", "white", "white", "white"), minRadius = 1.11, maxRadius = 1.11, shape = 'rect', size = 4)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("green", "green", "white", "white", "white", "white", "green", "green", "white", "white", "white"), minRadius = 1.01, maxRadius = 1.01, shape = 'rect', size = 4)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("orange", "orange", "orange", "white", "white", "white", "orange", "white", "white", "white", "white"), minRadius = 0.91, maxRadius = 0.91, shape = 'rect', size = 4)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("red", "red", "red", "red", "red", "red", "red", "red", "red", "red", "red"), minRadius = 0.81, maxRadius = 0.81, shape = 'rect', size = 4)

# Chromosomes on which the ligand should be displayed
points_chromosomes_receptor = c('Receptor', 'Receptor')
# Coordinates on which the ligand should be displayed length.out = 26 
points_coordinates_receptor = seq(3, 253, length.out = 26)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique receptors
points_values_receptor = 1:length(points_chromosomes_receptor)

# Peripheral sensory peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "magenta", "white", "white", "white", "white", "magenta", "magenta", "white", "magenta", "magenta", "white", "white", "white", "white", "white", "white", "magenta", "magenta"), minRadius = 0.80, maxRadius = 0.80)

# Peripheral sensory neurofilament neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("gold", "gold", "gold", "gold", "gold", "gold", "gold", "gold", "gold", "white", "white", "white", "white", "white", "white", "gold", "white", "gold", "gold", "gold", "white", "white", "white", "white", "white", "white"), minRadius = 0.95, maxRadius = 0.95)

# Peripheral sensory non-peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "white", "darkgray", "darkgray", "darkgray", "darkgray", "white", "white", "white", "white", "white", "white", "white", "darkgray", "darkgray", "darkgray", "darkgray", "white", "white"), minRadius = 1.10, maxRadius = 1.10)

#tracklist = tracklist + BioCircosTextTrack("testText", 'mDRG to Synovial Fibroblast', weight = "bold", size = '1em', x = -0.46, y = -1.25, displayGenomeBorder = TRUE)

links_chromosomes_1 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should start
links_chromosomes_2 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should end

# Coordinates to place start of links. The coordinates below correspond to the sequence given by seq(3, 101, length.out = 11) Their starting coordinate must be repeated for the number of connections between them and the number of ligands they are connected to.
links_pos_1 = c(3, 3, 3, 12.8, 3, 3, 3, 12.8, 12.8, 3, 3, 3, 12.8, 22.6, 52, 52, 12.8, 3, 3, 3, 12.8, 32.4, 22.6, 3, 3, 3, 12.8, 3, 22.6, 3, 12.8, 3, 22.6, 3, 12.8, 3, 42.2, 52, 42.2, 61.8, 71.6, 61.8, 71.6, 61.8, 52, 71.6, 81.4, 91.2, 52, 52, 101, 101, 52, 52, 61.8, 71.6, 101)
# Coordinates to place end of links. The coordinates below correspond to the sequence given by seq(3, 253, length.out = 26)
links_pos_2 = c(3, 13, 23, 33, 3, 13, 23, 43, 33, 3, 13, 23, 43, 53, 63, 73, 33, 3, 13, 23, 43, 83, 53, 93, 103, 93, 113, 103, 123, 93, 113, 103, 123, 93, 113, 103, 133, 73, 133, 143, 153, 143, 163, 143, 173, 153, 183, 193, 203, 213, 223, 233, 173, 243, 143, 163, 253)
# The labels are given at the end of the link. 
links_labels_12 = c("FGF1", "FGF11", "FGF9", "EDIL3", "", "", "", "LTBP3", "", "", "", "", "", "SHH", "CLSTN1", "CLSTN3", "", "", "", "", "", "NFASC", "", "FGF12", "NCAM1", "", "ITGB3BP", "", "DHH", "", "", "", "", "", "", "", "PTN", "", "", 'NTRK1', "EPHA10", "", "EPHA5", "", "NGFR", "", "UNC5A", "PLXNA1", "CAV1", "SLC45A3", "CD44", "CD82", "", "TNFRSF21", "", "", "CD9") 

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_1, links_pos_1,
  links_pos_1, links_chromosomes_2, links_pos_2, links_pos_2,
  maxRadius = 0.5, width = "0.01em", color = 'brown', labelSize = "0.75em", labels = links_labels_12)

links_chromosomes_3 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should start
links_chromosomes_4 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should end

# Now we start at the ligands
links_pos_3 = c(3, 13, 23, 33, 3, 13, 23, 43, 33, 3, 13, 23, 43, 53, 63, 73, 33, 3, 13, 23, 43, 83, 53, 93, 103, 93, 113, 103, 123, 93, 113, 103, 123, 93, 113, 103, 133, 73, 133, 143, 153, 143, 163, 143, 173, 153, 183, 193, 203, 213, 223, 233, 173, 243, 143, 163, 253)
# Now we end at the receptors
links_pos_4 = c(3, 3, 3, 12.8, 3, 3, 3, 12.8, 12.8, 3, 3, 3, 12.8, 22.6, 52, 52, 12.8, 3, 3, 3, 12.8, 32.4, 22.6, 3, 3, 3, 12.8, 3, 22.6, 3, 12.8, 3, 22.6, 3, 12.8, 3, 42.2, 52, 42.2, 61.8, 71.6, 61.8, 71.6, 61.8, 52, 71.6, 81.4, 91.2, 52, 52, 101, 101, 52, 52, 61.8, 71.6, 101)
# These links are for the receptors
links_labels_34 = c('FGFR1', "", "", "ITGB5", "", "", "", "", "", "", "", "", "", "CDON", "APP", "", "", "", "", "", "", "ANK3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "PTPRS", "", "", "CTGF", "EFNA5", "", "", "", "", "", "NTN4", "SEMA6D", "", "", "HBEGF", "", "", "", "", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_3, links_pos_3,
  links_pos_3, links_chromosomes_4, links_pos_4, links_pos_4,
  maxRadius = 0.5, width = "0.01em", color = 'brown', labelSize = "0.75em", labels = links_labels_34, labelPadding = 3)

links_chromosomes_5 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should start
links_chromosomes_6 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should end
links_pos_5 = c(143, 153, 143, 163, 143, 173, 153, 183, 193, 203, 213, 223, 233, 173, 243, 143, 163, 253)
links_pos_6 = c(61.8, 71.6, 61.8, 71.6, 61.8, 52, 71.6, 81.4, 91.2, 52, 52, 101, 101, 52, 52, 61.8, 71.6, 101)
links_labels_56 = c("", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_5, links_pos_5,
  links_pos_5, links_chromosomes_6, links_pos_6, links_pos_6,
  maxRadius = 0.5, width = "0.01em", color = 'cyan', labelSize = "0.5em", labels = links_labels_56)

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra = c('Ligand') # Chromosomes on which the arcs should be displayed
arcs_begin_ra = c(0)
arcs_end_ra = c(56)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra, arcs_begin_ra, arcs_end_ra, minRadius = 1.15, maxRadius = 1.20, colors = 'forestgreen')

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra2 = c('Ligand') # Chromosomes on which the arcs should be displayed
arcs_begin_ra2 = c(50)
arcs_end_ra2 = c(104)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra2, arcs_begin_ra2, arcs_end_ra2, minRadius = 1.20, maxRadius = 1.25, colors = 'deeppink3')

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_hdrg = c('Receptor') # Chromosomes on which the arcs should be displayed
arcs_begin_hdrg = c(0)
arcs_end_hdrg = c(136)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_hdrg, arcs_begin_hdrg, arcs_end_hdrg, minRadius = 1.20, maxRadius = 1.25, colors = 'deeppink3')

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_hdrg2 = c('Receptor') # Chromosomes on which the arcs should be displayed
arcs_begin_hdrg2 = c(136)
arcs_end_hdrg2 = c(256)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_hdrg2, arcs_begin_hdrg2, arcs_end_hdrg2, minRadius = 1.15, maxRadius = 1.20, colors = 'forestgreen')

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Mouse DRG to Human RA - Specific Neurons', x = -0.6, y = -1.30, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Receptor', x #= 0.75, y = -1, size = "1em",
##  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Ligand', x = #-0.95, y = -1, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes, arcs_begin, arcs_end, minRadius = 0.85, maxRadius = 0.99, colors = 'forestgreen') 

BioCircos(tracklist, genome = myGenome, genomeFillColor = c('white', 'white'), chrPad = 0.05, displayGenomeBorder = FALSE, yChr =  FALSE,
  genomeTicksDisplay = FALSE,  genomeLabelTextSize = 0, genomeLabelDy = 0, width = 650, height = 650)
```

Below we make a legend for the ligand and receptor tracts.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Receptor', 'Ligand'), bty='n', lwd = 8, cex = 1.2, col = c("forestgreen", "deeppink3"), lty = c(1, 1), pch = c(NA, NA))
 #   col = c('forestgreen', 'deeppink3'))
#mtext("mDRG Neuronal Cell Types", at=0.19, cex=1.1)
```


Below we make a legend for the fibroblast subtypes.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (3)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (4)', 'DKK3+ sublining fibroblasts (SC-F3) (4)', 'CD55+ lining fibroblasts (SC-F4) (11)'), pch=15, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes", at=0.24, cex=2)
```

Below we split the legend in half - one for the ligands and one for the recpetors.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (2)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (2)', 'DKK3+ sublining fibroblasts (SC-F3) (3)', 'CD55+ lining fibroblasts (SC-F4) (6)'), pch=15, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes For Pain-Associated Receptors", at=0.24, cex=1.6)
```

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (1)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (2)', 'DKK3+ sublining fibroblasts (SC-F3) (1)', 'CD55+ lining fibroblasts (SC-F4) (6)'), pch=15, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes For Pain-Associated Ligands", at=0.24, cex=1.6)
```

Below we plot the legend for the cell types.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Peripheral Sensory Non-peptidergic Neurons (8)', 'Peripheral Sensory Neurofilament Neurons (13)', 'Peripheral Sensory Peptidergic Neurons (7)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('darkgrey', 'gold', 'magenta'))
mtext("Neuronal Cell Types", at=0.19, cex=1.1)
```

Below we split the legend in half - one for the ligands and one for the recpetors.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Peripheral Sensory Non-peptidergic Neurons (4)', 'Peripheral Sensory Neurofilament Neurons (4)', 'Peripheral Sensory Peptidergic Neurons (5)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('darkgrey', 'gold', 'magenta'))
mtext("Neuronal Cell Types For Receptors", at=0.19, cex=1.1)
```

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Peripheral Sensory Non-peptidergic Neurons (4)', 'Peripheral Sensory Neurofilament Neurons (9)', 'Peripheral Sensory Peptidergic Neurons (2)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('darkgrey', 'gold', 'magenta'))
mtext("Neuronal Cell Types For Ligands", at=0.19, cex=1.1)
```

Below we create plots for the number of connections for mDRG to RA and RA to mDRG.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Mouse Nervous System to Synovial Fibroblast (39 connections)', 'Synovial Fibroblast to Mouse Nervous System (18 connections)'), bty='n', lwd = 8, cex = 1.2, col = c('brown', 'cyan'), lty = c(1, 1, 1, 1), pch = c(NA, NA, NA, NA))
 #   col = c('brown', 'cyan'))
#mtext("mDRG Neuronal Cell Types", at=0.19, cex=1.1)
```

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (1 Connection)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (3 Connections)', 'DKK3+ sublining fibroblasts (SC-F3) (1 Connection)', 'CD55+ lining fibroblasts (SC-F4) (13 Connections)'), pch=15, pt.cex=3, cex=1.2, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Connections Synovial Fibroblast to Mouse Nervous System", at=0.34, cex=1.3)
```

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (6 Connections)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (8 Connections)', 'DKK3+ sublining fibroblasts (SC-F3) (10 Connections)', 'CD55+ lining fibroblasts (SC-F4) (15 Connections)'), pch=15, pt.cex=3, cex=1.2, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Connections Mouse Nervous System to Synovial Fibroblast", at=0.34, cex=1.3)
```

Below we create the connection legends for cell types.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Peripheral Sensory Non-peptidergic Neurons (23 Connections)', 'Peripheral Sensory Neurofilament Neurons (13 Connections)', 'Peripheral Sensory Peptidergic Neurons (3 Connections)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('darkgrey', 'gold', 'magenta'))
mtext("Neuronal Cell Type Connections Mouse Nervous System to Synovial Fibroblast", at=0.4, cex=1.1)
```

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Peripheral Sensory Non-peptidergic Neurons (4 Connections)', 'Peripheral Sensory Neurofilament Neurons (5 Connections)', 'Peripheral Sensory Peptidergic Neurons (9 Connections)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('darkgrey', 'gold', 'magenta'))
mtext("Neuronal Cell Type Connections Synovial Fibroblast to Mouse Nervous System", at=0.4, cex=1.1)
```


### All Neurons Plots

Below we create the interactome plot from the mouse DRG ligand to the human RA receptors.

```{r, fig.height=10,fig.width=10}
library(BioCircos)

# Chromosomes on which the Receptor points should be displayed
points_chromosomes_ligand = c('Ligand', 'Ligand')
# Coordinates on which the Receptor points should be displayed (length.out = 10 because there are 10 unique genes in the receptor column
points_coordinates_ligand = seq(3, 85, length.out = 10)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique ligands
points_values_ligand = 1:length(points_coordinates_ligand)

# Size of the outside of the circle for ligands and receptors
# Since there are uniquely 31 ligands and 10 receptor genes, we make the slices of the circle 10/41*360 = 88 for the receptors and 360 - 88 for the ligands
myGenome = list("Ligand" = 88,
        "Receptor" = 272)

# Create circle for cell type identification to lie in
tracklist = BioCircosBackgroundTrack("myBackgroundTrack", minRadius = 0.78, maxRadius = 1.23,
  borderColors = "#AAAAAA", borderSize = 0.6, fillColors = "White") 

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("blue", "white", "white", "white", "blue", "blue", "white", "white", "blue", "white"), minRadius = 1.21, maxRadius = 1.21)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("green", "white", "white", "white", "white", "white", "green", "white", "white", "white"), minRadius = 1.08, maxRadius = 1.08)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("orange", "white", "white", "white", "orange", "white", "orange", "orange", "white", "orange"), minRadius = 0.95, maxRadius = 0.95)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("red", "red", "red", "red", "white", "red", "red", "white", "white", "red"), minRadius = 0.81, maxRadius = 0.81)

# Chromosomes on which the ligand should be displayed
points_chromosomes_receptor = c('Receptor', 'Receptor')
# Coordinates on which the ligand should be displayed length.out = 31
points_coordinates_receptor = seq(3, 269, length.out = 31)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique receptors
points_values_receptor = 1:length(points_chromosomes_receptor)

# Cerebellum neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("antiquewhite4", "antiquewhite4", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.80, maxRadius = 0.80)

# Cholinergic and monoaminergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "black", "black", "black", "black", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.832, maxRadius = 0.832)

# Di- and mesencphalon excitatory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "blue", "white", "white", "blue", "white", "white", "white", "blue", "blue", "blue", "blue", "blue", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.863, maxRadius = 0.863)

# Di- and mesencphalon inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("blueviolet", "blueviolet", "blueviolet", "white", "blueviolet", "white", "white", "white", "blueviolet", "blueviolet", "blueviolet", "blueviolet", "blueviolet", "blueviolet", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.895, maxRadius = 0.895)

# Enteric neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "brown", "brown", "white", "white", "brown", "brown", "white", "brown", "brown", "brown", "white", "white", "brown", "brown", "brown", "brown", "brown", "brown", "brown", "brown", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.926, maxRadius = 0.926)

# Hindbrain neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "cadetblue1", "cadetblue1", "white", "cadetblue1", "white", "white", "white", "cadetblue1", "cadetblue1", "white", "cadetblue1", "cadetblue1", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cadetblue1", "cadetblue1", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.946, maxRadius = 0.946)

# Peripheral sensory neurofilament neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "gold", "white", "gold", "white", "gold", "white", "white", "white", "gold", "gold", "gold", "gold", "white", "white", "white", "white", "gold", "white", "white", "white", "white", "gold", "gold", "gold", "gold", "gold", "white", "white", "white"), minRadius = 0.976, maxRadius = 0.976)

# Peripheral sensory non-peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "darkgrey", "white", "white", "white", "darkgrey", "white", "white", "white", "white", "white", "white", "white", "darkgrey", "white", "white", "white", "darkgrey", "white", "white", "white", "white", "darkgrey", "darkgrey", "white", "white", "white", "darkgrey", "white", "white"), minRadius = 1.01, maxRadius = 1.01)

# Peripheral sensory peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "magenta", "white", "white", "white", "magenta", "white", "white", "magenta", "magenta", "white", "white", "white", "white", "white", "white", "white", "magenta", "white", "white", "white", "white", "magenta", "white", "white", "magenta", "white", "white", "white", "white"), minRadius = 1.034, maxRadius = 1.034)

# Sympathetic cholinergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "white", "chartreuse3", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "white", "white", "white", "chartreuse3", "chartreuse3"), minRadius = 1.064, maxRadius = 1.064)

# Sympathetic nonadrenergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "white", "cyan", "white", "white", "white", "white"), minRadius = 1.093, maxRadius = 1.093)

# Dendate gyrus granule neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "orange", "white", "white", "orange", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.122, maxRadius = 1.122)

# Olfactory inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "coral", "white", "white", "white", "white", "white", "white", "coral", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "coral", "white", "white", "white", "white", "white", "white"), minRadius = 1.151, maxRadius = 1.151)

# Spinal cord inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "firebrick", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.21, maxRadius = 1.21)

#tracklist = tracklist + BioCircosTextTrack("testText", 'mDRG to Synovial Fibroblast', weight = "bold", size = '1em', x = -0.46, y = -1.25, displayGenomeBorder = TRUE)

links_chromosomes_1 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should start
links_chromosomes_2 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should end

# Coordinates to place start of links. The coordinates below correspond to the sequence given by seq(3, 85, length.out = 10) Their starting coordinate must be repeated for the number of connections between them and the number of ligands they are connected to.
links_pos_1 = c(3, 3, 3, 12.11, 3, 3, 3, 3, 3, 3, 3, 12.11, 3, 3, 21.22, 30.33, 3, 39.44, 3, 3, 39.44, 30.33, 3, 48.56, 48.56, 57.67, 57.67, 57.67, 66.78, 57.67, 12.11, 12.11, 57.67, 48.56, 57.67, 21.22, 48.56, 3, 3, 48.56, 48.56, 57.67, 3, 3, 57.67, 57.67, 3, 3, 66.78, 57.67, 12.11, 12.11, 57.67, 3, 3, 48.56, 57.67, 21.22, 48.56, 75.89, 3, 3, 48.56, 48.56, 3, 48.56, 39.44, 39.44, 57.67, 57.67, 3, 3, 57.67, 57.67, 57.67, 3, 57.67, 57.67, 57.67, 3, 3, 57.67, 66.78, 57.67, 57.67, 3, 57.67, 39.44, 39.44, 57.67, 57.67, 3, 3, 48.56, 57.67, 48.56, 57.67, 57.67, 3, 48.56, 57.67, 3, 3, 48.56, 48.56, 57.67, 3, 3, 57.67, 57.67, 3, 3, 66.78, 57.67, 57.67, 12.11, 12.11, 3, 3, 48.56, 57.67, 21.22, 48.56, 3, 48.56, 3, 3, 3, 48.56, 3, 3, 3, 3, 3, 39.44, 57.67, 57.67, 57.67, 3, 3, 3, 3, 3, 57.67, 57.67, 57.67, 57.67, 3, 3, 3, 3, 3, 57.67, 85, 39.44, 57.67, 12.11, 57.67, 57.67, 3, 3, 3, 3, 3, 57.67, 21.22, 85, 75.89, 3, 3, 3, 57.67, 57.67, 3, 3, 3, 57.67, 57.67, 85, 3, 3, 3, 57.67, 57.67, 85, 3, 3, 3, 3, 3, 48.56, 39.44, 57.67, 57.67, 57.67, 3, 3, 57.67, 57.67, 57.67, 3, 3, 39.44, 57.67, 57.67, 57.67, 3, 3, 48.56, 12.11, 3, 3, 48.56, 39.44, 3, 3, 57.67, 57.67, 3, 3, 57.67, 57.67, 39.44, 3, 3, 57.67, 48.56, 57.67, 3, 39.44, 39.44, 3, 3, 39.44, 39.44, 3)
# Coordinates to place end of links. The coordinates below correspond to the sequence given by seq(3, 269, length.out = 31)
links_pos_2 = c(3, 3, 3, 11.87, 3, 20.73, 29.6, 3, 29.6, 3, 29.6, 11.87, 3, 29.6, 38.47, 47.33, 56.2, 65.07, 56.2, 56.2, 65.07, 47.33, 56.2, 73.93, 82.8, 91.67, 100.53, 91.67, 73.93, 100.53, 109.4, 11.87, 91.67, 73.93, 100.53, 38.47, 82.8, 118.27, 3, 73.93, 82.8, 91.67, 118.27, 3, 100.53, 91.67, 118.27, 3, 73.93, 100.53, 109.4, 11.87, 91.67, 118.27, 3, 73.93, 100.53, 38.47, 82.8, 127.13, 20.73, 56.2, 136, 144.87, 29.6, 82.8, 153.73, 65.07, 162.6, 91.67, 20.73, 56.2, 171.47, 180.33, 100.53, 29.6, 189.2, 162.6, 91.67, 20.73, 56.2, 171.47, 144.87, 180.33, 100.53, 29.6, 189.2, 153.73, 65.07, 162.6, 91.67, 20.73, 56.2, 136, 171.47, 144.87, 180.33, 100.53, 29.6, 82.8, 189.2, 20.73, 198.07, 73.93, 82.8, 206.93, 20.73, 198.07, 100.53, 206.93, 20.73, 198.07, 73.93, 100.53, 206.93, 109.4, 11.87, 20.73, 198.07, 73.93, 100.53, 38.47, 82.8, 215.8, 82.8, 215.8, 215.8, 215.8, 82.8, 20.73, 118.27, 215.8, 56.2, 224.67, 233.53, 206.93, 162.6, 91.67, 20.73, 118.27, 215.8, 56.2, 224.67, 100.53, 206.93, 162.6, 91.67, 20.73, 118.27, 215.8, 56.2, 224.67, 100.53, 242.4, 233.53, 206.93, 109.4, 162.6, 91.67, 20.73, 118.27, 215.8, 56.2, 224.67, 100.53, 38.47, 242.4, 127.13, 20.73, 215.8, 56.2, 206.93, 162.6, 20.73, 215.8, 56.2, 206.93, 162.6, 251.27, 20.73, 215.8, 56.2, 206.93, 162.6, 251.27, 20.73, 215.8, 56.2, 20.73, 56.2, 82.8, 233.53, 206.93, 162.6, 91.67, 20.73, 56.2, 206.93, 162.6, 91.67, 20.73, 56.2, 233.53, 206.93, 162.6, 91.67, 20.77, 56.2, 82.8, 11.87, 20.73, 260.13, 82.8, 153.73, 20.73, 260.13, 171.47, 269, 20.73, 260.13, 171.47, 269, 153.73, 20.73, 260.13, 171.47, 82.8, 269, 20.73, 153.73, 233.53, 20.73, 20.73, 153.73, 233.53, 20.73)
# The labels are given at the end of the link. 
links_labels_12 = c("FGF9", "", "", "CLSTN3", "", "FGF1", "NCAM1", "", "", "", "", "", "", "", "NFASC", "CXCL12", "FGF13", "TGFB2", "", "", "", "", "", "LRRC4B", "PTN", "EDIL3", "LTBP3", "", "", "", "CLSTN1", "", "", "", "", "", "", "FGF11", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "EFEMP2", "", "", "HSPG2", "LRFN4", "", "", "TGFB1", "", "CYR61", "", "", "", "ITGB3BP", "LTBP1", "", "", "SORBS1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF14", "", "", "ADAM9", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF12", "", "", "", "", "", "", "", "", "", "FGF18", "TGFB3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "SHH", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "DHH", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF7", "", "", "", "", "", "SERPINE1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "") 

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_1, links_pos_1,
  links_pos_1, links_chromosomes_2, links_pos_2, links_pos_2,
  maxRadius = 0.5, width = "0.01em", color = 'brown', labelSize = "0.75em", labels = links_labels_12)

links_chromosomes_3 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should start
links_chromosomes_4 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should end

# Now we start at the ligands
links_pos_3 = c(3, 3, 3, 11.87, 3, 20.73, 29.6, 3, 29.6, 3, 29.6, 11.87, 3, 29.6, 38.47, 47.33, 56.2, 65.07, 56.2, 56.2, 65.07, 47.33, 56.2, 73.93, 82.8, 91.67, 100.53, 91.67, 73.93, 100.53, 109.4, 11.87, 91.67, 73.93, 100.53, 38.47, 82.8, 118.27, 3, 73.93, 82.8, 91.67, 118.27, 3, 100.53, 91.67, 118.27, 3, 73.93, 100.53, 109.4, 11.87, 91.67, 118.27, 3, 73.93, 100.53, 38.47, 82.8, 127.13, 20.73, 56.2, 136, 144.87, 29.6, 82.8, 153.73, 65.07, 162.6, 91.67, 20.73, 56.2, 171.47, 180.33, 100.53, 29.6, 189.2, 162.6, 91.67, 20.73, 56.2, 171.47, 144.87, 180.33, 100.53, 29.6, 189.2, 153.73, 65.07, 162.6, 91.67, 20.73, 56.2, 136, 171.47, 144.87, 180.33, 100.53, 29.6, 82.8, 189.2, 20.73, 198.07, 73.93, 82.8, 206.93, 20.73, 198.07, 100.53, 206.93, 20.73, 198.07, 73.93, 100.53, 206.93, 109.4, 11.87, 20.73, 198.07, 73.93, 100.53, 38.47, 82.8, 215.8, 82.8, 215.8, 215.8, 215.8, 82.8, 20.73, 118.27, 215.8, 56.2, 224.67, 233.53, 206.93, 162.6, 91.67, 20.73, 118.27, 215.8, 56.2, 224.67, 100.53, 206.93, 162.6, 91.67, 20.73, 118.27, 215.8, 56.2, 224.67, 100.53, 242.4, 233.53, 206.93, 109.4, 162.6, 91.67, 20.73, 118.27, 215.8, 56.2, 224.67, 100.53, 38.47, 242.4, 127.13, 20.73, 215.8, 56.2, 206.93, 162.6, 20.73, 215.8, 56.2, 206.93, 162.6, 251.27, 20.73, 215.8, 56.2, 206.93, 162.6, 251.27, 20.73, 215.8, 56.2, 20.73, 56.2, 82.8, 233.53, 206.93, 162.6, 91.67, 20.73, 56.2, 206.93, 162.6, 91.67, 20.73, 56.2, 233.53, 206.93, 162.6, 91.67, 20.77, 56.2, 82.8, 11.87, 20.73, 260.13, 82.8, 153.73, 20.73, 260.13, 171.47, 269, 20.73, 260.13, 171.47, 269, 153.73, 20.73, 260.13, 171.47, 82.8, 269, 20.73, 153.73, 233.53, 20.73, 20.73, 153.73, 233.53, 20.73)
# Now we end at the receptors
links_pos_4 = c(3, 3, 3, 12.11, 3, 3, 3, 3, 3, 3, 3, 12.11, 3, 3, 21.22, 30.33, 3, 39.44, 3, 3, 39.44, 30.33, 3, 48.56, 48.56, 57.67, 57.67, 57.67, 66.78, 57.67, 12.11, 12.11, 57.67, 48.56, 57.67, 21.22, 48.56, 3, 3, 48.56, 48.56, 57.67, 3, 3, 57.67, 57.67, 3, 3, 66.78, 57.67, 12.11, 12.11, 57.67, 3, 3, 48.56, 57.67, 21.22, 48.56, 75.89, 3, 3, 48.56, 48.56, 3, 48.56, 39.44, 39.44, 57.67, 57.67, 3, 3, 57.67, 57.67, 57.67, 3, 57.67, 57.67, 57.67, 3, 3, 57.67, 66.78, 57.67, 57.67, 3, 57.67, 39.44, 39.44, 57.67, 57.67, 3, 3, 48.56, 57.67, 48.56, 57.67, 57.67, 3, 48.56, 57.67, 3, 3, 48.56, 48.56, 57.67, 3, 3, 57.67, 57.67, 3, 3, 66.78, 57.67, 57.67, 12.11, 12.11, 3, 3, 48.56, 57.67, 21.22, 48.56, 3, 48.56, 3, 3, 3, 48.56, 3, 3, 3, 3, 3, 39.44, 57.67, 57.67, 57.67, 3, 3, 3, 3, 3, 57.67, 57.67, 57.67, 57.67, 3, 3, 3, 3, 3, 57.67, 85, 39.44, 57.67, 12.11, 57.67, 57.67, 3, 3, 3, 3, 3, 57.67, 21.22, 85, 75.89, 3, 3, 3, 57.67, 57.67, 3, 3, 3, 57.67, 57.67, 85, 3, 3, 3, 57.67, 57.67, 85, 3, 3, 3, 3, 3, 48.56, 39.44, 57.67, 57.67, 57.67, 3, 3, 57.67, 57.67, 57.67, 3, 3, 39.44, 57.67, 57.67, 57.67, 3, 3, 48.56, 12.11, 3, 3, 48.56, 39.44, 3, 3, 57.67, 57.67, 3, 3, 57.67, 57.67, 39.44, 3, 3, 57.67, 48.56, 57.67, 3, 39.44, 39.44, 3, 3, 39.44, 39.44, 3)
# These links are for the receptors
links_labels_34 = c('FGFR1', "", "", "APP", "", "", "", "", "", "", "", "", "", "", "ANK3", "ACKR3", "", "TGFBR3", "", "", "", "", "", "PTPRS", "", "ITGB5", "", "", "PTPRD", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "PLSCR4", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "CDON", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_3, links_pos_3,
  links_pos_3, links_chromosomes_4, links_pos_4, links_pos_4,
  maxRadius = 0.5, width = "0.01em", color = 'brown', labelSize = "0.75em", labels = links_labels_34, labelPadding = 3)

#links_chromosomes_5 = c('ligand', 'hDRG', 'hDRG') # Chromosomes on which the links should start
#links_chromosomes_6 = c('RA', 'RA', 'RA') # Chromosomes on which the links should end
#links_pos_5 = c(180.08, 187.46, 194.84, 202.22, 209.59, 216.97, 224.35, 231.73, 239.11, 216.97, 246.49, 253.86, 261.24, 268.62, 276)
#links_pos_6 = c(48, 55.5, 55.5, 55.5, 55.5, 55.5, 55.5, 55.5, 63, 63, 63, 63, 70.5, 70.5, 78)
#links_labels_56 = c("", "", '', '', '', '', '', '', "", '', '', '', "", '', "")

#tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_5, links_pos_5,
#  links_pos_5, links_chromosomes_6, links_pos_6, links_pos_6,
#  maxRadius = 0.5, width = "0.01em", color = 'brown', labelSize = "0.5em", labels = links_labels_56)

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra = c('Ligand') # Chromosomes on which the arcs should be displayed
arcs_begin_ra = c(0)
arcs_end_ra = c(88)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra, arcs_begin_ra, arcs_end_ra, minRadius = 1.25, maxRadius = 1.30, colors = 'forestgreen') 

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_hdrg = c('Receptor') # Chromosomes on which the arcs should be displayed
arcs_begin_hdrg = c(0)
arcs_end_hdrg = c(272)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_hdrg, arcs_begin_hdrg, arcs_end_hdrg, minRadius = 1.25, maxRadius = 1.30, colors = 'deeppink3')

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Mouse DRG to Human RA - All Neurons', x = -0.6, y = -1.37, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Receptor', x = 0.85, y = -1.1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Ligand', x = -1.05, y = -1.1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes, arcs_begin, arcs_end, minRadius = 0.85, maxRadius = 0.99, colors = 'forestgreen') 

BioCircos(tracklist, genome = myGenome, genomeFillColor = c('white', 'white'), chrPad = 0.05, displayGenomeBorder = FALSE, yChr =  FALSE,
  genomeTicksDisplay = FALSE,  genomeLabelTextSize = 0, genomeLabelDy = 0, width = 650, height = 650)
```

Below we make a legend for the fibroblast subtypes.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (4)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (2)', 'DKK3+ sublining fibroblasts (SC-F3) (5)', 'CD55+ lining fibroblasts (SC-F4) (7)'), pch=16, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes", at=0.24, cex=2)
```

Below we plot the legend for the cell types.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Cerebellum Neurons (2)', 'Cholinergic and Monoaminergic Neurons (4)', 'Di- and Mesencphalon Excitatory Neurons (7)', 'Di- and Mesencphalon Inhibitory Neurons (10)', 'Enteric Neurons (15)', 'Hindbrain Neurons (9)', 'Peripheral Sensory Neurofilament Neurons (13)', 'Peripheral Sensory Non-peptidergic Neurons (7)', 'Peripheral Sensory Peptidergic Neurons (7)', 'Sympathetic Cholinergic Neurons (6)', 'Sympathetic Noradrenergic Neurons (3)', 'Dentate Gyrus Granule Neurons (4)', 'Olfactory Inhibitory Neurons (3)', 'Spinal Cord Inhibitory Neurons (1)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('antiquewhite4', 'black', 'blue', 'blueviolet', 'brown', 'cadetblue1', 'gold', 'darkgrey', 'magenta', 'chartreuse3', 'cyan', 'orange', 'coral', 'firebrick'
            ))
mtext("mDRG Neuronal Cell Types", at=0.19, cex=1.1)
```

Below we create the same interactome plot from the human RA receptors to the mouse DRG ligands.

```{r, fig.height=10,fig.width=10}
library(BioCircos)

# Chromosomes on which the Receptor points should be displayed
points_chromosomes_ligand = c('Ligand', 'Ligand')
# Coordinates on which the Receptor points should be displayed (length.out = 11 because there are 11 unique genes in the Ligand column
points_coordinates_ligand = seq(3, 91, length.out = 11)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique ligands
points_values_ligand = 1:length(points_coordinates_ligand)

# Size of the outside of the circle for ligands and receptors
# Since there are uniquely 11 ligands and 31 receptor genes, we make the slices of the circle 31/42*360 = 266 for the receptors and 360 - 266 for the ligands
myGenome = list("Ligand" = 94, "Receptor" = 266)

# Create circle for cell type identification to lie in
tracklist = BioCircosBackgroundTrack("myBackgroundTrack", minRadius = 0.78, maxRadius = 1.23,
  borderColors = "#AAAAAA", borderSize = 0.6, fillColors = "White") 

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("blue", "blue", "blue", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.21, maxRadius = 1.21)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("green", "green", "green", "green", "green", "white", "white", "white", "white", "white", "white"), minRadius = 1.08, maxRadius = 1.08)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("orange", "white", "orange", "white", "orange", "white", "white", "white", "white", "white", "white"), minRadius = 0.95, maxRadius = 0.95)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("red", "red", "red", "red", "red", "red", "red", "red", "red", "red", "red"), minRadius = 0.81, maxRadius = 0.81)

# Chromosomes on which the ligand should be displayed
points_chromosomes_receptor = c('Receptor', 'Receptor')
# Coordinates on which the ligand should be displayed length.out = 31 
points_coordinates_receptor = seq(3, 263, length.out = 31)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique receptors
points_values_receptor = 1:length(points_chromosomes_receptor)

# Cerebellum neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("antiquewhite4", "antiquewhite4", "antiquewhite4", "white", "white", "white", "white", "white", "antiquewhite4", "white", "white", "white", "white", "white", "white", "white", "white", "antiquewhite4", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.80, maxRadius = 0.80)

# Cholinergic and monoaminergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "black", "black", "white", "white", "white", "black", "white", "white", "white", "white", "white", "white", "white", "white", "black", "black", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.827, maxRadius = 0.827)

# Di- and mesencphalon excitatory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("blue", "white", "blue", "white", "white", "white", "white", "white", "blue", "blue", "white", "white", "white", "white", "white", "white", "white", "blue", "white", "blue", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.855, maxRadius = 0.855)

# Di- and mesencphalon inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "blueviolet", "white", "white", "white", "white", "white", "blueviolet", "blueviolet", "white", "white", "white", "blueviolet", "white", "white", "white", "blueviolet", "white", "blueviolet", "white", "white", "blueviolet", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.88, maxRadius = 0.88)

# Enteric neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "brown", "brown", "brown", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "brown", "brown", "brown", "white", "brown", "brown", "brown", "brown", "brown", "white", "white", "white", "white"), minRadius = 0.91, maxRadius = 0.91)

# Hindbrain neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "cadetblue1", "cadetblue1", "white", "white", "cadetblue1", "white", "cadetblue1", "white", "white", "white", "white", "cadetblue1", "cadetblue1", "cadetblue1", "white", "white", "white", "cadetblue1", "cadetblue1", "cadetblue1", "cadetblue1", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.937, maxRadius = 0.937)

# Peripheral sensory neurofilament neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "gold", "gold", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "gold", "white", "gold", "white", "gold", "gold", "gold", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.964, maxRadius = 0.964)

# Peripheral sensory non-peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "darkgrey", "darkgrey", "white", "white", "darkgrey", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "darkgrey", "white", "darkgrey", "darkgrey", "darkgrey", "white", "darkgrey", "darkgrey", "darkgrey", "darkgrey", "darkgrey", "darkgrey"), minRadius = 0.99, maxRadius = 0.99)

# Peripheral sensory peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "magenta", "magenta", "white", "white", "magenta", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "magenta", "white", "magenta", "white", "magenta", "white", "magenta", "magenta", "white", "magenta", "magenta", "magenta"), minRadius = 1.02, maxRadius = 1.02)

# Sympathetic cholinergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "chartreuse3", "chartreuse3", "white", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "chartreuse3", "white", "chartreuse3", "white", "white", "chartreuse3", "white", "white", "white", "white"), minRadius = 1.046, maxRadius = 1.046)

# Sympathetic nonadrenergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "cyan", "cyan", "white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cyan", "white", "cyan", "white", "white", "cyan", "white", "white", "white", "white"), minRadius = 1.073, maxRadius = 1.073)

# Dendate gyrus granule neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "orange", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.10, maxRadius = 1.10)

# Olfactory inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "coral", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.13, maxRadius = 1.13)

# Spinal cord excitatory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "white", "darkgreen", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.155, maxRadius = 1.155)

# Spinal cord inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "white", "firebrick", "white", "white", "white", "white", "white", "white", "firebrick", "white", "white", "white", "white", "firebrick", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.18, maxRadius = 1.18)

# Peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "purple", "white", "white", "white", "white"), minRadius = 1.21, maxRadius = 1.21)

#tracklist = tracklist + BioCircosTextTrack("testText", 'mDRG to Synovial Fibroblast', weight = "bold", size = '1em', x = -0.46, y = -1.25, displayGenomeBorder = TRUE)

links_chromosomes_1 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should start
links_chromosomes_2 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should end

# Coordinates to place start of links. The coordinates below correspond to the sequence given by seq(3, 263, length.out = 31) Their starting coordinate must be repeated for the number of connections between them and the number of ligands they are connected to.
links_pos_1 = c(3, 11.67, 3, 20.33, 29, 37.67, 3, 3, 20.33, 20.33, 46.33, 29, 37.67, 55, 20.33, 29, 29, 37.67, 63.67, 29, 37.67, 63.67, 29, 37.67, 63.67, 29, 37.67, 63.67, 29, 37.67, 3, 11.67, 72.33, 3, 20.33, 81, 29, 37.67, 89.67, 98.33, 107, 3, 72.33, 81, 3, 20.33, 115.67, 72.33, 81, 20.33, 46.33, 29, 37.67, 55, 115.67, 124.33, 72.33, 133, 20.33, 29, 98.33, 29, 37.67, 63.67, 29, 37.67, 63.67, 29, 37.67, 81, 141.67, 81, 63.67, 29, 37.67, 63.67, 29, 37.67, 3, 11.67, 29, 37.67, 3, 29, 37.67, 55, 133, 29, 29, 37.67, 63.67, 29, 37.67, 63.67, 29, 37.67, 63.67, 29, 37.67, 63.67, 29, 37.67, 3, 3, 11.67, 72.33, 150.33, 3, 20.33, 159, 167.67, 81, 29, 37.67, 89.67, 98.33, 107, 176.33, 185, 3, 167.67, 3, 72.33, 81, 150.33, 3, 20.33, 193.67, 167.67, 115.67, 72.33, 81, 150.33, 20.33, 193.67, 159, 202.33, 167.67, 211, 219.67, 228.33, 176.33, 46.33, 29, 37.67, 193.67, 167.67, 55, 55, 115.67, 124.33, 72.33, 55, 133, 176.33, 185, 20.33, 29, 98.33, 228.33, 193.67, 202.33, 167.67, 185, 150.33, 29, 37.67, 237, 193.67, 202.33, 245.67, 167.67, 63.67, 219.67, 254.33, 228.33, 185, 263, 29, 37.67, 202.33, 245.67, 167.67, 63.67, 219.67, 254.33, 228.33, 185, 263, 29, 37.67, 81, 141.67, 81, 185, 202.33, 167.67, 63.67, 219.67, 228.33, 29, 37.67, 202.33, 63.67, 228.33, 185, 29, 37.67)
# Coordinates to place end of links. The coordinates below correspond to the sequence given by seq(3, 91, length.out = 11)
links_pos_2 = c(3, 3, 11.8, 11.8, 20.6, 20.6, 3, 11.8, 11.8, 11.8, 11.8, 20.6, 20.6, 3, 11.8, 20.6, 20.6, 20.6, 3, 20.6, 20.6, 3, 20.6, 20.6, 3, 20.6, 20.6, 3, 20.6, 20.6, 3, 3, 29.4, 11.8, 11.8, 29.4, 20.6, 20.6, 29.4, 29.4, 29.4, 3, 29.4, 29.4, 11.8, 11.8, 29.4, 29.4, 29.4, 11.8, 11.8, 20.6, 20.6, 3, 29.4, 29.4, 29.4, 38.2, 11.8, 20.6, 29.4, 20.6, 20.6, 3, 20.6, 20.6, 3, 20.6, 20.6, 29.4, 29.4, 29.4, 3, 20.6, 20.6, 3, 20.6, 20.6, 3, 3, 20.6, 20.6, 3, 20.6, 20.6, 3, 38.2, 20.6, 20.6, 20.6, 3, 20.6, 20.6, 3, 20.6, 20.6, 3, 20.6, 20.6, 3, 20.6, 20.6, 47, 3, 3, 29.4, 55.8, 11.8, 11.8, 47, 47, 29.4, 20.6, 20.6, 29.4, 29.4, 29.4, 64.6, 73.4, 47, 47, 3, 29.4, 29.4, 55.8, 11.8, 11.8, 47, 47, 29.4, 29.4, 29.4, 55.8, 11.8, 47, 47, 47, 47, 82.2, 91, 91, 64.6, 11.8, 20.6, 20.6, 47, 47, 82.2, 3, 29.4, 29.4, 29.4, 91, 38.2, 64.6, 73.4, 11.8, 20.6, 29.4, 91, 47, 47, 47, 73.4, 55.8, 20.6, 20.6, 47, 47, 47, 47, 47, 3, 91, 91, 91, 73.4, 73.4, 20.6, 20.6, 47, 47, 47, 3, 91, 91, 91, 73.4, 73.4, 20.6, 20.6, 29.4, 29.4, 29.4, 73.4, 47, 47, 3, 91, 91, 20.6, 20.6, 47, 3, 91, 73.4, 20.6, 20.6)
# The labels are given at the end of the link. 
links_labels_12 = c("CTGF", "", "TFPI", "", "TIMP2", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "EFNA5", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "LTBP3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "APP", "", "", "", "SEMA6D", "", "", "", "", "", "", "", "", "", "", "NTN4", "SEMA3E", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "BTC", "HBEGF", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "") 

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_1, links_pos_1,
  links_pos_1, links_chromosomes_2, links_pos_2, links_pos_2,
  maxRadius = 0.5, width = "0.01em", color = 'cyan', labelSize = "0.75em", labels = links_labels_12)

links_chromosomes_3 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should start
links_chromosomes_4 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should end

# Now we start at the ligands
links_pos_3 = c(3, 3, 11.8, 11.8, 20.6, 20.6, 3, 11.8, 11.8, 11.8, 11.8, 20.6, 20.6, 3, 11.8, 20.6, 20.6, 20.6, 3, 20.6, 20.6, 3, 20.6, 20.6, 3, 20.6, 20.6, 3, 20.6, 20.6, 3, 3, 29.4, 11.8, 11.8, 29.4, 20.6, 20.6, 29.4, 29.4, 29.4, 3, 29.4, 29.4, 11.8, 11.8, 29.4, 29.4, 29.4, 11.8, 11.8, 20.6, 20.6, 3, 29.4, 29.4, 29.4, 38.2, 11.8, 20.6, 29.4, 20.6, 20.6, 3, 20.6, 20.6, 3, 20.6, 20.6, 29.4, 29.4, 29.4, 3, 20.6, 20.6, 3, 20.6, 20.6, 3, 3, 20.6, 20.6, 3, 20.6, 20.6, 3, 38.2, 20.6, 20.6, 20.6, 3, 20.6, 20.6, 3, 20.6, 20.6, 3, 20.6, 20.6, 3, 20.6, 20.6, 47, 3, 3, 29.4, 55.8, 11.8, 11.8, 47, 47, 29.4, 20.6, 20.6, 29.4, 29.4, 29.4, 64.6, 73.4, 47, 47, 3, 29.4, 29.4, 55.8, 11.8, 11.8, 47, 47, 29.4, 29.4, 29.4, 55.8, 11.8, 47, 47, 47, 47, 82.2, 91, 91, 64.6, 11.8, 20.6, 20.6, 47, 47, 82.2, 3, 29.4, 29.4, 29.4, 91, 38.2, 64.6, 73.4, 11.8, 20.6, 29.4, 91, 47, 47, 47, 73.4, 55.8, 20.6, 20.6, 47, 47, 47, 47, 47, 3, 91, 91, 91, 73.4, 73.4, 20.6, 20.6, 47, 47, 47, 3, 91, 91, 91, 73.4, 73.4, 20.6, 20.6, 29.4, 29.4, 29.4, 73.4, 47, 47, 3, 91, 91, 20.6, 20.6, 47, 3, 91, 73.4, 20.6, 20.6)
# Now we end at the receptors
links_pos_4 = c(3, 11.67, 3, 20.33, 29, 37.67, 3, 3, 20.33, 20.33, 46.33, 29, 37.67, 55, 20.33, 29, 29, 37.67, 63.67, 29, 37.67, 63.67, 29, 37.67, 63.67, 29, 37.67, 63.67, 29, 37.67, 3, 11.67, 72.33, 3, 20.33, 81, 29, 37.67, 89.67, 98.33, 107, 3, 72.33, 81, 3, 20.33, 115.67, 72.33, 81, 20.33, 46.33, 29, 37.67, 55, 115.67, 124.33, 72.33, 133, 20.33, 29, 98.33, 29, 37.67, 63.67, 29, 37.67, 63.67, 29, 37.67, 81, 141.67, 81, 63.67, 29, 37.67, 63.67, 29, 37.67, 3, 11.67, 29, 37.67, 3, 29, 37.67, 55, 133, 29, 29, 37.67, 63.67, 29, 37.67, 63.67, 29, 37.67, 63.67, 29, 37.67, 63.67, 29, 37.67, 3, 3, 11.67, 72.33, 150.33, 3, 20.33, 159, 167.67, 81, 29, 37.67, 89.67, 98.33, 107, 176.33, 185, 3, 167.67, 3, 72.33, 81, 150.33, 3, 20.33, 193.67, 167.67, 115.67, 72.33, 81, 150.33, 20.33, 193.67, 159, 202.33, 167.67, 211, 219.67, 228.33, 176.33, 46.33, 29, 37.67, 193.67, 167.67, 55, 55, 115.67, 124.33, 72.33, 55, 133, 176.33, 185, 20.33, 29, 98.33, 228.33, 193.67, 202.33, 167.67, 185, 150.33, 29, 37.67, 237, 193.67, 202.33, 245.67, 167.67, 63.67, 219.67, 254.33, 228.33, 185, 263, 29, 37.67, 202.33, 245.67, 167.67, 63.67, 219.67, 254.33, 228.33, 185, 263, 29, 37.67, 81, 141.67, 81, 185, 202.33, 167.67, 63.67, 219.67, 228.33, 29, 37.67, 202.33, 63.67, 228.33, 185, 29, 37.67)
# These links are for the receptors
links_labels_34 = c("LRP1", "LRP6", "", "VLDLR", "ITGA3", "ITGB1", "", "", "", "", "SDC4", "", "", "ERBB4", "", "", "", "", "NTRK1", "", "", "", "", "", "", "", "", "", "", "", "", "", "EPHB1", "", "", "EPHB6", "", "", "EPHA4", "EPHA5", "EPHA7", "", "", "", "", "", "EPHA10", "", "", "", "", "", "", "", "", "EPHA3", "", "ITBG5", "", "", "", "", "", "", "", "", "", "", "", "", "EPHB2", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "PLXNA1", "", "", "NCSTN", "TNFRSF21", "", "", "", "", "", "", "UNC5A", "NRP1", "", "", "", "", "", "", "", "", "GPC1", "", "", "", "", "", "", "", "", "NGFR", "", "ERBB3", "CD44", "CD9", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "CAV1", "", "", "SLC45A3", "", "", "", "CD82", "", "", "PLXND1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_3, links_pos_3,
  links_pos_3, links_chromosomes_4, links_pos_4, links_pos_4,
  maxRadius = 0.5, width = "0.01em", color = 'cyan', labelSize = "0.75em", labels = links_labels_34, labelPadding = 3)

#links_chromosomes_5 = c('ligand', 'hDRG', 'hDRG') # Chromosomes on which the links should start
#links_chromosomes_6 = c('RA', 'RA', 'RA') # Chromosomes on which the links should end
#links_pos_5 = c(180.08, 187.46, 194.84, 202.22, 209.59, 216.97, 224.35, 231.73, 239.11, 216.97, 246.49, 253.86, 261.24, 268.62, 276)
#links_pos_6 = c(48, 55.5, 55.5, 55.5, 55.5, 55.5, 55.5, 55.5, 63, 63, 63, 63, 70.5, 70.5, 78)
#links_labels_56 = c("", "", '', '', '', '', '', '', "", '', '', '', "", '', "")

#tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_5, links_pos_5,
#  links_pos_5, links_chromosomes_6, links_pos_6, links_pos_6,
#  maxRadius = 0.5, width = "0.01em", color = 'brown', labelSize = "0.5em", labels = links_labels_56)

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra = c('Receptor') # Chromosomes on which the arcs should be displayed
arcs_begin_ra = c(0)
arcs_end_ra = c(266)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra, arcs_begin_ra, arcs_end_ra, minRadius = 1.25, maxRadius = 1.30, colors = 'forestgreen') 

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_hdrg = c('Ligand') # Chromosomes on which the arcs should be displayed
arcs_begin_hdrg = c(0)
arcs_end_hdrg = c(94)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_hdrg, arcs_begin_hdrg, arcs_end_hdrg, minRadius = 1.25, maxRadius = 1.30, colors = 'deeppink3')

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Human RA to Mouse DRG - All Neurons', x = -0.7, y = -1.37, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Ligand', x = 0.85, y = -1.1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Receptor', x = -1.05, y = -1.1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes, arcs_begin, arcs_end, minRadius = 0.85, maxRadius = 0.99, colors = 'forestgreen') 

BioCircos(tracklist, genome = myGenome, genomeFillColor = c('white', 'white'), chrPad = 0.05, displayGenomeBorder = FALSE, yChr =  FALSE,
  genomeTicksDisplay = FALSE,  genomeLabelTextSize = 0, genomeLabelDy = 0, width = 650, height = 650)
```

Below we make a legend for the fibroblast subtypes.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (3)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (5)', 'DKK3+ sublining fibroblasts (SC-F3) (3)', 'CD55+ lining fibroblasts (SC-F4) (11)'), pch=16, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes", at=0.24, cex=2)
```

Below we plot the legend for the cell types.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Cerebellum Neurons (5)', 'Cholinergic and Monoaminergic Neurons (5)', 'Di- and Mesencphalon Excitatory Neurons (6)', 'Di- and Mesencphalon Inhibitory Neurons (7)', 'Enteric Neurons (11)', 'Hindbrain Neurons (11)', 'Peripheral Sensory Neurofilament Neurons (7)', 'Peripheral Sensory Non-peptidergic Neurons (13)', 'Peripheral Sensory Peptidergic Neurons (11)', 'Sympathetic Cholinergic Neurons (7)', 'Sympathetic Noradrenergic Neurons (6)', 'Dentate Gyrus Granule Neurons (5)', 'Olfactory Inhibitory Neurons (1)', 'Spinal Cord Excitatory Neurons (1)', 'Spinal Cord Inhibitory Neurons (3)', 'Peptidergic Neurons (1)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('antiquewhite4', 'black', 'blue', 'blueviolet', 'brown', 'cadetblue1', 'gold', 'darkgrey', 'magenta', 'chartreuse3', 'cyan', 'orange', 'coral', 'darkgreen', 'firebrick', 'purple'))
mtext("mDRG Neuronal Cell Types", at=0.19, cex=1.1)
```

Below we create one plot with both mDRG to human RA and human RA to mDRG.

```{r, fig.height=10,fig.width=10}
library(BioCircos)

# Chromosomes on which the Receptor points should be displayed
points_chromosomes_ligand = c('Ligand', 'Ligand')
# Coordinates on which the Receptor points should be displayed (length.out = 20 because there are 20 unique genes in the receptor column
points_coordinates_ligand = seq(3, 85, length.out = 20)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique ligands
points_values_ligand = 1:length(points_coordinates_ligand)

# Size of the outside of the circle for ligands and receptors
# Since there are uniquely 20 ligands and 62 receptor genes, we make the slices of the circle 20/82*360 = 88 for the receptors and 360 - 88 for the ligands
myGenome = list("Ligand" = 88,
        "Receptor" = 272)

# Create circle for cell type identification to lie in
tracklist = BioCircosBackgroundTrack("myBackgroundTrack", minRadius = 0.83, maxRadius = 1.28,
  borderColors = "#AAAAAA", borderSize = 0.6, fillColors = "White") 

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("blue", "white", "white", "blue", "blue", "white", "white", "blue", "white", "white", "blue", "blue", "blue", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.26, maxRadius = 1.26, shape = 'rect', size = 4)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("green", "white", "white", "white", "white", "green", "white", "white", "white", "white", "green", "green", "green", "green", "green", "white", "white", "white", "white", "white"), minRadius = 1.13, maxRadius = 1.13, shape = 'rect', size = 4)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("orange", "white", "white", "orange", "white", "orange", "orange", "white", "orange", "white", "orange", "white", "orange", "white", "orange", "white", "white", "white", "white", "white"), minRadius = 1.00, maxRadius = 1.00, shape = 'rect', size = 4)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("red", "red", "red", "white", "red", "red", "white", "white", "red", "red", "red", "red", "red", "red", "red", "red", "red", "red", "red", "red", "red"), minRadius = 0.86, maxRadius = 0.86, shape = 'rect', size = 4)

# Chromosomes on which the ligand should be displayed
points_chromosomes_receptor = c('Receptor', 'Receptor')
# Coordinates on which the ligand should be displayed length.out = 62
points_coordinates_receptor = seq(3, 269, length.out = 62)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique receptors
points_values_receptor = 1:length(points_chromosomes_receptor)

# Cerebellum neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("antiquewhite4", "antiquewhite4", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "antiquewhite4", "antiquewhite4", "antiquewhite4", "white", "white", "white", "white", "white", "antiquewhite4", "white", "white", "white", "white", "white", "white", "white", "white", "antiquewhite4", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.85, maxRadius = 0.85)

# Cholinergic and monoaminergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "black", "black", "black", "black", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "black", "black", "white", "white", "white", "black", "white", "white", "white", "white", "white", "white", "white", "white", "black", "black", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.877, maxRadius = 0.877)

# Di- and mesencphalon excitatory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "blue", "white", "white", "blue", "white", "white", "white", "blue", "blue", "blue", "blue", "blue", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "blue", "white", "blue", "white", "white", "white", "white", "white", "blue", "blue", "white", "white", "white", "white", "white", "white", "white", "blue", "white", "blue", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.903, maxRadius = 0.903)

# Di- and mesencphalon inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("blueviolet", "blueviolet", "blueviolet", "white", "blueviolet", "white", "white", "white", "blueviolet", "blueviolet", "blueviolet", "blueviolet", "blueviolet", "blueviolet", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "blueviolet", "white", "white", "white", "white", "white", "blueviolet", "blueviolet", "white", "white", "white", "blueviolet", "white", "white", "white", "blueviolet", "white", "blueviolet", "white", "white", "blueviolet", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.93, maxRadius = 0.93)

# Enteric neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "brown", "brown", "white", "white", "brown", "brown", "white", "brown", "brown", "brown", "white", "white", "brown", "brown", "brown", "brown", "brown", "brown", "brown", "brown", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "brown", "brown", "brown", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "brown", "brown", "brown", "white", "brown", "brown", "brown", "brown", "brown", "white", "white", "white", "white"), minRadius = 0.957, maxRadius = 0.957)

# Hindbrain neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "cadetblue1", "cadetblue1", "white", "cadetblue1", "white", "white", "white", "cadetblue1", "cadetblue1", "white", "cadetblue1", "cadetblue1", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cadetblue1", "cadetblue1", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cadetblue1", "cadetblue1", "white", "white", "cadetblue1", "white", "cadetblue1", "white", "white", "white", "white", "cadetblue1", "cadetblue1", "cadetblue1", "white", "white", "white", "cadetblue1", "cadetblue1", "cadetblue1", "cadetblue1", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.983, maxRadius = 0.983)

# Peripheral sensory neurofilament neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "gold", "white", "gold", "white", "gold", "white", "white", "white", "gold", "gold", "gold", "gold", "white", "white", "white", "white", "gold", "white", "white", "white", "white", "gold", "gold", "gold", "gold", "gold", "white", "white", "white", "white", "white", "white", "gold", "gold", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "gold", "white", "gold", "white", "gold", "gold", "gold", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.016, maxRadius = 1.016)

# Peripheral sensory non-peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "darkgrey", "white", "white", "white", "darkgrey", "white", "white", "white", "white", "white", "white", "white", "darkgrey", "white", "white", "white", "darkgrey", "white", "white", "white", "white", "darkgrey", "darkgrey", "white", "white", "white", "darkgrey", "white", "white", "white", "white", "white", "darkgrey", "darkgrey", "white", "white", "darkgrey", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "darkgrey", "white", "darkgrey", "darkgrey", "darkgrey", "white", "darkgrey", "darkgrey", "darkgrey", "darkgrey", "darkgrey", "darkgrey"), minRadius = 1.037, maxRadius = 1.037)

# Peripheral sensory peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "magenta", "white", "white", "white", "magenta", "white", "white", "magenta", "magenta", "white", "white", "white", "white", "white", "white", "white", "magenta", "white", "white", "white", "white", "magenta", "white", "white", "magenta", "white", "white", "white", "white", "white", "white", "white", "magenta", "magenta", "white", "white", "magenta", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "magenta", "white", "magenta", "white", "magenta", "white", "magenta", "magenta", "white", "magenta", "magenta", "magenta"), minRadius = 1.063, maxRadius = 1.063)

# Sympathetic cholinergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "white", "chartreuse3", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "white", "white", "white", "chartreuse3", "chartreuse3", "white", "white", "white", "chartreuse3", "chartreuse3", "white", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "chartreuse3", "white", "chartreuse3", "white", "white", "chartreuse3", "white", "white", "white", "white"), minRadius = 1.09, maxRadius = 1.09)

# Sympathetic nonadrenergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "cyan", "cyan", "white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cyan", "white", "cyan", "white", "white", "cyan", "white", "white", "white", "white"), minRadius = 1.117, maxRadius = 1.117)

# Dendate gyrus granule neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "orange", "white", "white", "orange", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "orange", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.143, maxRadius = 1.143)

# Olfactory inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "coral", "white", "white", "white", "white", "white", "white", "coral", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "coral", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "coral", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.17, maxRadius = 1.17)

# Spinal cord excitatory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "darkgreen", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.197, maxRadius = 1.197)

# Spinal cord inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "firebrick", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "firebrick", "white", "white", "white", "white", "white", "white", "firebrick", "white", "white", "white", "white", "firebrick", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.223, maxRadius = 1.223)

# Peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "purple", "white", "white", "white", "white"), minRadius = 1.25, maxRadius = 1.25)

#tracklist = tracklist + BioCircosTextTrack("testText", 'mDRG to Synovial Fibroblast', weight = "bold", size = '1em', x = -0.46, y = -1.25, displayGenomeBorder = TRUE)

links_chromosomes_1 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should start
links_chromosomes_2 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should end

# Coordinates to place start of links. The coordinates below correspond to the sequence given by seq(3, 85, length.out = 20) Their starting coordinate must be repeated for the number of connections between them and the number of ligands they are connected to.
links_pos_1 = c(3, 3, 3, 41.84, 3, 3, 3, 3, 3, 3, 3, 41.84, 3, 3, 7.32, 11.63, 3, 15.95, 3, 3, 15.95, 11.63, 3, 20.26, 20.26, 24.58, 24.58, 24.58, 28.89, 24.58, 41.84, 41.84, 24.58, 20.26, 24.58, 7.32, 20.26, 3, 3, 20.26, 20.26, 24.58, 3, 3, 24.58, 24.58, 3, 3, 28.89, 24.58, 41.84, 41.84, 24.58, 3, 3, 20.26, 24.58, 7.32, 20.26, 33.21, 3, 3, 20.26, 20.26, 3, 20.26, 15.95, 15.95, 24.58, 24.58, 3, 3, 24.58, 24.58, 24.58, 3, 24.58, 24.58, 24.58, 3, 3, 24.58, 28.89, 24.58, 24.58, 3, 24.58, 15.95, 15.95, 24.58, 24.58, 3, 3, 20.26, 24.58, 20.26, 24.58, 24.58, 3, 20.26, 24.58, 3, 3, 20.26, 20.26, 24.58, 3, 3, 24.58, 24.58, 3, 3, 28.89, 24.58, 24.58, 41.84, 41.84, 3, 3, 20.26, 24.58, 7.32, 20.26, 3, 20.26, 3, 3, 3, 20.26, 3, 3, 3, 3, 3, 15.95, 24.58, 24.58, 24.58, 3, 3, 3, 3, 3, 24.58, 24.58, 24.58, 24.58, 3, 3, 3, 3, 3, 24.58, 37.53, 15.95, 24.58, 41.84, 24.58, 24.58, 3, 3, 3, 3, 3, 24.58, 7.32, 37.53, 33.21, 3, 3, 3, 24.58, 24.58, 3, 3, 3, 24.58, 24.58, 37.53, 3, 3, 3, 24.58, 24.58, 37.53, 3, 3, 3, 3, 3, 20.26, 15.95, 24.58, 24.58, 24.58, 3, 3, 24.58, 24.58, 24.58, 3, 3, 15.95, 24.58, 24.58, 24.58, 3, 3, 20.26, 41.84, 3, 3, 20.26, 15.95, 3, 3, 24.58, 24.58, 3, 3, 24.58, 24.58, 15.95, 3, 3, 24.58, 20.26, 24.58, 3, 15.95, 15.95, 3, 3, 15.95, 15.95, 3, 46.16, 46.16, 50.47, 50.47, 54.79, 54.79, 46.16, 50.47, 50.47, 50.47, 50.47, 54.79, 54.79, 46.16, 50.47, 54.79, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 46.16, 59.11, 50.47, 50.47, 59.11, 54.79, 54.79, 59.11, 59.11, 59.11, 46.16, 59.11, 59.11, 50.47, 50.47, 59.11, 59.11, 59.11, 50.47, 50.47, 54.79, 54.79, 46.16, 59.11, 59.11, 59.11, 63.42, 50.47, 54.79, 59.11, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 59.11, 59.11, 59.11, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 63.42, 54.79, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 41.84, 46.16, 46.16, 59.11, 67.74, 50.47, 50.47, 41.84, 41.84, 59.11, 54.79, 54.79, 59.11, 59.11, 59.11, 72.05, 76.37, 41.84, 41.84, 46.16, 59.11, 59.11, 67.74, 50.47, 50.47, 41.84, 41.84, 59.11, 59.11, 59.11, 67.74, 50.47, 41.84, 41.84, 41.84, 41.84, 80.68, 85, 85, 72.05, 50.47, 54.79, 54.79, 41.84, 41.84, 80.68, 46.16, 59.11, 59.11, 59.11, 85, 63.42, 72.05, 76.37, 50.47, 54.79, 59.11, 85, 41.84, 41.84, 41.84, 76.37, 67.74, 54.79, 54.79, 41.84, 41.84, 41.84, 41.84, 41.84, 46.16, 85, 85, 85, 76.37, 76.37, 54.79, 54.79, 41.84, 41.84, 41.84, 46.16, 85, 85, 85, 76.37, 76.37, 54.79, 54.79, 59.11, 59.11, 59.11, 76.37, 41.84, 41.84, 46.16, 85, 85, 54.79, 54.79, 41.84, 46.16, 85, 76.37, 54.79, 54.79)

# Coordinates to place end of links. The coordinates below correspond to the sequence given by seq(3, 269, length.out = 62)
links_pos_2 = c(3, 3, 3, 7.36, 3, 11.72, 16.08, 3, 16.08, 3, 16.08, 7.36, 3, 16.08, 20.44, 24.80, 29.16, 33.52, 29.16, 29.16, 33.52, 24.80, 29.16, 37.89, 42.25, 46.61, 50.97, 46.61, 37.89, 50.97, 55.33, 7.36, 46.61, 37.89, 50.97, 20.44, 42.25, 59.69, 3, 37.89, 42.25, 46.61, 59.69, 3, 50.97, 46.61, 59.69, 3, 37.89, 50.97, 55.33, 7.36, 46.61, 59.69, 3, 37.89, 50.97, 20.44, 42.25, 64.05, 11.72, 29.16, 68.41, 72.77, 16.08, 42.25, 77.13, 33.52, 81.49, 46.61, 11.72, 29.16, 85.85, 90.21, 50.97, 16.08, 94.57, 81.49, 46.61, 11.72, 29.16, 85.85, 72.77, 90.21, 50.97, 16.08, 94.57, 77.13, 33.52, 81.49, 46.61, 11.72, 29.16, 68.41, 85.85, 72.77, 90.21, 50.97, 16.08, 42.25, 94.57, 11.72, 98.93, 37.89, 42.25, 103.30, 11.72, 98.93, 50.97, 103.30, 11.72, 98.93, 37.89, 50.97, 103.30, 55.33, 7.36, 11.72, 98.93, 37.89, 50.97, 20.44, 42.25, 107.66, 42.25, 107.66, 107.66, 107.66, 42.25, 11.72, 59.69, 107.66, 29.16, 112.02, 116.38, 103.30, 81.49, 46.61, 11.72, 59.69, 107.66, 29.16, 112.02, 50.97, 103.30, 81.49, 46.61, 11.72, 59.69, 107.66, 29.16, 112.02, 50.97, 120.74, 116.38, 103.30, 55.33, 81.49, 46.61, 11.72, 59.69, 107.66, 29.16, 112.02, 50.97, 20.44, 120.74, 64.05, 11.72, 107.66, 29.16, 103.30, 81.49, 11.72, 107.66, 29.16, 103.30, 81.49, 125.10, 11.72, 107.66, 29.16, 103.30, 81.49, 125.10, 11.72, 107.66, 29.16, 11.72, 29.16, 42.25, 116.38, 103.30, 81.49, 46.61, 11.72, 29.16, 103.30, 81.49, 46.61, 11.72, 29.16, 116.38, 103.30, 81.49, 46.61, 20.77, 29.16, 42.25, 7.36, 11.72, 129.46, 42.25, 77.13, 11.72, 129.46, 85.85, 133.82, 11.72, 129.46, 85.85, 133.82, 77.13, 11.72, 129.46, 85.85, 42.25, 133.82, 11.72, 77.13, 116.38, 11.72, 11.72, 77.13, 116.38, 11.72, 138.18, 142.54, 138.18, 146.90, 151.26, 155.62, 138.18, 138.18, 146.90, 146.90, 159.98, 151.26, 155.62, 164.34, 146.90, 151.26, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 142.54, 173.07, 138.18, 146.90, 177.43, 151.26, 155.62, 181.79, 186.15, 190.51, 138.18, 173.07, 177.43, 138.18, 146.90, 194.87, 173.07, 177.43, 146.90, 159.98, 151.26, 155.62, 164.34, 194.87, 199.23, 173.07, 203.59, 146.90, 151.26, 186.15, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 177.43, 207.95, 177.43, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 142.54, 151.26, 155.62, 138.18, 151.26, 155.62, 164.34, 203.59, 151.26, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 138.18, 142.54, 173.07, 212.31, 138.18, 146.90, 216.67, 221.03, 177.43, 151.26, 155.62, 181.79, 186.15, 190.51, 225.39, 229.75, 138.18, 221.03, 138.18, 173.07, 177.43, 212.31, 138.18, 146.90, 234.11, 221.03, 194.87, 173.07, 177.43, 212.31, 146.90, 234.11, 216.67, 238.48, 221.03, 242.84, 247.20, 251.56, 225.39, 159.98, 151.26, 155.62, 234.11, 221.03, 164.34, 164.34, 194.87, 199.23, 173.07, 164.34, 203.59, 225.39, 229.75, 146.90, 151.26, 186.15, 251.56, 234.11, 238.48, 221.03, 229.75, 212.31, 151.26, 155.62, 255.92, 234.11, 238.48, 260.28, 221.03, 168.70, 247.20, 264.64, 251.56, 229.75, 269, 151.26, 155.62, 238.48, 260.28, 221.03, 168.70, 247.20, 264.64, 251.56, 229.75, 269, 151.26, 155.62, 177.43, 207.95, 177.43, 229.75, 238.48, 221.03, 168.70, 247.20, 251.56, 151.26, 155.62, 238.48, 168.70, 251.56, 229.75, 151.26, 155.62)


# The labels are given at the end of the link. 
links_labels_12 = c("FGF9", "", "", "CLSTN3", "", "FGF1", "NCAM1", "", "", "", "", "", "", "", "NFASC", "CXCL12", "FGF13", "TGFB2", "", "", "", "", "", "LRRC4B", "PTN", "EDIL3", "LTBP3", "", "", "", "CLSTN1", "", "", "", "", "", "", "FGF11", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "EFEMP2", "", "", "HSPG2", "LRFN4", "", "", "TGFB1", "", "CYR61", "", "", "", "ITGB3BP", "LTBP1", "", "", "SORBS1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF14", "", "", "ADAM9", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF12", "", "", "", "", "", "", "", "", "", "FGF18", "TGFB3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "SHH", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "DHH", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF7", "", "", "", "", "", "SERPINE1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "LRP1", "LRP6", "", "VLDLR", "ITGA3", "ITGB1", "", "", "", "", "SDC4", "", "", "ERBB4", "", "", "", "", "NTRK1", "", "", "", "", "", "", "", "", "", "", "", "", "", "EPHB1", "", "", "EPHB6", "", "", "EPHA4", "EPHA5", "EPHA7", "", "", "", "", "", "EPHA10", "", "", "", "", "", "", "", "", "EPHA3", "", "ITBG5", "", "", "", "", "", "", "", "", "", "", "", "", "EPHB2", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "PLXNA1", "", "", "NCSTN", "TNFRSF21", "", "", "", "", "", "", "UNC5A", "NRP1", "", "", "", "", "", "", "", "", "GPC1", "", "", "", "", "", "", "", "", "NGFR", "", "ERBB3", "CD44", "CD9", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "CAV1", "", "", "SLC45A3", "", "", "", "CD82", "", "", "PLXND1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "") 

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_1, links_pos_1,
  links_pos_1, links_chromosomes_2, links_pos_2, links_pos_2,
  maxRadius = 0.55, width = "0.01em", color = 'brown', labelSize = "0.75em", labels = links_labels_12)

links_chromosomes_3 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should start
links_chromosomes_4 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should end

# Now we start at the ligands
links_pos_3 = c(3, 3, 3, 7.36, 3, 11.72, 16.08, 3, 16.08, 3, 16.08, 7.36, 3, 16.08, 20.44, 24.80, 29.16, 33.52, 29.16, 29.16, 33.52, 24.80, 29.16, 37.89, 42.25, 46.61, 50.97, 46.61, 37.89, 50.97, 55.33, 7.36, 46.61, 37.89, 50.97, 20.44, 42.25, 59.69, 3, 37.89, 42.25, 46.61, 59.69, 3, 50.97, 46.61, 59.69, 3, 37.89, 50.97, 55.33, 7.36, 46.61, 59.69, 3, 37.89, 50.97, 20.44, 42.25, 64.05, 11.72, 29.16, 68.41, 72.77, 16.08, 42.25, 77.13, 33.52, 81.49, 46.61, 11.72, 29.16, 85.85, 90.21, 50.97, 16.08, 94.57, 81.49, 46.61, 11.72, 29.16, 85.85, 72.77, 90.21, 50.97, 16.08, 94.57, 77.13, 33.52, 81.49, 46.61, 11.72, 29.16, 68.41, 85.85, 72.77, 90.21, 50.97, 16.08, 42.25, 94.57, 11.72, 98.93, 37.89, 42.25, 103.30, 11.72, 98.93, 50.97, 103.30, 11.72, 98.93, 37.89, 50.97, 103.30, 55.33, 7.36, 11.72, 98.93, 37.89, 50.97, 20.44, 42.25, 107.66, 42.25, 107.66, 107.66, 107.66, 42.25, 11.72, 59.69, 107.66, 29.16, 112.02, 116.38, 103.30, 81.49, 46.61, 11.72, 59.69, 107.66, 29.16, 112.02, 50.97, 103.30, 81.49, 46.61, 11.72, 59.69, 107.66, 29.16, 112.02, 50.97, 120.74, 116.38, 103.30, 55.33, 81.49, 46.61, 11.72, 59.69, 107.66, 29.16, 112.02, 50.97, 20.44, 120.74, 64.05, 11.72, 107.66, 29.16, 103.30, 81.49, 11.72, 107.66, 29.16, 103.30, 81.49, 125.10, 11.72, 107.66, 29.16, 103.30, 81.49, 125.10, 11.72, 107.66, 29.16, 11.72, 29.16, 42.25, 116.38, 103.30, 81.49, 46.61, 11.72, 29.16, 103.30, 81.49, 46.61, 11.72, 29.16, 116.38, 103.30, 81.49, 46.61, 20.77, 29.16, 42.25, 7.36, 11.72, 129.46, 42.25, 77.13, 11.72, 129.46, 85.85, 133.82, 11.72, 129.46, 85.85, 133.82, 77.13, 11.72, 129.46, 85.85, 42.25, 133.82, 11.72, 77.13, 116.38, 11.72, 11.72, 77.13, 116.38, 11.72, 138.18, 142.54, 138.18, 146.90, 151.26, 155.62, 138.18, 138.18, 146.90, 146.90, 159.98, 151.26, 155.62, 164.34, 146.90, 151.26, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 142.54, 173.07, 138.18, 146.90, 177.43, 151.26, 155.62, 181.79, 186.15, 190.51, 138.18, 173.07, 177.43, 138.18, 146.90, 194.87, 173.07, 177.43, 146.90, 159.98, 151.26, 155.62, 164.34, 194.87, 199.23, 173.07, 203.59, 146.90, 151.26, 186.15, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 177.43, 207.95, 177.43, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 142.54, 151.26, 155.62, 138.18, 151.26, 155.62, 164.34, 203.59, 151.26, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 138.18, 142.54, 173.07, 212.31, 138.18, 146.90, 216.67, 221.03, 177.43, 151.26, 155.62, 181.79, 186.15, 190.51, 225.39, 229.75, 138.18, 221.03, 138.18, 173.07, 177.43, 212.31, 138.18, 146.90, 234.11, 221.03, 194.87, 173.07, 177.43, 212.31, 146.90, 234.11, 216.67, 238.48, 221.03, 242.84, 247.20, 251.56, 225.39, 159.98, 151.26, 155.62, 234.11, 221.03, 164.34, 164.34, 194.87, 199.23, 173.07, 164.34, 203.59, 225.39, 229.75, 146.90, 151.26, 186.15, 251.56, 234.11, 238.48, 221.03, 229.75, 212.31, 151.26, 155.62, 255.92, 234.11, 238.48, 260.28, 221.03, 168.70, 247.20, 264.64, 251.56, 229.75, 269, 151.26, 155.62, 238.48, 260.28, 221.03, 168.70, 247.20, 264.64, 251.56, 229.75, 269, 151.26, 155.62, 177.43, 207.95, 177.43, 229.75, 238.48, 221.03, 168.70, 247.20, 251.56, 151.26, 155.62, 238.48, 168.70, 251.56, 229.75, 151.26, 155.62)
# Now we end at the receptors
links_pos_4 = c(3, 3, 3, 41.84, 3, 3, 3, 3, 3, 3, 3, 41.84, 3, 3, 7.32, 11.63, 3, 15.95, 3, 3, 15.95, 11.63, 3, 20.26, 20.26, 24.58, 24.58, 24.58, 28.89, 24.58, 41.84, 41.84, 24.58, 20.26, 24.58, 7.32, 20.26, 3, 3, 20.26, 20.26, 24.58, 3, 3, 24.58, 24.58, 3, 3, 28.89, 24.58, 41.84, 41.84, 24.58, 3, 3, 20.26, 24.58, 7.32, 20.26, 33.21, 3, 3, 20.26, 20.26, 3, 20.26, 15.95, 15.95, 24.58, 24.58, 3, 3, 24.58, 24.58, 24.58, 3, 24.58, 24.58, 24.58, 3, 3, 24.58, 28.89, 24.58, 24.58, 3, 24.58, 15.95, 15.95, 24.58, 24.58, 3, 3, 20.26, 24.58, 20.26, 24.58, 24.58, 3, 20.26, 24.58, 3, 3, 20.26, 20.26, 24.58, 3, 3, 24.58, 24.58, 3, 3, 28.89, 24.58, 24.58, 41.84, 41.84, 3, 3, 20.26, 24.58, 7.32, 20.26, 3, 20.26, 3, 3, 3, 20.26, 3, 3, 3, 3, 3, 15.95, 24.58, 24.58, 24.58, 3, 3, 3, 3, 3, 24.58, 24.58, 24.58, 24.58, 3, 3, 3, 3, 3, 24.58, 37.53, 15.95, 24.58, 41.84, 24.58, 24.58, 3, 3, 3, 3, 3, 24.58, 7.32, 37.53, 33.21, 3, 3, 3, 24.58, 24.58, 3, 3, 3, 24.58, 24.58, 37.53, 3, 3, 3, 24.58, 24.58, 37.53, 3, 3, 3, 3, 3, 20.26, 15.95, 24.58, 24.58, 24.58, 3, 3, 24.58, 24.58, 24.58, 3, 3, 15.95, 24.58, 24.58, 24.58, 3, 3, 20.26, 41.84, 3, 3, 20.26, 15.95, 3, 3, 24.58, 24.58, 3, 3, 24.58, 24.58, 15.95, 3, 3, 24.58, 20.26, 24.58, 3, 15.95, 15.95, 3, 3, 15.95, 15.95, 3, 46.16, 46.16, 50.47, 50.47, 54.79, 54.79, 46.16, 50.47, 50.47, 50.47, 50.47, 54.79, 54.79, 46.16, 50.47, 54.79, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 46.16, 59.11, 50.47, 50.47, 59.11, 54.79, 54.79, 59.11, 59.11, 59.11, 46.16, 59.11, 59.11, 50.47, 50.47, 59.11, 59.11, 59.11, 50.47, 50.47, 54.79, 54.79, 46.16, 59.11, 59.11, 59.11, 63.42, 50.47, 54.79, 59.11, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 59.11, 59.11, 59.11, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 63.42, 54.79, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 41.84, 46.16, 46.16, 59.11, 67.74, 50.47, 50.47, 41.84, 41.84, 59.11, 54.79, 54.79, 59.11, 59.11, 59.11, 72.05, 76.37, 41.84, 41.84, 46.16, 59.11, 59.11, 67.74, 50.47, 50.47, 41.84, 41.84, 59.11, 59.11, 59.11, 67.74, 50.47, 41.84, 41.84, 41.84, 41.84, 80.68, 85, 85, 72.05, 50.47, 54.79, 54.79, 41.84, 41.84, 80.68, 46.16, 59.11, 59.11, 59.11, 85, 63.42, 72.05, 76.37, 50.47, 54.79, 59.11, 85, 41.84, 41.84, 41.84, 76.37, 67.74, 54.79, 54.79, 41.84, 41.84, 41.84, 41.84, 41.84, 46.16, 85, 85, 85, 76.37, 76.37, 54.79, 54.79, 41.84, 41.84, 41.84, 46.16, 85, 85, 85, 76.37, 76.37, 54.79, 54.79, 59.11, 59.11, 59.11, 76.37, 41.84, 41.84, 46.16, 85, 85, 54.79, 54.79, 41.84, 46.16, 85, 76.37, 54.79, 54.79)
# These links are for the receptors
links_labels_34 = c('FGFR1', "", "", "APP", "", "", "", "", "", "", "", "", "", "", "ANK3", "ACKR3", "", "TGFBR3", "", "", "", "", "", "PTPRS", "", "ITGB5", "", "", "PTPRD", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "PLSCR4", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "CDON", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "CTGF", "", "TFPI", "", "TIMP2", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "EFNA5", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "LTBP3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "APP", "", "", "", "SEMA6D", "", "", "", "", "", "", "", "", "", "", "NTN4", "SEMA3E", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "BTC", "HBEGF", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_3, links_pos_3,
  links_pos_3, links_chromosomes_4, links_pos_4, links_pos_4,
  maxRadius = 0.55, width = "0.01em", color = 'brown', labelSize = "0.75em", labels = links_labels_34, labelPadding = 3)

links_chromosomes_5 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should start
links_chromosomes_6 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should end
links_pos_5 = c(138.18, 142.54, 138.18, 146.90, 151.26, 155.62, 138.18, 138.18, 146.90, 146.90, 159.98, 151.26, 155.62, 164.34, 146.90, 151.26, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 142.54, 173.07, 138.18, 146.90, 177.43, 151.26, 155.62, 181.79, 186.15, 190.51, 138.18, 173.07, 177.43, 138.18, 146.90, 194.87, 173.07, 177.43, 146.90, 159.98, 151.26, 155.62, 164.34, 194.87, 199.23, 173.07, 203.59, 146.90, 151.26, 186.15, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 177.43, 207.95, 177.43, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 142.54, 151.26, 155.62, 138.18, 151.26, 155.62, 164.34, 203.59, 151.26, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 138.18, 142.54, 173.07, 212.31, 138.18, 146.90, 216.67, 221.03, 177.43, 151.26, 155.62, 181.79, 186.15, 190.51, 225.39, 229.75, 138.18, 221.03, 138.18, 173.07, 177.43, 212.31, 138.18, 146.90, 234.11, 221.03, 194.87, 173.07, 177.43, 212.31, 146.90, 234.11, 216.67, 238.48, 221.03, 242.84, 247.20, 251.56, 225.39, 159.98, 151.26, 155.62, 234.11, 221.03, 164.34, 164.34, 194.87, 199.23, 173.07, 164.34, 203.59, 225.39, 229.75, 146.90, 151.26, 186.15, 251.56, 234.11, 238.48, 221.03, 229.75, 212.31, 151.26, 155.62, 255.92, 234.11, 238.48, 260.28, 221.03, 168.70, 247.20, 264.64, 251.56, 229.75, 269, 151.26, 155.62, 238.48, 260.28, 221.03, 168.70, 247.20, 264.64, 251.56, 229.75, 269, 151.26, 155.62, 177.43, 207.95, 177.43, 229.75, 238.48, 221.03, 168.70, 247.20, 251.56, 151.26, 155.62, 238.48, 168.70, 251.56, 229.75, 151.26, 155.62)
links_pos_6 = c(46.16, 46.16, 50.47, 50.47, 54.79, 54.79, 46.16, 50.47, 50.47, 50.47, 50.47, 54.79, 54.79, 46.16, 50.47, 54.79, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 46.16, 59.11, 50.47, 50.47, 59.11, 54.79, 54.79, 59.11, 59.11, 59.11, 46.16, 59.11, 59.11, 50.47, 50.47, 59.11, 59.11, 59.11, 50.47, 50.47, 54.79, 54.79, 46.16, 59.11, 59.11, 59.11, 63.42, 50.47, 54.79, 59.11, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 59.11, 59.11, 59.11, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 63.42, 54.79, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 41.84, 46.16, 46.16, 59.11, 67.74, 50.47, 50.47, 41.84, 41.84, 59.11, 54.79, 54.79, 59.11, 59.11, 59.11, 72.05, 76.37, 41.84, 41.84, 46.16, 59.11, 59.11, 67.74, 50.47, 50.47, 41.84, 41.84, 59.11, 59.11, 59.11, 67.74, 50.47, 41.84, 41.84, 41.84, 41.84, 80.68, 85, 85, 72.05, 50.47, 54.79, 54.79, 41.84, 41.84, 80.68, 46.16, 59.11, 59.11, 59.11, 85, 63.42, 72.05, 76.37, 50.47, 54.79, 59.11, 85, 41.84, 41.84, 41.84, 76.37, 67.74, 54.79, 54.79, 41.84, 41.84, 41.84, 41.84, 41.84, 46.16, 85, 85, 85, 76.37, 76.37, 54.79, 54.79, 41.84, 41.84, 41.84, 46.16, 85, 85, 85, 76.37, 76.37, 54.79, 54.79, 59.11, 59.11, 59.11, 76.37, 41.84, 41.84, 46.16, 85, 85, 54.79, 54.79, 41.84, 46.16, 85, 76.37, 54.79, 54.79)
links_labels_56 = c("", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", '', "", "", "", "", "", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_5, links_pos_5,
  links_pos_5, links_chromosomes_6, links_pos_6, links_pos_6,
  maxRadius = 0.55, width = "0.01em", color = 'cyan', labelSize = "0.5em", labels = links_labels_56)

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra = c('Ligand') # Chromosomes on which the arcs should be displayed
arcs_begin_ra = c(0)
arcs_end_ra = c(43)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra, arcs_begin_ra, arcs_end_ra, minRadius = 1.3, maxRadius = 1.35, colors = 'forestgreen')

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra2 = c('Ligand') # Chromosomes on which the arcs should be displayed
arcs_begin_ra2 = c(39)
arcs_end_ra2 = c(88)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra2, arcs_begin_ra2, arcs_end_ra2, minRadius = 1.35, maxRadius = 1.4, colors = 'deeppink3')

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_hdrg = c('Receptor') # Chromosomes on which the arcs should be displayed
arcs_begin_hdrg = c(0)
arcs_end_hdrg = c(138)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_hdrg, arcs_begin_hdrg, arcs_end_hdrg, minRadius = 1.35, maxRadius = 1.4, colors = 'deeppink3')

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_hdrg2 = c('Receptor') # Chromosomes on which the arcs should be displayed
arcs_begin_hdrg2 = c(139)
arcs_end_hdrg2 = c(272)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_hdrg2, arcs_begin_hdrg2, arcs_end_hdrg2, minRadius = 1.3, maxRadius = 1.35, colors = 'forestgreen')

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Mouse DRG to Human RA - Specific Neurons', x = -0.6, y = -1.30, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Receptor', x #= 0.75, y = -1, size = "1em",
##  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Ligand', x = #-0.95, y = -1, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes, arcs_begin, arcs_end, minRadius = 0.85, maxRadius = 0.99, colors = 'forestgreen') 

BioCircos(tracklist, genome = myGenome, genomeFillColor = c('white', 'white'), chrPad = 0.05, displayGenomeBorder = FALSE, yChr =  FALSE,
  genomeTicksDisplay = FALSE,  genomeLabelTextSize = 0, genomeLabelDy = 0, width = 650, height = 650)
```

Below we make a legend for the ligand and receptor tracts.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Receptor', 'Ligand'), bty='n', lwd = 8, cex = 1.2, col = c("forestgreen", "deeppink3"), lty = c(1, 1), pch = c(NA, NA))
 #   col = c('forestgreen', 'deeppink3'))
#mtext("mDRG Neuronal Cell Types", at=0.19, cex=1.1)
```

Below we create one plot with both mDRG to human RA and human RA to mDRG.

```{r, fig.height=10,fig.width=10}
library(BioCircos)

# Chromosomes on which the Receptor points should be displayed
points_chromosomes_ligand = c('Ligand', 'Ligand')
# Coordinates on which the Receptor points should be displayed (length.out = 20 because there are 20 unique genes in the receptor column
points_coordinates_ligand = seq(3, 85, length.out = 20)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique ligands
points_values_ligand = 1:length(points_coordinates_ligand)

# Size of the outside of the circle for ligands and receptors
# Since there are uniquely 20 ligands and 62 receptor genes, we make the slices of the circle 20/82*360 = 88 for the receptors and 360 - 88 for the ligands
myGenome = list("Ligand" = 88,
        "Receptor" = 272)

# Create circle for cell type identification to lie in
tracklist = BioCircosBackgroundTrack("myBackgroundTrack", minRadius = 0.83, maxRadius = 1.28,
  borderColors = "#AAAAAA", borderSize = 0.6, fillColors = "White") 

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("blue", "white", "white", "blue", "blue", "white", "white", "blue", "white", "white", "blue", "blue", "blue", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.26, maxRadius = 1.26)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("green", "white", "white", "white", "white", "green", "white", "white", "white", "white", "green", "green", "green", "green", "green", "white", "white", "white", "white", "white"), minRadius = 1.13, maxRadius = 1.13)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("orange", "white", "white", "orange", "white", "orange", "orange", "white", "orange", "white", "orange", "white", "orange", "white", "orange", "white", "white", "white", "white", "white"), minRadius = 1.00, maxRadius = 1.00)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("red", "red", "red", "white", "red", "red", "white", "white", "red", "red", "red", "red", "red", "red", "red", "red", "red", "red", "red", "red", "red"), minRadius = 0.86, maxRadius = 0.86)

# Chromosomes on which the ligand should be displayed
points_chromosomes_receptor = c('Receptor', 'Receptor')
# Coordinates on which the ligand should be displayed length.out = 62
points_coordinates_receptor = seq(3, 269, length.out = 62)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique receptors
points_values_receptor = 1:length(points_chromosomes_receptor)

# Cerebellum neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("antiquewhite4", "antiquewhite4", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "antiquewhite4", "antiquewhite4", "antiquewhite4", "white", "white", "white", "white", "white", "antiquewhite4", "white", "white", "white", "white", "white", "white", "white", "white", "antiquewhite4", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.85, maxRadius = 0.85)

# Cholinergic and monoaminergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "black", "black", "black", "black", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "black", "black", "white", "white", "white", "black", "white", "white", "white", "white", "white", "white", "white", "white", "black", "black", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.877, maxRadius = 0.877)

# Di- and mesencphalon excitatory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "blue", "white", "white", "blue", "white", "white", "white", "blue", "blue", "blue", "blue", "blue", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "blue", "white", "blue", "white", "white", "white", "white", "white", "blue", "blue", "white", "white", "white", "white", "white", "white", "white", "blue", "white", "blue", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.903, maxRadius = 0.903)

# Di- and mesencphalon inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("blueviolet", "blueviolet", "blueviolet", "white", "blueviolet", "white", "white", "white", "blueviolet", "blueviolet", "blueviolet", "blueviolet", "blueviolet", "blueviolet", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "blueviolet", "white", "white", "white", "white", "white", "blueviolet", "blueviolet", "white", "white", "white", "blueviolet", "white", "white", "white", "blueviolet", "white", "blueviolet", "white", "white", "blueviolet", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.93, maxRadius = 0.93)

# Enteric neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "brown", "brown", "white", "white", "brown", "brown", "white", "brown", "brown", "brown", "white", "white", "brown", "brown", "brown", "brown", "brown", "brown", "brown", "brown", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "brown", "brown", "brown", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "brown", "brown", "brown", "white", "brown", "brown", "brown", "brown", "brown", "white", "white", "white", "white"), minRadius = 0.957, maxRadius = 0.957)

# Hindbrain neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "cadetblue1", "cadetblue1", "white", "cadetblue1", "white", "white", "white", "cadetblue1", "cadetblue1", "white", "cadetblue1", "cadetblue1", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cadetblue1", "cadetblue1", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cadetblue1", "cadetblue1", "white", "white", "cadetblue1", "white", "cadetblue1", "white", "white", "white", "white", "cadetblue1", "cadetblue1", "cadetblue1", "white", "white", "white", "cadetblue1", "cadetblue1", "cadetblue1", "cadetblue1", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.983, maxRadius = 0.983)

# Peripheral sensory neurofilament neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "gold", "white", "gold", "white", "gold", "white", "white", "white", "gold", "gold", "gold", "gold", "white", "white", "white", "white", "gold", "white", "white", "white", "white", "gold", "gold", "gold", "gold", "gold", "white", "white", "white", "white", "white", "white", "gold", "gold", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "gold", "white", "gold", "white", "gold", "gold", "gold", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.016, maxRadius = 1.016)

# Peripheral sensory non-peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "darkgrey", "white", "white", "white", "darkgrey", "white", "white", "white", "white", "white", "white", "white", "darkgrey", "white", "white", "white", "darkgrey", "white", "white", "white", "white", "darkgrey", "darkgrey", "white", "white", "white", "darkgrey", "white", "white", "white", "white", "white", "darkgrey", "darkgrey", "white", "white", "darkgrey", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "darkgrey", "white", "darkgrey", "darkgrey", "darkgrey", "white", "darkgrey", "darkgrey", "darkgrey", "darkgrey", "darkgrey", "darkgrey"), minRadius = 1.037, maxRadius = 1.037)

# Peripheral sensory peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "magenta", "white", "white", "white", "magenta", "white", "white", "magenta", "magenta", "white", "white", "white", "white", "white", "white", "white", "magenta", "white", "white", "white", "white", "magenta", "white", "white", "magenta", "white", "white", "white", "white", "white", "white", "white", "magenta", "magenta", "white", "white", "magenta", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "magenta", "white", "magenta", "white", "magenta", "white", "magenta", "magenta", "white", "magenta", "magenta", "magenta"), minRadius = 1.063, maxRadius = 1.063)

# Sympathetic cholinergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "white", "chartreuse3", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "white", "white", "white", "chartreuse3", "chartreuse3", "white", "white", "white", "chartreuse3", "chartreuse3", "white", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "chartreuse3", "white", "chartreuse3", "white", "white", "chartreuse3", "white", "white", "white", "white"), minRadius = 1.09, maxRadius = 1.09)

# Sympathetic nonadrenergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "cyan", "cyan", "white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cyan", "white", "cyan", "white", "white", "cyan", "white", "white", "white", "white"), minRadius = 1.117, maxRadius = 1.117)

# Dendate gyrus granule neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "orange", "white", "white", "orange", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "orange", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.143, maxRadius = 1.143)

# Olfactory inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "coral", "white", "white", "white", "white", "white", "white", "coral", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "coral", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "coral", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.17, maxRadius = 1.17)

# Spinal cord excitatory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "darkgreen", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.197, maxRadius = 1.197)

# Spinal cord inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "firebrick", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "firebrick", "white", "white", "white", "white", "white", "white", "firebrick", "white", "white", "white", "white", "firebrick", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.223, maxRadius = 1.223)

# Peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "purple", "white", "white", "white", "white"), minRadius = 1.25, maxRadius = 1.25)

#tracklist = tracklist + BioCircosTextTrack("testText", 'mDRG to Synovial Fibroblast', weight = "bold", size = '1em', x = -0.46, y = -1.25, displayGenomeBorder = TRUE)

links_chromosomes_1 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should start
links_chromosomes_2 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should end

# Coordinates to place start of links. The coordinates below correspond to the sequence given by seq(3, 85, length.out = 20) Their starting coordinate must be repeated for the number of connections between them and the number of ligands they are connected to.
links_pos_1 = c(3, 3, 3, 41.84, 3, 3, 3, 3, 3, 3, 3, 41.84, 3, 3, 7.32, 11.63, 3, 15.95, 3, 3, 15.95, 11.63, 3, 20.26, 20.26, 24.58, 24.58, 24.58, 28.89, 24.58, 41.84, 41.84, 24.58, 20.26, 24.58, 7.32, 20.26, 3, 3, 20.26, 20.26, 24.58, 3, 3, 24.58, 24.58, 3, 3, 28.89, 24.58, 41.84, 41.84, 24.58, 3, 3, 20.26, 24.58, 7.32, 20.26, 33.21, 3, 3, 20.26, 20.26, 3, 20.26, 15.95, 15.95, 24.58, 24.58, 3, 3, 24.58, 24.58, 24.58, 3, 24.58, 24.58, 24.58, 3, 3, 24.58, 28.89, 24.58, 24.58, 3, 24.58, 15.95, 15.95, 24.58, 24.58, 3, 3, 20.26, 24.58, 20.26, 24.58, 24.58, 3, 20.26, 24.58, 3, 3, 20.26, 20.26, 24.58, 3, 3, 24.58, 24.58, 3, 3, 28.89, 24.58, 24.58, 41.84, 41.84, 3, 3, 20.26, 24.58, 7.32, 20.26, 3, 20.26, 3, 3, 3, 20.26, 3, 3, 3, 3, 3, 15.95, 24.58, 24.58, 24.58, 3, 3, 3, 3, 3, 24.58, 24.58, 24.58, 24.58, 3, 3, 3, 3, 3, 24.58, 37.53, 15.95, 24.58, 41.84, 24.58, 24.58, 3, 3, 3, 3, 3, 24.58, 7.32, 37.53, 33.21, 3, 3, 3, 24.58, 24.58, 3, 3, 3, 24.58, 24.58, 37.53, 3, 3, 3, 24.58, 24.58, 37.53, 3, 3, 3, 3, 3, 20.26, 15.95, 24.58, 24.58, 24.58, 3, 3, 24.58, 24.58, 24.58, 3, 3, 15.95, 24.58, 24.58, 24.58, 3, 3, 20.26, 41.84, 3, 3, 20.26, 15.95, 3, 3, 24.58, 24.58, 3, 3, 24.58, 24.58, 15.95, 3, 3, 24.58, 20.26, 24.58, 3, 15.95, 15.95, 3, 3, 15.95, 15.95, 3, 46.16, 46.16, 50.47, 50.47, 54.79, 54.79, 46.16, 50.47, 50.47, 50.47, 50.47, 54.79, 54.79, 46.16, 50.47, 54.79, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 46.16, 59.11, 50.47, 50.47, 59.11, 54.79, 54.79, 59.11, 59.11, 59.11, 46.16, 59.11, 59.11, 50.47, 50.47, 59.11, 59.11, 59.11, 50.47, 50.47, 54.79, 54.79, 46.16, 59.11, 59.11, 59.11, 63.42, 50.47, 54.79, 59.11, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 59.11, 59.11, 59.11, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 63.42, 54.79, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 41.84, 46.16, 46.16, 59.11, 67.74, 50.47, 50.47, 41.84, 41.84, 59.11, 54.79, 54.79, 59.11, 59.11, 59.11, 72.05, 76.37, 41.84, 41.84, 46.16, 59.11, 59.11, 67.74, 50.47, 50.47, 41.84, 41.84, 59.11, 59.11, 59.11, 67.74, 50.47, 41.84, 41.84, 41.84, 41.84, 80.68, 85, 85, 72.05, 50.47, 54.79, 54.79, 41.84, 41.84, 80.68, 46.16, 59.11, 59.11, 59.11, 85, 63.42, 72.05, 76.37, 50.47, 54.79, 59.11, 85, 41.84, 41.84, 41.84, 76.37, 67.74, 54.79, 54.79, 41.84, 41.84, 41.84, 41.84, 41.84, 46.16, 85, 85, 85, 76.37, 76.37, 54.79, 54.79, 41.84, 41.84, 41.84, 46.16, 85, 85, 85, 76.37, 76.37, 54.79, 54.79, 59.11, 59.11, 59.11, 76.37, 41.84, 41.84, 46.16, 85, 85, 54.79, 54.79, 41.84, 46.16, 85, 76.37, 54.79, 54.79)

# Coordinates to place end of links. The coordinates below correspond to the sequence given by seq(3, 269, length.out = 62)
links_pos_2 = c(3, 3, 3, 7.36, 3, 11.72, 16.08, 3, 16.08, 3, 16.08, 7.36, 3, 16.08, 20.44, 24.80, 29.16, 33.52, 29.16, 29.16, 33.52, 24.80, 29.16, 37.89, 42.25, 46.61, 50.97, 46.61, 37.89, 50.97, 55.33, 7.36, 46.61, 37.89, 50.97, 20.44, 42.25, 59.69, 3, 37.89, 42.25, 46.61, 59.69, 3, 50.97, 46.61, 59.69, 3, 37.89, 50.97, 55.33, 7.36, 46.61, 59.69, 3, 37.89, 50.97, 20.44, 42.25, 64.05, 11.72, 29.16, 68.41, 72.77, 16.08, 42.25, 77.13, 33.52, 81.49, 46.61, 11.72, 29.16, 85.85, 90.21, 50.97, 16.08, 94.57, 81.49, 46.61, 11.72, 29.16, 85.85, 72.77, 90.21, 50.97, 16.08, 94.57, 77.13, 33.52, 81.49, 46.61, 11.72, 29.16, 68.41, 85.85, 72.77, 90.21, 50.97, 16.08, 42.25, 94.57, 11.72, 98.93, 37.89, 42.25, 103.30, 11.72, 98.93, 50.97, 103.30, 11.72, 98.93, 37.89, 50.97, 103.30, 55.33, 7.36, 11.72, 98.93, 37.89, 50.97, 20.44, 42.25, 107.66, 42.25, 107.66, 107.66, 107.66, 42.25, 11.72, 59.69, 107.66, 29.16, 112.02, 116.38, 103.30, 81.49, 46.61, 11.72, 59.69, 107.66, 29.16, 112.02, 50.97, 103.30, 81.49, 46.61, 11.72, 59.69, 107.66, 29.16, 112.02, 50.97, 120.74, 116.38, 103.30, 55.33, 81.49, 46.61, 11.72, 59.69, 107.66, 29.16, 112.02, 50.97, 20.44, 120.74, 64.05, 11.72, 107.66, 29.16, 103.30, 81.49, 11.72, 107.66, 29.16, 103.30, 81.49, 125.10, 11.72, 107.66, 29.16, 103.30, 81.49, 125.10, 11.72, 107.66, 29.16, 11.72, 29.16, 42.25, 116.38, 103.30, 81.49, 46.61, 11.72, 29.16, 103.30, 81.49, 46.61, 11.72, 29.16, 116.38, 103.30, 81.49, 46.61, 20.77, 29.16, 42.25, 7.36, 11.72, 129.46, 42.25, 77.13, 11.72, 129.46, 85.85, 133.82, 11.72, 129.46, 85.85, 133.82, 77.13, 11.72, 129.46, 85.85, 42.25, 133.82, 11.72, 77.13, 116.38, 11.72, 11.72, 77.13, 116.38, 11.72, 138.18, 142.54, 138.18, 146.90, 151.26, 155.62, 138.18, 138.18, 146.90, 146.90, 159.98, 151.26, 155.62, 164.34, 146.90, 151.26, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 142.54, 173.07, 138.18, 146.90, 177.43, 151.26, 155.62, 181.79, 186.15, 190.51, 138.18, 173.07, 177.43, 138.18, 146.90, 194.87, 173.07, 177.43, 146.90, 159.98, 151.26, 155.62, 164.34, 194.87, 199.23, 173.07, 203.59, 146.90, 151.26, 186.15, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 177.43, 207.95, 177.43, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 142.54, 151.26, 155.62, 138.18, 151.26, 155.62, 164.34, 203.59, 151.26, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 138.18, 142.54, 173.07, 212.31, 138.18, 146.90, 216.67, 221.03, 177.43, 151.26, 155.62, 181.79, 186.15, 190.51, 225.39, 229.75, 138.18, 221.03, 138.18, 173.07, 177.43, 212.31, 138.18, 146.90, 234.11, 221.03, 194.87, 173.07, 177.43, 212.31, 146.90, 234.11, 216.67, 238.48, 221.03, 242.84, 247.20, 251.56, 225.39, 159.98, 151.26, 155.62, 234.11, 221.03, 164.34, 164.34, 194.87, 199.23, 173.07, 164.34, 203.59, 225.39, 229.75, 146.90, 151.26, 186.15, 251.56, 234.11, 238.48, 221.03, 229.75, 212.31, 151.26, 155.62, 255.92, 234.11, 238.48, 260.28, 221.03, 168.70, 247.20, 264.64, 251.56, 229.75, 269, 151.26, 155.62, 238.48, 260.28, 221.03, 168.70, 247.20, 264.64, 251.56, 229.75, 269, 151.26, 155.62, 177.43, 207.95, 177.43, 229.75, 238.48, 221.03, 168.70, 247.20, 251.56, 151.26, 155.62, 238.48, 168.70, 251.56, 229.75, 151.26, 155.62)


# The labels are given at the end of the link. 
links_labels_12 = c("FGF9", "", "", "CLSTN3", "", "FGF1", "NCAM1", "", "", "", "", "", "", "", "NFASC", "CXCL12", "FGF13", "TGFB2", "", "", "", "", "", "LRRC4B", "PTN", "EDIL3", "LTBP3", "", "", "", "CLSTN1", "", "", "", "", "", "", "FGF11", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "EFEMP2", "", "", "HSPG2", "LRFN4", "", "", "TGFB1", "", "CYR61", "", "", "", "ITGB3BP", "LTBP1", "", "", "SORBS1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF14", "", "", "ADAM9", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF12", "", "", "", "", "", "", "", "", "", "FGF18", "TGFB3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "SHH", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "DHH", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF7", "", "", "", "", "", "SERPINE1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "LRP1", "LRP6", "", "VLDLR", "ITGA3", "ITGB1", "", "", "", "", "SDC4", "", "", "ERBB4", "", "", "", "", "NTRK1", "", "", "", "", "", "", "", "", "", "", "", "", "", "EPHB1", "", "", "EPHB6", "", "", "EPHA4", "EPHA5", "EPHA7", "", "", "", "", "", "EPHA10", "", "", "", "", "", "", "", "", "EPHA3", "", "ITBG5", "", "", "", "", "", "", "", "", "", "", "", "", "EPHB2", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "PLXNA1", "", "", "NCSTN", "TNFRSF21", "", "", "", "", "", "", "UNC5A", "NRP1", "", "", "", "", "", "", "", "", "GPC1", "", "", "", "", "", "", "", "", "NGFR", "", "ERBB3", "CD44", "CD9", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "CAV1", "", "", "SLC45A3", "", "", "", "CD82", "", "", "PLXND1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "") 

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_1, links_pos_1,
  links_pos_1, links_chromosomes_2, links_pos_2, links_pos_2,
  maxRadius = 0.55, width = "0.01em", color = 'brown', labelSize = "0.75em", labels = links_labels_12)

links_chromosomes_3 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should start
links_chromosomes_4 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should end

# Now we start at the ligands
links_pos_3 = c(3, 3, 3, 7.36, 3, 11.72, 16.08, 3, 16.08, 3, 16.08, 7.36, 3, 16.08, 20.44, 24.80, 29.16, 33.52, 29.16, 29.16, 33.52, 24.80, 29.16, 37.89, 42.25, 46.61, 50.97, 46.61, 37.89, 50.97, 55.33, 7.36, 46.61, 37.89, 50.97, 20.44, 42.25, 59.69, 3, 37.89, 42.25, 46.61, 59.69, 3, 50.97, 46.61, 59.69, 3, 37.89, 50.97, 55.33, 7.36, 46.61, 59.69, 3, 37.89, 50.97, 20.44, 42.25, 64.05, 11.72, 29.16, 68.41, 72.77, 16.08, 42.25, 77.13, 33.52, 81.49, 46.61, 11.72, 29.16, 85.85, 90.21, 50.97, 16.08, 94.57, 81.49, 46.61, 11.72, 29.16, 85.85, 72.77, 90.21, 50.97, 16.08, 94.57, 77.13, 33.52, 81.49, 46.61, 11.72, 29.16, 68.41, 85.85, 72.77, 90.21, 50.97, 16.08, 42.25, 94.57, 11.72, 98.93, 37.89, 42.25, 103.30, 11.72, 98.93, 50.97, 103.30, 11.72, 98.93, 37.89, 50.97, 103.30, 55.33, 7.36, 11.72, 98.93, 37.89, 50.97, 20.44, 42.25, 107.66, 42.25, 107.66, 107.66, 107.66, 42.25, 11.72, 59.69, 107.66, 29.16, 112.02, 116.38, 103.30, 81.49, 46.61, 11.72, 59.69, 107.66, 29.16, 112.02, 50.97, 103.30, 81.49, 46.61, 11.72, 59.69, 107.66, 29.16, 112.02, 50.97, 120.74, 116.38, 103.30, 55.33, 81.49, 46.61, 11.72, 59.69, 107.66, 29.16, 112.02, 50.97, 20.44, 120.74, 64.05, 11.72, 107.66, 29.16, 103.30, 81.49, 11.72, 107.66, 29.16, 103.30, 81.49, 125.10, 11.72, 107.66, 29.16, 103.30, 81.49, 125.10, 11.72, 107.66, 29.16, 11.72, 29.16, 42.25, 116.38, 103.30, 81.49, 46.61, 11.72, 29.16, 103.30, 81.49, 46.61, 11.72, 29.16, 116.38, 103.30, 81.49, 46.61, 20.77, 29.16, 42.25, 7.36, 11.72, 129.46, 42.25, 77.13, 11.72, 129.46, 85.85, 133.82, 11.72, 129.46, 85.85, 133.82, 77.13, 11.72, 129.46, 85.85, 42.25, 133.82, 11.72, 77.13, 116.38, 11.72, 11.72, 77.13, 116.38, 11.72, 138.18, 142.54, 138.18, 146.90, 151.26, 155.62, 138.18, 138.18, 146.90, 146.90, 159.98, 151.26, 155.62, 164.34, 146.90, 151.26, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 142.54, 173.07, 138.18, 146.90, 177.43, 151.26, 155.62, 181.79, 186.15, 190.51, 138.18, 173.07, 177.43, 138.18, 146.90, 194.87, 173.07, 177.43, 146.90, 159.98, 151.26, 155.62, 164.34, 194.87, 199.23, 173.07, 203.59, 146.90, 151.26, 186.15, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 177.43, 207.95, 177.43, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 142.54, 151.26, 155.62, 138.18, 151.26, 155.62, 164.34, 203.59, 151.26, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 138.18, 142.54, 173.07, 212.31, 138.18, 146.90, 216.67, 221.03, 177.43, 151.26, 155.62, 181.79, 186.15, 190.51, 225.39, 229.75, 138.18, 221.03, 138.18, 173.07, 177.43, 212.31, 138.18, 146.90, 234.11, 221.03, 194.87, 173.07, 177.43, 212.31, 146.90, 234.11, 216.67, 238.48, 221.03, 242.84, 247.20, 251.56, 225.39, 159.98, 151.26, 155.62, 234.11, 221.03, 164.34, 164.34, 194.87, 199.23, 173.07, 164.34, 203.59, 225.39, 229.75, 146.90, 151.26, 186.15, 251.56, 234.11, 238.48, 221.03, 229.75, 212.31, 151.26, 155.62, 255.92, 234.11, 238.48, 260.28, 221.03, 168.70, 247.20, 264.64, 251.56, 229.75, 269, 151.26, 155.62, 238.48, 260.28, 221.03, 168.70, 247.20, 264.64, 251.56, 229.75, 269, 151.26, 155.62, 177.43, 207.95, 177.43, 229.75, 238.48, 221.03, 168.70, 247.20, 251.56, 151.26, 155.62, 238.48, 168.70, 251.56, 229.75, 151.26, 155.62)
# Now we end at the receptors
links_pos_4 = c(3, 3, 3, 41.84, 3, 3, 3, 3, 3, 3, 3, 41.84, 3, 3, 7.32, 11.63, 3, 15.95, 3, 3, 15.95, 11.63, 3, 20.26, 20.26, 24.58, 24.58, 24.58, 28.89, 24.58, 41.84, 41.84, 24.58, 20.26, 24.58, 7.32, 20.26, 3, 3, 20.26, 20.26, 24.58, 3, 3, 24.58, 24.58, 3, 3, 28.89, 24.58, 41.84, 41.84, 24.58, 3, 3, 20.26, 24.58, 7.32, 20.26, 33.21, 3, 3, 20.26, 20.26, 3, 20.26, 15.95, 15.95, 24.58, 24.58, 3, 3, 24.58, 24.58, 24.58, 3, 24.58, 24.58, 24.58, 3, 3, 24.58, 28.89, 24.58, 24.58, 3, 24.58, 15.95, 15.95, 24.58, 24.58, 3, 3, 20.26, 24.58, 20.26, 24.58, 24.58, 3, 20.26, 24.58, 3, 3, 20.26, 20.26, 24.58, 3, 3, 24.58, 24.58, 3, 3, 28.89, 24.58, 24.58, 41.84, 41.84, 3, 3, 20.26, 24.58, 7.32, 20.26, 3, 20.26, 3, 3, 3, 20.26, 3, 3, 3, 3, 3, 15.95, 24.58, 24.58, 24.58, 3, 3, 3, 3, 3, 24.58, 24.58, 24.58, 24.58, 3, 3, 3, 3, 3, 24.58, 37.53, 15.95, 24.58, 41.84, 24.58, 24.58, 3, 3, 3, 3, 3, 24.58, 7.32, 37.53, 33.21, 3, 3, 3, 24.58, 24.58, 3, 3, 3, 24.58, 24.58, 37.53, 3, 3, 3, 24.58, 24.58, 37.53, 3, 3, 3, 3, 3, 20.26, 15.95, 24.58, 24.58, 24.58, 3, 3, 24.58, 24.58, 24.58, 3, 3, 15.95, 24.58, 24.58, 24.58, 3, 3, 20.26, 41.84, 3, 3, 20.26, 15.95, 3, 3, 24.58, 24.58, 3, 3, 24.58, 24.58, 15.95, 3, 3, 24.58, 20.26, 24.58, 3, 15.95, 15.95, 3, 3, 15.95, 15.95, 3, 46.16, 46.16, 50.47, 50.47, 54.79, 54.79, 46.16, 50.47, 50.47, 50.47, 50.47, 54.79, 54.79, 46.16, 50.47, 54.79, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 46.16, 59.11, 50.47, 50.47, 59.11, 54.79, 54.79, 59.11, 59.11, 59.11, 46.16, 59.11, 59.11, 50.47, 50.47, 59.11, 59.11, 59.11, 50.47, 50.47, 54.79, 54.79, 46.16, 59.11, 59.11, 59.11, 63.42, 50.47, 54.79, 59.11, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 59.11, 59.11, 59.11, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 63.42, 54.79, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 41.84, 46.16, 46.16, 59.11, 67.74, 50.47, 50.47, 41.84, 41.84, 59.11, 54.79, 54.79, 59.11, 59.11, 59.11, 72.05, 76.37, 41.84, 41.84, 46.16, 59.11, 59.11, 67.74, 50.47, 50.47, 41.84, 41.84, 59.11, 59.11, 59.11, 67.74, 50.47, 41.84, 41.84, 41.84, 41.84, 80.68, 85, 85, 72.05, 50.47, 54.79, 54.79, 41.84, 41.84, 80.68, 46.16, 59.11, 59.11, 59.11, 85, 63.42, 72.05, 76.37, 50.47, 54.79, 59.11, 85, 41.84, 41.84, 41.84, 76.37, 67.74, 54.79, 54.79, 41.84, 41.84, 41.84, 41.84, 41.84, 46.16, 85, 85, 85, 76.37, 76.37, 54.79, 54.79, 41.84, 41.84, 41.84, 46.16, 85, 85, 85, 76.37, 76.37, 54.79, 54.79, 59.11, 59.11, 59.11, 76.37, 41.84, 41.84, 46.16, 85, 85, 54.79, 54.79, 41.84, 46.16, 85, 76.37, 54.79, 54.79)
# These links are for the receptors
links_labels_34 = c('FGFR1', "", "", "APP", "", "", "", "", "", "", "", "", "", "", "ANK3", "ACKR3", "", "TGFBR3", "", "", "", "", "", "PTPRS", "", "ITGB5", "", "", "PTPRD", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "PLSCR4", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "CDON", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "CTGF", "", "TFPI", "", "TIMP2", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "EFNA5", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "LTBP3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "APP", "", "", "", "SEMA6D", "", "", "", "", "", "", "", "", "", "", "NTN4", "SEMA3E", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "BTC", "HBEGF", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_3, links_pos_3,
  links_pos_3, links_chromosomes_4, links_pos_4, links_pos_4,
  maxRadius = 0.55, width = "0.01em", color = 'brown', labelSize = "0.75em", labels = links_labels_34, labelPadding = 3)

links_chromosomes_5 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should start
links_chromosomes_6 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should end
links_pos_5 = c(138.18, 142.54, 138.18, 146.90, 151.26, 155.62, 138.18, 138.18, 146.90, 146.90, 159.98, 151.26, 155.62, 164.34, 146.90, 151.26, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 142.54, 173.07, 138.18, 146.90, 177.43, 151.26, 155.62, 181.79, 186.15, 190.51, 138.18, 173.07, 177.43, 138.18, 146.90, 194.87, 173.07, 177.43, 146.90, 159.98, 151.26, 155.62, 164.34, 194.87, 199.23, 173.07, 203.59, 146.90, 151.26, 186.15, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 177.43, 207.95, 177.43, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 142.54, 151.26, 155.62, 138.18, 151.26, 155.62, 164.34, 203.59, 151.26, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 168.70, 151.26, 155.62, 138.18, 138.18, 142.54, 173.07, 212.31, 138.18, 146.90, 216.67, 221.03, 177.43, 151.26, 155.62, 181.79, 186.15, 190.51, 225.39, 229.75, 138.18, 221.03, 138.18, 173.07, 177.43, 212.31, 138.18, 146.90, 234.11, 221.03, 194.87, 173.07, 177.43, 212.31, 146.90, 234.11, 216.67, 238.48, 221.03, 242.84, 247.20, 251.56, 225.39, 159.98, 151.26, 155.62, 234.11, 221.03, 164.34, 164.34, 194.87, 199.23, 173.07, 164.34, 203.59, 225.39, 229.75, 146.90, 151.26, 186.15, 251.56, 234.11, 238.48, 221.03, 229.75, 212.31, 151.26, 155.62, 255.92, 234.11, 238.48, 260.28, 221.03, 168.70, 247.20, 264.64, 251.56, 229.75, 269, 151.26, 155.62, 238.48, 260.28, 221.03, 168.70, 247.20, 264.64, 251.56, 229.75, 269, 151.26, 155.62, 177.43, 207.95, 177.43, 229.75, 238.48, 221.03, 168.70, 247.20, 251.56, 151.26, 155.62, 238.48, 168.70, 251.56, 229.75, 151.26, 155.62)
links_pos_6 = c(46.16, 46.16, 50.47, 50.47, 54.79, 54.79, 46.16, 50.47, 50.47, 50.47, 50.47, 54.79, 54.79, 46.16, 50.47, 54.79, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 46.16, 59.11, 50.47, 50.47, 59.11, 54.79, 54.79, 59.11, 59.11, 59.11, 46.16, 59.11, 59.11, 50.47, 50.47, 59.11, 59.11, 59.11, 50.47, 50.47, 54.79, 54.79, 46.16, 59.11, 59.11, 59.11, 63.42, 50.47, 54.79, 59.11, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 59.11, 59.11, 59.11, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 63.42, 54.79, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 46.16, 54.79, 54.79, 41.84, 46.16, 46.16, 59.11, 67.74, 50.47, 50.47, 41.84, 41.84, 59.11, 54.79, 54.79, 59.11, 59.11, 59.11, 72.05, 76.37, 41.84, 41.84, 46.16, 59.11, 59.11, 67.74, 50.47, 50.47, 41.84, 41.84, 59.11, 59.11, 59.11, 67.74, 50.47, 41.84, 41.84, 41.84, 41.84, 80.68, 85, 85, 72.05, 50.47, 54.79, 54.79, 41.84, 41.84, 80.68, 46.16, 59.11, 59.11, 59.11, 85, 63.42, 72.05, 76.37, 50.47, 54.79, 59.11, 85, 41.84, 41.84, 41.84, 76.37, 67.74, 54.79, 54.79, 41.84, 41.84, 41.84, 41.84, 41.84, 46.16, 85, 85, 85, 76.37, 76.37, 54.79, 54.79, 41.84, 41.84, 41.84, 46.16, 85, 85, 85, 76.37, 76.37, 54.79, 54.79, 59.11, 59.11, 59.11, 76.37, 41.84, 41.84, 46.16, 85, 85, 54.79, 54.79, 41.84, 46.16, 85, 76.37, 54.79, 54.79)
links_labels_56 = c("", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", '', "", "", "", "", "", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_5, links_pos_5,
  links_pos_5, links_chromosomes_6, links_pos_6, links_pos_6,
  maxRadius = 0.55, width = "0.01em", color = 'cyan', labelSize = "0.5em", labels = links_labels_56)

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra = c('Ligand') # Chromosomes on which the arcs should be displayed
arcs_begin_ra = c(0)
arcs_end_ra = c(43)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra, arcs_begin_ra, arcs_end_ra, minRadius = 1.3, maxRadius = 1.35, colors = 'forestgreen')

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra2 = c('Ligand') # Chromosomes on which the arcs should be displayed
arcs_begin_ra2 = c(39)
arcs_end_ra2 = c(88)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra2, arcs_begin_ra2, arcs_end_ra2, minRadius = 1.35, maxRadius = 1.4, colors = 'deeppink3')

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_hdrg = c('Receptor') # Chromosomes on which the arcs should be displayed
arcs_begin_hdrg = c(0)
arcs_end_hdrg = c(138)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_hdrg, arcs_begin_hdrg, arcs_end_hdrg, minRadius = 1.35, maxRadius = 1.4, colors = 'blue')

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_hdrg2 = c('Receptor') # Chromosomes on which the arcs should be displayed
arcs_begin_hdrg2 = c(139)
arcs_end_hdrg2 = c(272)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_hdrg2, arcs_begin_hdrg2, arcs_end_hdrg2, minRadius = 1.3, maxRadius = 1.35, colors = 'yellow')

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Mouse DRG to Human RA - Specific Neurons', x = -0.6, y = -1.30, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Receptor', x #= 0.75, y = -1, size = "1em",
##  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Ligand', x = #-0.95, y = -1, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes, arcs_begin, arcs_end, minRadius = 0.85, maxRadius = 0.99, colors = 'forestgreen') 

BioCircos(tracklist, genome = myGenome, genomeFillColor = c('white', 'white'), chrPad = 0.05, displayGenomeBorder = FALSE, yChr =  FALSE,
  genomeTicksDisplay = FALSE,  genomeLabelTextSize = 0, genomeLabelDy = 0, width = 650, height = 650)
```

Below we make a legend for the ligand and receptor tracts.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('mDRG Neuronal Cell Type Ligands (236 connections)', 'mDRG Neuronal Cell Type Receptors (208 connections)', 'Synovial Fibroblast Subtype Ligands (208 connections)', 'Synovial Fibroblast Subtype Receptors (236 connections)'), bty='n', lwd = 8, cex = 1.2, col = c("forestgreen", "deeppink3", 'blue', 'yellow'), lty = c(1, 1, 1, 1), pch = c(NA, NA, NA, NA))
 #   col = c('forestgreen', 'deeppink3', 'blue', 'yellow'))
#mtext("mDRG Neuronal Cell Types", at=0.19, cex=1.1)
```

Below we make a legend for the fibroblast subtypes.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (7)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (7)', 'DKK3+ sublining fibroblasts (SC-F3) (8)', 'CD55+ lining fibroblasts (SC-F4) (17)'), pch=15, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes", at=0.24, cex=2)
```

Below we split the legend in half - one for the ligands and one for the recpetors.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (4)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (2)', 'DKK3+ sublining fibroblasts (SC-F3) (5)', 'CD55+ lining fibroblasts (SC-F4) (7)'), pch=15, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes for Pain-associated Receptors", at=0.24, cex=1.6)
```

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (3)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (5)', 'DKK3+ sublining fibroblasts (SC-F3) (3)', 'CD55+ lining fibroblasts (SC-F4) (11)'), pch=15, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes For Pain-associated Ligands", at=0.24, cex=1.6)
```

Below we plot the legend for the cell types.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Cerebellum Neurons (6)', 'Cholinergic and Monoaminergic Neurons (9)', 'Di- and Mesencphalon Excitatory Neurons (13)', 'Di- and Mesencphalon Inhibitory Neurons (17)', 'Enteric Neurons (26)', 'Hindbrain Neurons (20)', 'Peripheral Sensory Neurofilament Neurons (20)', 'Peripheral Sensory Non-peptidergic Neurons (20)', 'Peripheral Sensory Peptidergic Neurons (18)', 'Sympathetic Cholinergic Neurons (12)', 'Sympathetic Noradrenergic Neurons (9)', 'Dentate Gyrus Granule Neurons (9)', 'Olfactory Inhibitory Neurons (3)', 'Spinal Cord Excitatory Neurons (1)', 'Spinal Cord Inhibitory Neurons (4)', 'Peptidergic Neurons (1)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('antiquewhite4', 'black', 'blue', 'blueviolet', 'brown', 'cadetblue1', 'gold', 'darkgrey', 'magenta', 'chartreuse3', 'cyan', 'orange', 'coral', 'darkgreen', 'firebrick', 'purple'))
mtext("Mouse Nervous System Neuronal Cell Types", at=0.19, cex=1.1)
```

Below we split the legend in half - one for the ligands and one for the recpetors.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Cerebellum Neurons (2)', 'Cholinergic and Monoaminergic Neurons (4)', 'Di- and Mesencphalon Excitatory Neurons (7)', 'Di- and Mesencphalon Inhibitory Neurons (10)', 'Enteric Neurons (15)', 'Hindbrain Neurons (9)', 'Peripheral Sensory Neurofilament Neurons (13)', 'Peripheral Sensory Non-peptidergic Neurons (7)', 'Peripheral Sensory Peptidergic Neurons (7)', 'Sympathetic Cholinergic Neurons (6)', 'Sympathetic Noradrenergic Neurons (3)', 'Dentate Gyrus Granule Neurons (4)', 'Olfactory Inhibitory Neurons (3)', 'Spinal Cord Inhibitory Neurons (1)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('antiquewhite4', 'black', 'blue', 'blueviolet', 'brown', 'cadetblue1', 'gold', 'darkgrey', 'magenta', 'chartreuse3', 'cyan', 'orange', 'coral', 'firebrick'
            ))
mtext("Neuronal Cell Types For Ligands", at=0.19, cex=1.1)
```

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Cerebellum Neurons (5)', 'Cholinergic and Monoaminergic Neurons (5)', 'Di- and Mesencphalon Excitatory Neurons (6)', 'Di- and Mesencphalon Inhibitory Neurons (7)', 'Enteric Neurons (11)', 'Hindbrain Neurons (11)', 'Peripheral Sensory Neurofilament Neurons (7)', 'Peripheral Sensory Non-peptidergic Neurons (13)', 'Peripheral Sensory Peptidergic Neurons (11)', 'Sympathetic Cholinergic Neurons (7)', 'Sympathetic Noradrenergic Neurons (6)', 'Dentate Gyrus Granule Neurons (5)', 'Olfactory Inhibitory Neurons (1)', 'Spinal Cord Excitatory Neurons (1)', 'Spinal Cord Inhibitory Neurons (3)', 'Peptidergic Neurons (1)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('antiquewhite4', 'black', 'blue', 'blueviolet', 'brown', 'cadetblue1', 'gold', 'darkgrey', 'magenta', 'chartreuse3', 'cyan', 'orange', 'coral', 'darkgreen', 'firebrick', 'purple'))
mtext("Neuronal Cell Types For Receptors", at=0.19, cex=1.1)
```

Below we create plots for the number of connections for mDRG to RA and RA to mDRG.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Mouse Nervous System to Synovial Fibroblast (236 connections)', 'Synovial Fibroblast to Mouse Nervous System (208 connections)'), bty='n', lwd = 8, cex = 1.2, col = c('brown', 'cyan'), lty = c(1, 1, 1, 1), pch = c(NA, NA, NA, NA))
 #   col = c('cyan', 'brown'))
#mtext("Mouse Nervous System Neuronal Cell Types", at=0.19, cex=1.1)
```

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (48 Connections)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (48 Connections)', 'DKK3+ sublining fibroblasts (SC-F3) (62 Connections)', 'CD55+ lining fibroblasts (SC-F4) (78 Connections)'), pch=15, pt.cex=3, cex=1.2, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Connections Synovial Fibroblast to Mouse Nervous System", at=0.34, cex= 1.3)
```

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (30 Connections)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (48 Connections)', 'DKK3+ sublining fibroblasts (SC-F3) (24 Connections)', 'CD55+ lining fibroblasts (SC-F4) (106 Connections)'), pch=15, pt.cex=3, cex=1.2, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Connections Mouse Nervous System to Synovial Fibroblast", at=0.34, cex=1.3)
```

Below we create the connections legend for the cell types for mouse nervous system to RA.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Cerebellum Neurons (5 Connections)', 'Cholinergic and Monoaminergic Neurons (10 Connections)', 'Di- and Mesencphalon Excitatory Neurons (14 Connections)', 'Di- and Mesencphalon Inhibitory Neurons (22 Connections)', 'Enteric Neurons (42 Connections)', 'Hindbrain Neurons (22 Connections)', 'Peripheral Sensory Neurofilament Neurons (38)', 'Peripheral Sensory Non-peptidergic Neurons (21 Connections)', 'Peripheral Sensory Peptidergic Neurons (21 Connections)', 'Sympathetic Cholinergic Neurons (18 Connections)', 'Sympathetic Noradrenergic Neurons (8 Connections)', 'Dentate Gyrus Granule Neurons (8 Connections)', 'Olfactory Inhibitory Neurons (6 Connections)', 'Spinal Cord Inhibitory Neurons (1 Connection)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('antiquewhite4', 'black', 'blue', 'blueviolet', 'brown', 'cadetblue1', 'gold', 'darkgrey', 'magenta', 'chartreuse3', 'cyan', 'orange', 'coral', 'firebrick'))
mtext("Neuronal Cell Type Connections Mouse Nervous System to Synovial Fibroblast", at=0.4, cex=1.1)
```

Below we create the connections legend for the cell types for RA to mouse nervous system.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Cerebellum Neurons (18 Connections)', 'Cholinergic and Monoaminergic Neurons (12 Connections)', 'Di- and Mesencphalon Excitatory Neurons (17 Connections)', 'Di- and Mesencphalon Inhibitory Neurons (12 Connections)', 'Enteric Neurons (19 Connections)', 'Hindbrain Neurons (26 Connections)', 'Peripheral Sensory Neurofilament Neurons (12 Connections)', 'Peripheral Sensory Non-peptidergic Neurons (21 Connections)', 'Peripheral Sensory Peptidergic Neurons (20 Connections)', 'Sympathetic Cholinergic Neurons (12 Connections)', 'Sympathetic Noradrenergic Neurons (15 Connections)', 'Dentate Gyrus Granule Neurons (8 Connections)', 'Olfactory Inhibitory Neurons (2 Connections)', 'Spinal Cord Excitatory Neurons (2 Connections)', 'Spinal Cord Inhibitory Neurons (5 Connections)', 'Peptidergic Neurons (1 Connetion)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('antiquewhite4', 'black', 'blue', 'blueviolet', 'brown', 'cadetblue1', 'gold', 'darkgrey', 'magenta', 'chartreuse3', 'cyan', 'orange', 'coral', 'darkgreen', 'firebrick', 'purple'))
mtext("Neuronal Cell Type Connections Synovial Fibroblast to Mouse Nervous System", at=0.4, cex=1.1)
```


### All Cell Type Plots

For all cell types, we subset to only neurons. 

```{r}
mdrg_to_ra_all_types_final <- filter(mdrg_to_ra_all_types, !grepl('glia|Schwann', ligand_type))
ra_to_mdrg_all_types_final <- filter(ra_to_mdrg_all_types, !grepl('glia|Schwann', receptor_type))
all_types_data_final <- rbind(mdrg_to_ra_all_types_final, ra_to_mdrg_all_types_final)
```

Below we repeat the analysis for finding the statistics of all cell types using the subset interactome.

```{r}
all_types_data_f1 <- all_types_data_final[(all_types_data_final$ligand_type == 'SC-F1') | (all_types_data_final$receptor_type == 'SC-F1'),]
print(nrow(all_types_data_f1))
all_types_data_f2 <- all_types_data_final[(all_types_data_final$ligand_type == 'SC-F2') | (all_types_data_final$receptor_type == 'SC-F2'),]
print(nrow(all_types_data_f2))
all_types_data_f3 <- all_types_data_final[(all_types_data_final$ligand_type == 'SC-F3') | (all_types_data_final$receptor_type == 'SC-F3'),]
print(nrow(all_types_data_f3))
all_types_data_f4 <- all_types_data_final[(all_types_data_final$ligand_type == 'SC-F4') | (all_types_data_final$receptor_type == 'SC-F4'),]
print(nrow(all_types_data_f4))
```

Below we get statistics for a multitude of things for all types.

```{r}
print('Order: Total Unique Ligands')
print('Total Unique Receptors')
print('mDRG Ligand to Synovial Receptor Connections')
print('mDRG Ligands')
print('mDRG Receptors')
print('Synovial Ligand to mDRG Receptor Connections')
print('Synovial Ligands')
print('Synovial Receptors')
print('-------------SC-F1------------------')
length(unique(all_types_data_f1$ligand))
length(unique(all_types_data_f1$receptor))
all_types_data_f1_mdrg = all_types_data_f1[all_types_data_f1$location == 'mDRG',]
nrow(all_types_data_f1_mdrg)
length(unique(all_types_data_f1_mdrg$ligand))
length(unique(all_types_data_f1_mdrg$receptor))
all_types_data_f1_ra = all_types_data_f1[all_types_data_f1$location == 'RA sc',]
nrow(all_types_data_f1_ra)
length(unique(all_types_data_f1_ra$ligand))
length(unique(all_types_data_f1_ra$receptor))

print('-------------SC-F2------------------')
length(unique(all_types_data_f2$ligand))
length(unique(all_types_data_f2$receptor))
all_types_data_f2_mdrg = all_types_data_f2[all_types_data_f2$location == 'mDRG',]
nrow(all_types_data_f2_mdrg)
length(unique(all_types_data_f2_mdrg$ligand))
length(unique(all_types_data_f2_mdrg$receptor))
all_types_data_f2_ra = all_types_data_f2[all_types_data_f2$location == 'RA sc',]
nrow(all_types_data_f2_ra)
length(unique(all_types_data_f2_ra$ligand))
length(unique(all_types_data_f2_ra$receptor))

print('-------------SC-F3------------------')
length(unique(all_types_data_f3$ligand))
length(unique(all_types_data_f3$receptor))
all_types_data_f3_mdrg = all_types_data_f3[all_types_data_f3$location == 'mDRG',]
nrow(all_types_data_f3_mdrg)
length(unique(all_types_data_f3_mdrg$ligand))
length(unique(all_types_data_f3_mdrg$receptor))
all_types_data_f3_ra = all_types_data_f3[all_types_data_f3$location == 'RA sc',]
nrow(all_types_data_f3_ra)
length(unique(all_types_data_f3_ra$ligand))
length(unique(all_types_data_f3_ra$receptor))

print('-------------SC-F4------------------')
length(unique(all_types_data_f4$ligand))
length(unique(all_types_data_f4$receptor))
all_types_data_f4_mdrg = all_types_data_f4[all_types_data_f4$location == 'mDRG',]
nrow(all_types_data_f4_mdrg)
length(unique(all_types_data_f4_mdrg$ligand))
length(unique(all_types_data_f4_mdrg$receptor))
all_types_data_f4_ra = all_types_data_f4[all_types_data_f4$location == 'RA sc',]
nrow(all_types_data_f4_ra)
length(unique(all_types_data_f4_ra$ligand))
length(unique(all_types_data_f4_ra$receptor))
```

Below we specify by the three specific neuron cell types for all types before splitting by fibroblast subtype.

```{r}
print('------------Peripheral sensory neurofilament neurons--------------')
all_types_data_final_peripheral_sensory_neurofilament_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Peripheral sensory neurofilament neurons') | (all_types_data_final$receptor_type == 'Peripheral sensory neurofilament neurons'),]

print('------------Peripheral sensory non-peptidergic neurons--------------')
all_types_data_final_peripheral_sensory_nonpeptidergic_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Peripheral sensory non-peptidergic neurons') | (all_types_data_final$receptor_type == 'Peripheral sensory non-peptidergic neurons'),]

print('------------Peripheral sensory peptidergic neurons--------------')
all_types_data_final_peripheral_sensory_peptidergic_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Peripheral sensory peptidergic neurons') | (all_types_data_final$receptor_type == 'Peripheral sensory peptidergic neurons'),]

print('------------Cerebellum neurons--------------')
all_types_data_final_cerebellum_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Cerebellum neurons') | (all_types_data_final$receptor_type == 'Cerebellum neurons'),]

print('------------Cholinergic and monoaminergic neurons--------------')
all_types_data_final_cholinergic_and_monoaminergic_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Cholinergic and monoaminergic neurons') | (all_types_data_final$receptor_type == 'Cholinergic and monoaminergic neurons'),]

print('------------Dentate gyrus granule neurons--------------')
all_types_data_final_dentate_gyrus_granule_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Dentate gyrus granule neurons') | (all_types_data_final$receptor_type == 'Dentate gyrus granule neurons'),]

print('------------Di- and mesencephalon excitatory neurons--------------')
all_types_data_final_di_and_mesencephalon_excitatory_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Di- and mesencephalon excitatory neurons') | (all_types_data_final$receptor_type == 'Di- and mesencephalon excitatory neurons'),]

print('------------Di- and mesencephalon inhibitory neurons--------------')
all_types_data_final_di_and_mesencephalon_inhibitory_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Di- and mesencephalon inhibitory neurons') | (all_types_data_final$receptor_type == 'Di- and mesencephalon inhibitory neurons'),]

print('------------Enteric neurons--------------')
all_types_data_final_enteric_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Enteric neurons') | (all_types_data_final$receptor_type == 'Enteric neurons'),]

print('------------Hindbrain neurons--------------')
all_types_data_final_hindbrain_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Hindbrain neurons') | (all_types_data_final$receptor_type == 'Hindbrain neurons'),]

print('------------Olfactory inhibitory neurons--------------')
all_types_data_final_olfactory_inhibitory_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Olfactory inhibitory neurons') | (all_types_data_final$receptor_type == 'Olfactory inhibitory neurons'),]

print('------------Peptidergic neurons--------------')
all_types_data_final_peptidergic_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Peptidergic neurons') | (all_types_data_final$receptor_type == 'Peptidergic neurons'),]

print('------------Spinal cord inhibitory neurons--------------')
all_types_data_final_spinal_cord_inhibitory_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Spinal cord inhibitory neurons') | (all_types_data_final$receptor_type == 'Spinal cord inhibitory neurons'),]

print('------------Spinal cord excitatory neurons--------------')
all_types_data_final_spinal_cord_excitatory_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Spinal cord excitatory neurons') | (all_types_data_final$receptor_type == 'Spinal cord excitatory neurons'),]

print('------------Sympathetic cholinergic neurons--------------')
all_types_data_final_sympathetic_cholinergic_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Sympathetic cholinergic neurons') | (all_types_data_final$receptor_type == 'Sympathetic cholinergic neurons'),]

print('------------Sympathetic noradrenergic neurons--------------')
all_types_data_final_sympathetic_nonadrenergic_neurons <- all_types_data_final[(all_types_data_final$ligand_type == 'Sympathetic noradrenergic neurons') | (all_types_data_final$receptor_type == 'Sympathetic noradrenergic neurons'),]
```

Below we look at some statistics for the unique cell types.

```{r}
print('Order: Total Unique Ligands')
print('Total Unique Receptors')
print('mDRG Ligand to Synovial Receptor Connections')
print('mDRG Ligands')
print('mDRG Receptors')
print('Synovial Ligand to mDRG Receptor Connections')
print('Synovial Ligands')
print('Synovial Receptors')

print('------------Cerebellum neurons--------------')
length(unique(all_types_data_final_cerebellum_neurons$ligand))
length(unique(all_types_data_final_cerebellum_neurons$receptor))
all_types_data_final_cerebellum_neurons_mdrg = all_types_data_final_cerebellum_neurons[all_types_data_final_cerebellum_neurons$location == 'mDRG',]
nrow(all_types_data_final_cerebellum_neurons_mdrg)
length(unique(all_types_data_final_cerebellum_neurons_mdrg$ligand))
length(unique(all_types_data_final_cerebellum_neurons_mdrg$receptor))
all_types_data_final_cerebellum_neurons_ra = all_types_data_final_cerebellum_neurons[all_types_data_final_cerebellum_neurons$location == 'RA sc',]
nrow(all_types_data_final_cerebellum_neurons_ra)
length(unique(all_types_data_final_cerebellum_neurons_ra$ligand))
length(unique(all_types_data_final_cerebellum_neurons_ra$receptor))

print('------------Cholinergic and monoaminergic neurons--------------')
length(unique(all_types_data_final_cholinergic_and_monoaminergic_neurons$ligand))
length(unique(all_types_data_final_cholinergic_and_monoaminergic_neurons$receptor))
all_types_data_final_cholinergic_and_monoaminergic_neurons_mdrg = all_types_data_final_cholinergic_and_monoaminergic_neurons[all_types_data_final_cholinergic_and_monoaminergic_neurons$location == 'mDRG',]
nrow(all_types_data_final_cholinergic_and_monoaminergic_neurons_mdrg)
length(unique(all_types_data_final_cholinergic_and_monoaminergic_neurons_mdrg$ligand))
length(unique(all_types_data_final_cholinergic_and_monoaminergic_neurons_mdrg$receptor))
all_types_data_final_cholinergic_and_monoaminergic_neurons_ra = all_types_data_final_cholinergic_and_monoaminergic_neurons[all_types_data_final_cholinergic_and_monoaminergic_neurons$location == 'RA sc',]
nrow(all_types_data_final_cholinergic_and_monoaminergic_neurons_ra)
length(unique(all_types_data_final_cholinergic_and_monoaminergic_neurons_ra$ligand))
length(unique(all_types_data_final_cholinergic_and_monoaminergic_neurons_ra$receptor))
print('------------Dentate gyrus granule neurons--------------')
length(unique(all_types_data_final_dentate_gyrus_granule_neurons$ligand))
length(unique(all_types_data_final_dentate_gyrus_granule_neurons$receptor))
all_types_data_final_dentate_gyrus_granule_neurons_mdrg = all_types_data_final_dentate_gyrus_granule_neurons[all_types_data_final_dentate_gyrus_granule_neurons$location == 'mDRG',]
nrow(all_types_data_final_dentate_gyrus_granule_neurons_mdrg)
length(unique(all_types_data_final_dentate_gyrus_granule_neurons_mdrg$ligand))
length(unique(all_types_data_final_dentate_gyrus_granule_neurons_mdrg$receptor))
all_types_data_final_dentate_gyrus_granule_neurons_ra = all_types_data_final_dentate_gyrus_granule_neurons[all_types_data_final_dentate_gyrus_granule_neurons$location == 'RA sc',]
nrow(all_types_data_final_dentate_gyrus_granule_neurons_ra)
length(unique(all_types_data_final_dentate_gyrus_granule_neurons_ra$ligand))
length(unique(all_types_data_final_dentate_gyrus_granule_neurons_ra$receptor))
print('------------Di- and mesencephalon excitatory neurons--------------')
length(unique(all_types_data_final_di_and_mesencephalon_excitatory_neurons$ligand))
length(unique(all_types_data_final_di_and_mesencephalon_excitatory_neurons$receptor))
all_types_data_final_di_and_mesencephalon_excitatory_neurons_mdrg = all_types_data_final_di_and_mesencephalon_excitatory_neurons[all_types_data_final_di_and_mesencephalon_excitatory_neurons$location == 'mDRG',]
nrow(all_types_data_final_di_and_mesencephalon_excitatory_neurons_mdrg)
length(unique(all_types_data_final_di_and_mesencephalon_excitatory_neurons_mdrg$ligand))
length(unique(all_types_data_final_di_and_mesencephalon_excitatory_neurons_mdrg$receptor))
all_types_data_final_di_and_mesencephalon_excitatory_neurons_ra = all_types_data_final_di_and_mesencephalon_excitatory_neurons[all_types_data_final_di_and_mesencephalon_excitatory_neurons$location == 'RA sc',]
nrow(all_types_data_final_di_and_mesencephalon_excitatory_neurons_ra)
length(unique(all_types_data_final_di_and_mesencephalon_excitatory_neurons_ra$ligand))
length(unique(all_types_data_final_di_and_mesencephalon_excitatory_neurons_ra$receptor))
print('------------Di- and mesencephalon inhibitory neurons--------------')
length(unique(all_types_data_final_di_and_mesencephalon_inhibitory_neurons$ligand))
length(unique(all_types_data_final_di_and_mesencephalon_inhibitory_neurons$receptor))
all_types_data_final_di_and_mesencephalon_inhibitory_neurons_mdrg = all_types_data_final_di_and_mesencephalon_inhibitory_neurons[all_types_data_final_di_and_mesencephalon_inhibitory_neurons$location == 'mDRG',]
nrow(all_types_data_final_di_and_mesencephalon_inhibitory_neurons_mdrg)
length(unique(all_types_data_final_di_and_mesencephalon_inhibitory_neurons_mdrg$ligand))
length(unique(all_types_data_final_di_and_mesencephalon_inhibitory_neurons_mdrg$receptor))
all_types_data_final_di_and_mesencephalon_inhibitory_neurons_ra = all_types_data_final_di_and_mesencephalon_inhibitory_neurons[all_types_data_final_di_and_mesencephalon_inhibitory_neurons$location == 'RA sc',]
nrow(all_types_data_final_di_and_mesencephalon_inhibitory_neurons_ra)
length(unique(all_types_data_final_di_and_mesencephalon_inhibitory_neurons_ra$ligand))
length(unique(all_types_data_final_di_and_mesencephalon_inhibitory_neurons_ra$receptor))
print('------------Enteric neurons--------------')
length(unique(all_types_data_final_enteric_neurons$ligand))
length(unique(all_types_data_final_enteric_neurons$receptor))
all_types_data_final_enteric_neurons_mdrg = all_types_data_final_enteric_neurons[all_types_data_final_enteric_neurons$location == 'mDRG',]
nrow(all_types_data_final_enteric_neurons_mdrg)
length(unique(all_types_data_final_enteric_neurons_mdrg$ligand))
length(unique(all_types_data_final_enteric_neurons_mdrg$receptor))
all_types_data_final_enteric_neurons_ra = all_types_data_final_enteric_neurons[all_types_data_final_enteric_neurons$location == 'RA sc',]
nrow(all_types_data_final_enteric_neurons_ra)
length(unique(all_types_data_final_enteric_neurons_ra$ligand))
length(unique(all_types_data_final_enteric_neurons_ra$receptor))

print('------------Hindbrain neurons--------------')
length(unique(all_types_data_final_hindbrain_neurons$ligand))
length(unique(all_types_data_final_hindbrain_neurons$receptor))
all_types_data_final_hindbrain_neurons_mdrg = all_types_data_final_hindbrain_neurons[all_types_data_final_hindbrain_neurons$location == 'mDRG',]
nrow(all_types_data_final_hindbrain_neurons_mdrg)
length(unique(all_types_data_final_hindbrain_neurons_mdrg$ligand))
length(unique(all_types_data_final_hindbrain_neurons_mdrg$receptor))
all_types_data_final_hindbrain_neurons_ra = all_types_data_final_hindbrain_neurons[all_types_data_final_hindbrain_neurons$location == 'RA sc',]
nrow(all_types_data_final_hindbrain_neurons_ra)
length(unique(all_types_data_final_hindbrain_neurons_ra$ligand))
length(unique(all_types_data_final_hindbrain_neurons_ra$receptor))

print('------------Olfactory inhibitory neurons--------------')
length(unique(all_types_data_final_olfactory_inhibitory_neurons$ligand))
length(unique(all_types_data_final_olfactory_inhibitory_neurons$receptor))
all_types_data_final_olfactory_inhibitory_neurons_mdrg = all_types_data_final_olfactory_inhibitory_neurons[all_types_data_final_olfactory_inhibitory_neurons$location == 'mDRG',]
nrow(all_types_data_final_olfactory_inhibitory_neurons_mdrg)
length(unique(all_types_data_final_olfactory_inhibitory_neurons_mdrg$ligand))
length(unique(all_types_data_final_olfactory_inhibitory_neurons_mdrg$receptor))
all_types_data_final_olfactory_inhibitory_neurons_ra = all_types_data_final_olfactory_inhibitory_neurons[all_types_data_final_olfactory_inhibitory_neurons$location == 'RA sc',]
nrow(all_types_data_final_olfactory_inhibitory_neurons_ra)
length(unique(all_types_data_final_olfactory_inhibitory_neurons_ra$ligand))
length(unique(all_types_data_final_olfactory_inhibitory_neurons_ra$receptor))

print('------------Peptidergic neurons--------------')
length(unique(all_types_data_final_peptidergic_neurons$ligand))
length(unique(all_types_data_final_peptidergic_neurons$receptor))
all_types_data_final_peptidergic_neurons_mdrg = all_types_data_final_peptidergic_neurons[all_types_data_final_peptidergic_neurons$location == 'mDRG',]
nrow(all_types_data_final_peptidergic_neurons_mdrg)
length(unique(all_types_data_final_peptidergic_neurons_mdrg$ligand))
length(unique(all_types_data_final_peptidergic_neurons_mdrg$receptor))
all_types_data_final_peptidergic_neurons_ra = all_types_data_final_peptidergic_neurons[all_types_data_final_peptidergic_neurons$location == 'RA sc',]
nrow(all_types_data_final_peptidergic_neurons_ra)
length(unique(all_types_data_final_peptidergic_neurons_ra$ligand))
length(unique(all_types_data_final_peptidergic_neurons_ra$receptor))

print('------------Spinal cord inhibitory neurons--------------')
length(unique(all_types_data_final_spinal_cord_inhibitory_neurons$ligand))
length(unique(all_types_data_final_spinal_cord_inhibitory_neurons$receptor))
all_types_data_final_spinal_cord_inhibitory_neurons_mdrg = all_types_data_final_spinal_cord_inhibitory_neurons[all_types_data_final_spinal_cord_inhibitory_neurons$location == 'mDRG',]
nrow(all_types_data_final_spinal_cord_inhibitory_neurons_mdrg)
length(unique(all_types_data_final_spinal_cord_inhibitory_neurons_mdrg$ligand))
length(unique(all_types_data_final_spinal_cord_inhibitory_neurons_mdrg$receptor))
all_types_data_final_spinal_cord_inhibitory_neurons_ra = all_types_data_final_spinal_cord_inhibitory_neurons[all_types_data_final_spinal_cord_inhibitory_neurons$location == 'RA sc',]
nrow(all_types_data_final_spinal_cord_inhibitory_neurons_ra)
length(unique(all_types_data_final_spinal_cord_inhibitory_neurons_ra$ligand))
length(unique(all_types_data_final_spinal_cord_inhibitory_neurons_ra$receptor))

print('------------Spinal cord excitatory neurons--------------')
length(unique(all_types_data_final_spinal_cord_excitatory_neurons$ligand))
length(unique(all_types_data_final_spinal_cord_excitatory_neurons$receptor))
all_types_data_final_spinal_cord_excitatory_neurons_mdrg = all_types_data_final_spinal_cord_excitatory_neurons[all_types_data_final_spinal_cord_excitatory_neurons$location == 'mDRG',]
nrow(all_types_data_final_spinal_cord_excitatory_neurons_mdrg)
length(unique(all_types_data_final_spinal_cord_excitatory_neurons_mdrg$ligand))
length(unique(all_types_data_final_spinal_cord_excitatory_neurons_mdrg$receptor))
all_types_data_final_spinal_cord_excitatory_neurons_ra = all_types_data_final_spinal_cord_excitatory_neurons[all_types_data_final_spinal_cord_excitatory_neurons$location == 'RA sc',]
nrow(all_types_data_final_spinal_cord_excitatory_neurons_ra)
length(unique(all_types_data_final_spinal_cord_excitatory_neurons_ra$ligand))
length(unique(all_types_data_final_spinal_cord_excitatory_neurons_ra$receptor))

print('------------Sympathetic cholinergic neurons--------------')
length(unique(all_types_data_final_sympathetic_cholinergic_neurons$ligand))
length(unique(all_types_data_final_sympathetic_cholinergic_neurons$receptor))
all_types_data_final_sympathetic_cholinergic_neurons_mdrg = all_types_data_final_sympathetic_cholinergic_neurons[all_types_data_final_sympathetic_cholinergic_neurons$location == 'mDRG',]
nrow(all_types_data_final_sympathetic_cholinergic_neurons_mdrg)
length(unique(all_types_data_final_sympathetic_cholinergic_neurons_mdrg$ligand))
length(unique(all_types_data_final_sympathetic_cholinergic_neurons_mdrg$receptor))
all_types_data_final_sympathetic_cholinergic_neurons_ra = all_types_data_final_sympathetic_cholinergic_neurons[all_types_data_final_sympathetic_cholinergic_neurons$location == 'RA sc',]
nrow(all_types_data_final_sympathetic_cholinergic_neurons_ra)
length(unique(all_types_data_final_sympathetic_cholinergic_neurons_ra$ligand))
length(unique(all_types_data_final_sympathetic_cholinergic_neurons_ra$receptor))
print('------------Sympathetic noradrenergic neurons--------------')
length(unique(all_types_data_final_sympathetic_nonadrenergic_neurons$ligand))
length(unique(all_types_data_final_sympathetic_nonadrenergic_neurons$receptor))
all_types_data_final_sympathetic_nonadrenergic_neurons_mdrg = all_types_data_final_sympathetic_nonadrenergic_neurons[all_types_data_final_sympathetic_nonadrenergic_neurons$location == 'mDRG',]
nrow(all_types_data_final_sympathetic_nonadrenergic_neurons_mdrg)
length(unique(all_types_data_final_sympathetic_nonadrenergic_neurons_mdrg$ligand))
length(unique(all_types_data_final_sympathetic_nonadrenergic_neurons_mdrg$receptor))
all_types_data_final_sympathetic_nonadrenergic_neurons_ra = all_types_data_final_sympathetic_nonadrenergic_neurons[all_types_data_final_sympathetic_nonadrenergic_neurons$location == 'RA sc',]
nrow(all_types_data_final_sympathetic_nonadrenergic_neurons_ra)
length(unique(all_types_data_final_sympathetic_nonadrenergic_neurons_ra$ligand))
length(unique(all_types_data_final_sympathetic_nonadrenergic_neurons_ra$receptor))

print('-------------Peripheral sensory neurofilament------------------')
length(unique(all_types_data_final_peripheral_sensory_neurofilament_neurons$ligand))
length(unique(all_types_data_final_peripheral_sensory_neurofilament_neurons$receptor))
all_types_data_final_peripheral_sensory_neurofilament_neurons_mdrg = all_types_data_final_peripheral_sensory_neurofilament_neurons[all_types_data_final_peripheral_sensory_neurofilament_neurons$location == 'mDRG',]
nrow(all_types_data_final_peripheral_sensory_neurofilament_neurons_mdrg)
length(unique(all_types_data_final_peripheral_sensory_neurofilament_neurons_mdrg$ligand))
length(unique(all_types_data_final_peripheral_sensory_neurofilament_neurons_mdrg$receptor))
all_types_data_final_peripheral_sensory_neurofilament_neurons_ra = all_types_data_final_peripheral_sensory_neurofilament_neurons[all_types_data_final_peripheral_sensory_neurofilament_neurons$location == 'RA sc',]
nrow(all_types_data_final_peripheral_sensory_neurofilament_neurons_ra)
length(unique(all_types_data_final_peripheral_sensory_neurofilament_neurons_ra$ligand))
length(unique(all_types_data_final_peripheral_sensory_neurofilament_neurons_ra$receptor))

print('-------------Peripheral sensory non-peptidergic neurons------------------')
length(unique(all_types_data_final_peripheral_sensory_nonpeptidergic_neurons$ligand))
length(unique(all_types_data_final_peripheral_sensory_nonpeptidergic_neurons$receptor))
all_types_data_final_peripheral_sensory_nonpeptidergic_neurons_mdrg = all_types_data_final_peripheral_sensory_nonpeptidergic_neurons[all_types_data_final_peripheral_sensory_nonpeptidergic_neurons$location == 'mDRG',]
nrow(all_types_data_final_peripheral_sensory_nonpeptidergic_neurons_mdrg)
length(unique(all_types_data_final_peripheral_sensory_nonpeptidergic_neurons_mdrg$ligand))
length(unique(all_types_data_final_peripheral_sensory_nonpeptidergic_neurons_mdrg$receptor))
all_types_data_final_peripheral_sensory_nonpeptidergic_neurons_ra = all_types_data_final_peripheral_sensory_nonpeptidergic_neurons[all_types_data_final_peripheral_sensory_nonpeptidergic_neurons$location == 'RA sc',]
nrow(all_types_data_final_peripheral_sensory_nonpeptidergic_neurons_ra)
length(unique(all_types_data_final_peripheral_sensory_nonpeptidergic_neurons_ra$ligand))
length(unique(all_types_data_final_peripheral_sensory_nonpeptidergic_neurons_ra$receptor))

print('-------------Peripheral sensory peptidergic neurons------------------')
length(unique(all_types_data_final_peripheral_sensory_peptidergic_neurons$ligand))
length(unique(all_types_data_final_peripheral_sensory_peptidergic_neurons$receptor))
all_types_data_final_peripheral_sensory_peptidergic_neurons_mdrg = all_types_data_final_peripheral_sensory_peptidergic_neurons[all_types_data_final_peripheral_sensory_peptidergic_neurons$location == 'mDRG',]
nrow(all_types_data_final_peripheral_sensory_peptidergic_neurons_mdrg)
length(unique(all_types_data_final_peripheral_sensory_peptidergic_neurons_mdrg$ligand))
length(unique(all_types_data_final_peripheral_sensory_peptidergic_neurons_mdrg$receptor))
all_types_data_final_peripheral_sensory_peptidergic_neurons_ra = all_types_data_final_peripheral_sensory_peptidergic_neurons[all_types_data_final_peripheral_sensory_peptidergic_neurons$location == 'RA sc',]
nrow(all_types_data_final_peripheral_sensory_peptidergic_neurons_ra)
length(unique(all_types_data_final_peripheral_sensory_peptidergic_neurons_ra$ligand))
length(unique(all_types_data_final_peripheral_sensory_peptidergic_neurons_ra$receptor))
```





Below we create the interactome plot from the mouse DRG ligand to the human RA receptors.

```{r, fig.height=10,fig.width=10}
library(BioCircos)

# Chromosomes on which the Receptor points should be displayed
points_chromosomes_ligand = c('Ligand', 'Ligand')
# Coordinates on which the Receptor points should be displayed (length.out = 6 because there are 6 unique genes in the receptor column
points_coordinates_ligand = seq(3, 98, length.out = 9)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique ligands
points_values_ligand = 1:length(points_coordinates_ligand)

# Size of the outside of the circle for ligands and receptors
# Since there are uniquely 9 ligands and 23 receptor genes, we make the slices of the circle 9/32*360 = 101 for the receptors and 360 - 101 for the ligands
myGenome = list("Ligand" = 101,
        "Receptor" = 259)

# Create circle for cell type identification to lie in
tracklist = BioCircosBackgroundTrack("myBackgroundTrack", minRadius = 0.78, maxRadius = 1.23,
  borderColors = "#AAAAAA", borderSize = 0.6, fillColors = "White") 

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("blue", "blue", "white", "white", "white", "blue", "white", "blue", "white"), minRadius = 1.21, maxRadius = 1.21)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("green", "white", "white", "white", "white", "white", "green", "white", "white"), minRadius = 1.08, maxRadius = 1.08)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("orange", "white", "orange", "white", "white", "white", "orange", "orange", "orange"), minRadius = 0.95, maxRadius = 0.95)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("red", "red", "white", "red", "red", "red", "red", "white", "red"), minRadius = 0.81, maxRadius = 0.81)

# Chromosomes on which the ligand should be displayed
points_chromosomes_receptor = c('Receptor', 'Receptor')
# Coordinates on which the ligand should be displayed length.out = 6 
points_coordinates_receptor = seq(3, 256, length.out = 23)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique receptors
points_values_receptor = 1:length(points_chromosomes_receptor)

# Cerebellum neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("antiquewhite4", "antiquewhite4", "antiquewhite4", "antiquewhite4", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.80, maxRadius = 0.80)

# Cholinergic and monoaminergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "black", "white", "black", "black", "black", "black", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.829, maxRadius = 0.829)

# Di- and mesencphalon excitatory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("blue", "blue", "blue", "blue", "blue", "white", "blue", "white", "blue", "blue", "blue", "blue", "blue", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.859, maxRadius = 0.859)

# Di- and mesencphalon inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("blueviolet", "blueviolet", "blueviolet", "blueviolet", "white", "white", "blueviolet", "white", "white", "blueviolet", "blueviolet", "white", "blueviolet", "blueviolet", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.888, maxRadius = 0.888)

# Enteric neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "brown", "brown", "brown", "white", "brown", "brown", "brown", "white", "white", "white", "brown", "brown", "brown", "white", "white", "white", "white", "white", "white"), minRadius = 0.917, maxRadius = 0.917)

# Hindbrain neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("cadetblue1", "cadetblue1", "cadetblue1", "cadetblue1", "cadetblue1", "white", "cadetblue1", "white", "white", "white", "white", "cadetblue1", "white", "white", "cadetblue1", "white", "white", "cadetblue1", "cadetblue1", "white", "white", "white", "white"), minRadius = 0.946, maxRadius = 0.946)

# Peripheral sensory neurofilament neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("gold", "white", "gold", "white", "gold", "white", "gold", "white", "gold", "gold", "gold", "white", "gold", "gold", "gold", "white", "white", "white", "gold", "gold", "gold", "gold", "white"), minRadius = 0.976, maxRadius = 0.976)

# Peripheral sensory non-peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("darkgrey", "white", "white", "white", "darkgrey", "white", "white", "white", "darkgrey", "white", "white", "white", "white", "white", "white", "white", "white", "white", "darkgrey", "white", "white", "white", "white"), minRadius = 1.01, maxRadius = 1.01)

# Peripheral sensory peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("magenta", "white", "white", "white", "magenta", "white", "white", "white", "magenta", "white", "white", "white", "white", "white", "white", "white", "white", "white", "magenta", "white", "magenta", "white", "white"), minRadius = 1.034, maxRadius = 1.034)

# Sympathetic cholinergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "chartreuse3"), minRadius = 1.064, maxRadius = 1.064)

# Sympathetic nonadrenergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cyan", "white", "white", "white", "white", "cyan", "white", "white"), minRadius = 1.093, maxRadius = 1.093)

# Dendate gyrus granule neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "orange", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.122, maxRadius = 1.122)

# Olfactory inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("coral", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.151, maxRadius = 1.151)

# Spinal cord inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "firebrick", "firebrick", "white", "white", "firebrick", "white", "firebrick", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.21, maxRadius = 1.21)

# Peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "purple", "white", "white", "white", "purple", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.21, maxRadius = 1.21)

#tracklist = tracklist + BioCircosTextTrack("testText", 'mDRG to Synovial Fibroblast', weight = "bold", size = '1em', x = -0.46, y = -1.25, displayGenomeBorder = TRUE)

links_chromosomes_1 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should start
links_chromosomes_2 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should end

# Coordinates to place start of links. The coordinates below correspond to the sequence given by seq(3, 98, length.out = 9) Their starting coordinate must be repeated for the number of connections between them and the number of ligands they are connected to.
links_pos_1 = c(3, 14.88, 3, 3, 26.75, 38.63, 38.63, 3, 14.88, 3, 14.88, 3, 3, 3, 3, 26.75, 3, 38.63, 3, 14.88, 3, 50.5, 62.38, 3, 74.25, 3, 74.25, 3, 62.38, 74.25, 3, 3, 3, 3, 3, 86.13, 14.88, 74.25, 3, 3, 3, 3, 74.25, 74.25, 3, 3, 3, 3, 86.13, 26.75, 74.25, 38.63, 38.63, 74.25, 3, 3, 3, 3, 14.88, 74.25, 50.5, 3, 3, 3, 14.88, 74.25, 3, 3, 3, 74.25, 74.25, 3, 3, 3, 26.75, 74.25, 38.63, 38.63, 74.25, 3, 3, 3, 14.88, 74.25, 50.5, 3, 3, 14.88, 3, 86.13, 74.25, 3, 3, 74.25, 3, 74.25, 3, 3, 26.75, 74.25, 3, 86.13, 74.25, 3, 3, 14.88, 74.25, 3, 50.5, 3, 3, 3, 3, 86.13, 14.88, 14.88, 74.25, 3, 3, 3, 3, 74.25, 3, 3, 3, 3, 86.13, 26.75, 26.75, 74.25, 38.63, 38.63, 3, 3, 3, 3, 14.88, 14.88, 50.5, 3, 3, 3, 3, 38.63, 50.5, 3, 3, 3, 3, 3, 3, 14.88, 86.13, 74.25, 74.25, 3, 3, 3, 3, 3, 3, 74.25, 74.25, 74.25, 3, 3, 3, 3, 3, 3, 26.75, 74.25, 98, 86.13, 74.25, 38.63, 74.25, 3, 3, 3, 3, 3, 3, 14.88, 74.25, 50.5, 98, 3, 3, 3, 74.25, 3, 3, 3, 74.25, 3, 3, 3, 74.25, 3, 3, 3, 3, 3, 3, 86.13, 74.25, 3, 3, 3, 74.25, 3, 3, 3, 86.13, 74.25, 3, 3, 3, 3, 3, 3, 38.63, 38.63, 3, 50.5, 3, 3, 3, 3, 3, 3, 3, 3, 3, 86.13, 86.13, 3, 3, 86.13, 86.13, 3)
# Coordinates to place end of links. The coordinates below correspond to the sequence given by seq(3, 256, length.out = 23)
links_pos_2 = c(3, 14.5, 3, 3, 14.5, 26, 37.5, 3, 14.5, 49, 14.5, 60.5, 49, 60.5, 49, 14.5, 60.5, 37.5, 49, 14.5, 60.5, 72, 83.5, 95, 106.5, 95, 106.5, 95, 83.5, 106.5, 95, 49, 3, 95, 118, 129.5, 14.5, 106.5, 49, 3, 95, 118, 141, 106.5, 49, 3, 95, 118, 129.5, 14.5, 141, 26, 37.5, 106.5, 49, 3, 95, 118, 14.5, 141, 72, 152.5, 3, 118, 14.5, 106.5, 152.5, 3, 118, 141, 106.5, 152.5, 3, 118, 14.5, 141, 26, 37.5, 106.5, 152.5, 3, 118, 14.5, 141, 72, 49, 95, 164, 60.5, 175.5, 106.5, 49, 95, 187, 60.5, 106.5, 49, 95, 164, 187, 60.5, 175.5, 106.5, 49, 95, 164, 187, 60.5, 72, 49, 3, 198.5, 118, 129.5, 164, 14.5, 210, 49, 3, 198.5, 118, 210, 49, 3, 198.5, 118, 129.5, 164, 14.5, 210, 26, 37.5, 49, 3, 198.5, 118, 164, 14.5, 72, 3, 3, 3, 3, 26, 72, 49, 152.5, 3, 95, 221.5, 118, 164, 233, 210, 106.5, 49, 152.5, 3, 95, 221.5, 118, 141, 210, 106.5, 49, 152.5, 3, 95, 221.5, 118, 164, 141, 244.5, 233, 210, 26, 106.5, 49, 152.5, 3, 95, 221.5, 118, 164, 141, 72, 244.5, 49, 3, 95, 210, 49, 3, 95, 210, 49, 3, 95, 210, 49, 3, 95, 49, 3, 95, 233, 210, 49, 3, 95, 210, 49, 3, 95, 233, 210, 49, 3, 95, 95, 95, 95, 26, 37.5, 95, 72, 49, 256, 49, 256, 49, 256, 49, 256, 49, 175.5, 233, 49, 49, 175.5, 233, 49)
# The labels are given at the end of the link. 
links_labels_12 = c("FGF12", "LRRC4B", "", "", "", "CLSTN1", "CLSTN3", "", "", "FGF1", "", "NCAM1", "", "", "", "", "", "", "", "", "", "NFASC", "CXCL12", "FGF13", "EDIL3", "", "", "", "", "", "", "", "", "", "FGF9", "INHA", "", "", "", "", "", "", "LTBP3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF11", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "LFRN4", "", "TGFB1", "", "", "", "LTBP1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF14", "", "", "", "", "ADAM9", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF18", "", "", "TGFB3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "SHH", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF7", "", "", "", "", "", "", "", "", "", "", "", "", "", "") 

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_1, links_pos_1,
  links_pos_1, links_chromosomes_2, links_pos_2, links_pos_2,
  maxRadius = 0.5, width = "0.01em", color = 'brown', labelSize = "0.75em", labels = links_labels_12)

links_chromosomes_3 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should start
links_chromosomes_4 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should end

# Now we start at the ligands
links_pos_3 = c(3, 14.5, 3, 3, 14.5, 26, 37.5, 3, 14.5, 49, 14.5, 60.5, 49, 60.5, 49, 14.5, 60.5, 37.5, 49, 14.5, 60.5, 72, 83.5, 95, 106.5, 95, 106.5, 95, 83.5, 106.5, 95, 49, 3, 95, 118, 129.5, 14.5, 106.5, 49, 3, 95, 118, 141, 106.5, 49, 3, 95, 118, 129.5, 14.5, 141, 26, 37.5, 106.5, 49, 3, 95, 118, 14.5, 141, 72, 152.5, 3, 118, 14.5, 106.5, 152.5, 3, 118, 141, 106.5, 152.5, 3, 118, 14.5, 141, 26, 37.5, 106.5, 152.5, 3, 118, 14.5, 141, 72, 49, 95, 164, 60.5, 175.5, 106.5, 49, 95, 187, 60.5, 106.5, 49, 95, 164, 187, 60.5, 175.5, 106.5, 49, 95, 164, 187, 60.5, 72, 49, 3, 198.5, 118, 129.5, 164, 14.5, 210, 49, 3, 198.5, 118, 210, 49, 3, 198.5, 118, 129.5, 164, 14.5, 210, 26, 37.5, 49, 3, 198.5, 118, 164, 14.5, 72, 3, 3, 3, 3, 26, 72, 49, 152.5, 3, 95, 221.5, 118, 164, 233, 210, 106.5, 49, 152.5, 3, 95, 221.5, 118, 141, 210, 106.5, 49, 152.5, 3, 95, 221.5, 118, 164, 141, 244.5, 233, 210, 26, 106.5, 49, 152.5, 3, 95, 221.5, 118, 164, 141, 72, 244.5, 49, 3, 95, 210, 49, 3, 95, 210, 49, 3, 95, 210, 49, 3, 95, 49, 3, 95, 233, 210, 49, 3, 95, 210, 49, 3, 95, 233, 210, 49, 3, 95, 95, 95, 95, 26, 37.5, 95, 72, 49, 256, 49, 256, 49, 256, 49, 256, 49, 175.5, 233, 49, 49, 175.5, 233, 49)
# Now we end at the receptors
links_pos_4 = c(3, 14.88, 3, 3, 26.75, 38.63, 38.63, 3, 14.88, 3, 14.88, 3, 3, 3, 3, 26.75, 3, 38.63, 3, 14.88, 3, 50.5, 62.38, 3, 74.25, 3, 74.25, 3, 62.38, 74.25, 3, 3, 3, 3, 3, 86.13, 14.88, 74.25, 3, 3, 3, 3, 74.25, 74.25, 3, 3, 3, 3, 86.13, 26.75, 74.25, 38.63, 38.63, 74.25, 3, 3, 3, 3, 14.88, 74.25, 50.5, 3, 3, 3, 14.88, 74.25, 3, 3, 3, 74.25, 74.25, 3, 3, 3, 26.75, 74.25, 38.63, 38.63, 74.25, 3, 3, 3, 14.88, 74.25, 50.5, 3, 3, 14.88, 3, 86.13, 74.25, 3, 3, 74.25, 3, 74.25, 3, 3, 26.75, 74.25, 3, 86.13, 74.25, 3, 3, 14.88, 74.25, 3, 50.5, 3, 3, 3, 3, 86.13, 14.88, 14.88, 74.25, 3, 3, 3, 3, 74.25, 3, 3, 3, 3, 86.13, 26.75, 26.75, 74.25, 38.63, 38.63, 3, 3, 3, 3, 14.88, 14.88, 50.5, 3, 3, 3, 3, 38.63, 50.5, 3, 3, 3, 3, 3, 3, 14.88, 86.13, 74.25, 74.25, 3, 3, 3, 3, 3, 3, 74.25, 74.25, 74.25, 3, 3, 3, 3, 3, 3, 26.75, 74.25, 98, 86.13, 74.25, 38.63, 74.25, 3, 3, 3, 3, 3, 3, 14.88, 74.25, 50.5, 98, 3, 3, 3, 74.25, 3, 3, 3, 74.25, 3, 3, 3, 74.25, 3, 3, 3, 3, 3, 3, 86.13, 74.25, 3, 3, 3, 74.25, 3, 3, 3, 86.13, 74.25, 3, 3, 3, 3, 3, 3, 38.63, 38.63, 3, 50.5, 3, 3, 3, 3, 3, 3, 3, 3, 3, 86.13, 86.13, 3, 3, 86.13, 86.13, 3)
# These links are for the receptors
links_labels_34 = c('FGFR1', "PTPRS", "", "", "PTPRD", "APP", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "ANK3", "ACKR3", "", "ITGB5", "", "", "", "", "", "", "", "", "", "", "TGFBR3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "CDON", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_3, links_pos_3,
  links_pos_3, links_chromosomes_4, links_pos_4, links_pos_4,
  maxRadius = 0.5, width = "0.01em", color = 'brown', labelSize = "0.75em", labels = links_labels_34, labelPadding = 3)

#links_chromosomes_5 = c('ligand', 'hDRG', 'hDRG') # Chromosomes on which the links should start
#links_chromosomes_6 = c('RA', 'RA', 'RA') # Chromosomes on which the links should end
#links_pos_5 = c(180.08, 187.46, 194.84, 202.22, 209.59, 216.97, 224.35, 231.73, 239.11, 216.97, 246.49, 253.86, 261.24, 268.62, 276)
#links_pos_6 = c(48, 55.5, 55.5, 55.5, 55.5, 55.5, 55.5, 55.5, 63, 63, 63, 63, 70.5, 70.5, 78)
#links_labels_56 = c("", "", '', '', '', '', '', '', "", '', '', '', "", '', "")

#tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_5, links_pos_5,
#  links_pos_5, links_chromosomes_6, links_pos_6, links_pos_6,
#  maxRadius = 0.5, width = "0.01em", color = 'brown', labelSize = "0.5em", labels = links_labels_56)

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra = c('Ligand') # Chromosomes on which the arcs should be displayed
arcs_begin_ra = c(0)
arcs_end_ra = c(101)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra, arcs_begin_ra, arcs_end_ra, minRadius = 1.25, maxRadius = 1.30, colors = 'forestgreen') 

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_hdrg = c('Receptor') # Chromosomes on which the arcs should be displayed
arcs_begin_hdrg = c(0)
arcs_end_hdrg = c(259)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_hdrg, arcs_begin_hdrg, arcs_end_hdrg, minRadius = 1.25, maxRadius = 1.30, colors = 'deeppink3')

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Mouse DRG to Human RA - All Cell Types Subset to Neurons', x = -1, y = -1.37, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Receptor', x = 0.85, y = -1.1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Ligand', x = -1.05, y = -1.1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes, arcs_begin, arcs_end, minRadius = 0.85, maxRadius = 0.99, colors = 'forestgreen') 

BioCircos(tracklist, genome = myGenome, genomeFillColor = c('white', 'white'), chrPad = 0.05, displayGenomeBorder = FALSE, yChr =  FALSE,
  genomeTicksDisplay = FALSE,  genomeLabelTextSize = 0, genomeLabelDy = 0, width = 650, height = 650)
```

Below we make a legend for the fibroblast subtypes.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (4)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (2)', 'DKK3+ sublining fibroblasts (SC-F3) (5)', 'CD55+ lining fibroblasts (SC-F4) (7)'), pch=16, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes", at=0.24, cex=2)
```

Below we plot the legend for the cell types.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Cerebellum Neurons (4)', 'Cholinergic and Monoaminergic Neurons (5)', 'Di- and Mesencphalon Excitatory Neurons (11)', 'Di- and Mesencphalon Inhibitory Neurons (9)', 'Enteric Neurons (9)', 'Hindbrain Neurons (11)', 'Peripheral Sensory Neurofilament Neurons (14)', 'Peripheral Sensory Non-peptidergic Neurons (4)', 'Peripheral Sensory Peptidergic Neurons (5)', 'Sympathetic Cholinergic Neurons (2)', 'Sympathetic Noradrenergic Neurons (3)', 'Dentate Gyrus Granule Neurons (3)', 'Olfactory Inhibitory Neurons (1)', 'Spinal Cord Inhibitory Neurons (4)', "Peptidergic Neurons (2)"), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('antiquewhite4', 'black', 'blue', 'blueviolet', 'brown', 'cadetblue1', 'gold', 'darkgrey', 'magenta', 'chartreuse3', 'cyan', 'orange', 'coral', 'darkgreen', 'firebrick'
            ))
mtext("mDRG Neuronal Cell Types", at=0.19, cex=1.1)
```

Below we create the same interactome plot from the human RA receptors to the mouse DRG ligands.

```{r, fig.height=10,fig.width=10}
library(BioCircos)

# Chromosomes on which the Receptor points should be displayed
points_chromosomes_ligand = c('Ligand', 'Ligand')
# Coordinates on which the Receptor points should be displayed (length.out = 11 because there are 11 unique genes in the Ligand column
points_coordinates_ligand = seq(3, 94, length.out = 11)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique ligands
points_values_ligand = 1:length(points_coordinates_ligand)

# Size of the outside of the circle for ligands and receptors
# Since there are uniquely 11 ligands and 31 receptor genes, we make the slices of the circle 30/41*360 = 263 for the receptors and 360 - 263 for the ligands
myGenome = list("Ligand" = 97, "Receptor" = 263)

# Create circle for cell type identification to lie in
tracklist = BioCircosBackgroundTrack("myBackgroundTrack", minRadius = 0.78, maxRadius = 1.23,
  borderColors = "#AAAAAA", borderSize = 0.6, fillColors = "White") 

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("blue", "blue", "blue", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.21, maxRadius = 1.21)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("green", "green", "green", "green", "green", "white", "white", "white", "white", "white", "white"), minRadius = 1.08, maxRadius = 1.08)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("orange", "white", "orange", "white", "orange", "white", "white", "white", "white", "white", "white"), minRadius = 0.95, maxRadius = 0.95)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("red", "red", "red", "red", "red", "red", "red", "red", "red", "red", "red"), minRadius = 0.81, maxRadius = 0.81)

# Chromosomes on which the ligand should be displayed
points_chromosomes_receptor = c('Receptor', 'Receptor')
# Coordinates on which the ligand should be displayed length.out = 30 
points_coordinates_receptor = seq(3, 260, length.out = 30)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique receptors
points_values_receptor = 1:length(points_chromosomes_receptor)

# Cerebellum neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("antiquewhite4", "antiquewhite4", "white", "white", "white", "white", "white", "antiquewhite4", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.80, maxRadius = 0.80)

# Cholinergic and monoaminergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "black", "black", "white", "white", "white", "white", "black", "white", "white", "white", "white", "white", "white", "white", "white", "black", "black", "black", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.829, maxRadius = 0.829)

# Di- and mesencphalon excitatory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("blue", "white", "white", "white", "blue", "white", "white", "white", "blue", "white", "white", "white", "blue", "blue", "white", "white", "white", "white", "blue", "blue", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.859, maxRadius = 0.859)

# Di- and mesencphalon inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "blueviolet", "white", "white", "white", "blueviolet", "blueviolet", "white", "white", "blueviolet", "blueviolet", "white", "white", "white", "white", "blueviolet", "blueviolet", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.888, maxRadius = 0.888)

# Enteric neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "brown", "brown", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "brown", "white", "white", "white", "brown", "white", "brown", "white", "brown", "brown", "brown", "white", "white", "white", "white", "white"), minRadius = 0.917, maxRadius = 0.917)

# Hindbrain neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "cadetblue1", "white", "cadetblue1", "cadetblue1", "white", "white", "cadetblue1", "cadetblue1", "white", "cadetblue1", "cadetblue1", "cadetblue1", "white", "cadetblue1", "cadetblue1", "white", "cadetblue1", "white", "cadetblue1", "white", "white", "white", "white", "cadetblue1", "white", "white", "white", "white"), minRadius = 0.946, maxRadius = 0.946)

# Peripheral sensory neurofilament neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "gold", "gold", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "gold", "gold", "white", "gold", "gold", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.976, maxRadius = 0.976)

# Peripheral sensory non-peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "darkgrey", "darkgrey", "white", "white", "darkgrey", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "darkgrey", "white", "white", "darkgrey", "darkgrey", "darkgrey", "white", "darkgrey", "darkgrey", "darkgrey", "darkgrey"), minRadius = 1.01, maxRadius = 1.01)

# Peripheral sensory peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "magenta", "magenta", "white", "white", "magenta", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "magenta", "white", "white", "magenta", "magenta", "magenta", "white", "white", "magenta", "magenta", "magenta"), minRadius = 1.034, maxRadius = 1.034)

# Sympathetic cholinergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "chartreuse3", "white", "white", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "chartreuse3", "white", "white", "white", "chartreuse3", "chartreuse3", "chartreuse3", "white", "white", "white", "white", "white"), minRadius = 1.064, maxRadius = 1.064)

# Sympathetic nonadrenergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "cyan", "cyan", "white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cyan", "white", "white", "white", "cyan", "white", "cyan", "white", "white", "white", "white", "white"), minRadius = 1.093, maxRadius = 1.093)

# Dendate gyrus granule neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "white", "orange", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "white", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.122, maxRadius = 1.122)

# Olfactory inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "coral", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.151, maxRadius = 1.151)

# Spinal cord excitatory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "darkgreen", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.181, maxRadius = 1.181)

# Spinal cord inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "firebrick", "white", "white", "white", "firebrick", "white", "white", "white", "white", "white", "white", "white", "white", "firebrick", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.21, maxRadius = 1.21)

#tracklist = tracklist + BioCircosTextTrack("testText", 'mDRG to Synovial Fibroblast', weight = "bold", size = '1em', x = -0.46, y = -1.25, displayGenomeBorder = TRUE)

links_chromosomes_1 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should start
links_chromosomes_2 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should end

# Coordinates to place start of links. The coordinates below correspond to the sequence given by seq(3, 260, length.out = 30) Their starting coordinate must be repeated for the number of connections between them and the number of ligands they are connected to.
links_pos_1 = c(3, 11.86, 3, 20.72, 29.59, 3, 3, 38.45, 38.45, 20.72, 29.59, 47.31, 38.45, 20.72, 20.72, 29.59, 56.17, 20.72, 29.59, 56.17, 20.72, 29.59, 56.17, 20.72, 56.17, 20.72, 29.59, 3, 11.86, 65.03, 3, 73.90, 20.72, 29.59, 82.76, 91.62, 100.48, 3, 109.34, 118.21, 73.90, 3, 38.45, 109.34, 82.76, 118.21, 73.90, 38.45, 127.07, 20.72, 29.59, 47.31, 109.34, 135.93, 82.76, 100.48, 118.21, 73.90, 144.79, 38.45, 20.72, 91.62, 20.72, 29.59, 56.17, 20.72, 29.59, 56.17, 20.72, 29.59, 73.90, 109.34, 73.90, 56.17, 20.72, 56.17, 20.72, 29.59, 3, 11.86, 20.72, 29.59, 3, 20.72, 29.59, 47.31, 144.79, 20.72, 20.72, 29.59, 56.17, 20.72, 29.59, 56.17, 20.72, 29.59, 56.17, 20.72, 56.17, 20.72, 29.59, 3, 3, 11.86, 65.03, 3, 153.66, 162.52, 73.90, 171.38, 20.72, 29.59, 82.76, 91.62, 100.48, 180.24, 189.10, 3, 162.52, 3, 109.34, 118.21, 73.90, 171.38, 3, 38.45, 162.52, 109.34, 82.76, 118.21, 73.90, 171.38, 38.45, 197.97, 162.52, 127.07, 206.83, 215.69, 180.24, 20.72, 29.59, 224.55, 162.52, 47.31, 47.31, 109.34, 135.93, 82.76, 100.48, 118.21, 73.90, 47.31, 144.79, 180.24, 38.45, 20.72, 91.62, 197.97, 162.52, 189.10, 171.38, 20.72, 29.59, 233.41, 197.97, 242.28, 162.52, 56.17, 206.83, 251.14, 215.69, 189.10, 260, 20.72, 29.59, 197.97, 242.28, 162.52, 56.17, 206.83, 251.14, 215.69, 189.10, 260, 20.72, 29.59, 73.90, 109.34, 73.90, 189.10, 197.97, 162.52, 56.17, 206.83, 215.69, 20.72, 197.97, 162.52, 56.17, 215.69, 20.72, 29.59)
# Coordinates to place end of links. The coordinates below correspond to the sequence given by seq(3, 94, length.out = 11)
links_pos_2 = c(3, 3, 12.1, 21.2, 21.2, 3, 12.1, 12.1, 12.1, 21.2, 21.2, 3, 12.1, 21.2, 21.2, 21.2, 3, 21.2, 21.2, 3, 21.2, 21.2, 3, 21.2, 3, 21.2, 21.2, 3, 3, 30.3, 12.1, 30.3, 21.2, 21.2, 30.3, 30.3, 30.3, 3, 30.3, 30.3, 30.3, 12.1, 21.1, 30.3, 30.3, 30.3, 30.3, 12.1, 30.3, 21.2, 21.2, 3, 30.3, 30.3, 30.3, 30.3, 30.3, 30.3, 39.4, 12.1, 21.2, 30.3, 21.2, 21.2, 3, 21.2, 21.2, 3, 21.2, 21.2, 30.3, 30.3, 30.3, 3, 21.2, 3, 21.2, 21.2, 3, 3, 21.2, 21.2, 3, 21.2, 21.2, 3, 39.4, 21.2, 21.2, 21.2, 3, 21.2, 21.2, 3, 21.2, 21.2, 3, 21.2, 3, 21.2, 21.2, 48.5, 3, 3, 30.3, 12.1, 48.5, 48.5, 30.3, 57.6, 21.2, 21.2, 30.3, 30.3, 30.3, 66.7, 75.8, 48.5, 48.5, 3, 30.3, 30.3, 30.3, 57.6, 12.1, 12.1, 48.5, 30.3, 30.3, 30.3, 30.3, 57.6, 12.1, 48.5, 48.5, 30.3, 84.9, 84.9, 66.7, 21.2, 21.2, 48.5, 48.5, 94, 3, 30.3, 30.3, 30.3, 30.3, 30.3, 30.3, 84.9, 39.4, 66.7, 12.1, 21.2, 30.3, 48.5, 48.5, 75.8, 57.6, 21.2, 21.2, 48.5, 48.5, 48.5, 48.5, 3, 84.9, 84.9, 84.9, 75.8, 75.8, 21.2, 21.2, 48.5, 48.5, 48.5, 3, 84.9, 84.9, 84.9, 75.8, 75.8, 21.2, 21.2, 30.3, 30.3, 30.3, 75.8, 48.5, 48.5, 3, 84.9, 84.9, 21.2, 48.5, 48.5, 3, 84.9, 21.2, 21.2)
# The labels are given at the end of the link. 
links_labels_12 = c("CTGF", "", "TFPI", "TIMP2", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "EFNA5", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "LTBP3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "APP", "", "", "", "", "", "", "", "SEMA6D", "", "", "", "", "", "NTN4", "SEMA3E", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "HBEGF", "", "", "", "", "", "", "BTC", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "") 

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_1, links_pos_1,
  links_pos_1, links_chromosomes_2, links_pos_2, links_pos_2,
  maxRadius = 0.5, width = "0.01em", color = 'cyan', labelSize = "0.75em", labels = links_labels_12)

links_chromosomes_3 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should start
links_chromosomes_4 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should end

# Now we start at the ligands
links_pos_3 = c(3, 3, 12.1, 21.2, 21.2, 3, 12.1, 12.1, 12.1, 21.2, 21.2, 3, 12.1, 21.2, 21.2, 21.2, 3, 21.2, 21.2, 3, 21.2, 21.2, 3, 21.2, 3, 21.2, 21.2, 3, 3, 30.3, 12.1, 30.3, 21.2, 21.2, 30.3, 30.3, 30.3, 3, 30.3, 30.3, 30.3, 12.1, 21.1, 30.3, 30.3, 30.3, 30.3, 12.1, 30.3, 21.2, 21.2, 3, 30.3, 30.3, 30.3, 30.3, 30.3, 30.3, 39.4, 12.1, 21.2, 30.3, 21.2, 21.2, 3, 21.2, 21.2, 3, 21.2, 21.2, 30.3, 30.3, 30.3, 3, 21.2, 3, 21.2, 21.2, 3, 3, 21.2, 21.2, 3, 21.2, 21.2, 3, 39.4, 21.2, 21.2, 21.2, 3, 21.2, 21.2, 3, 21.2, 21.2, 3, 21.2, 3, 21.2, 21.2, 48.5, 3, 3, 30.3, 12.1, 48.5, 48.5, 30.3, 57.6, 21.2, 21.2, 30.3, 30.3, 30.3, 66.7, 75.8, 48.5, 48.5, 3, 30.3, 30.3, 30.3, 57.6, 12.1, 12.1, 48.5, 30.3, 30.3, 30.3, 30.3, 57.6, 12.1, 48.5, 48.5, 30.3, 84.9, 84.9, 66.7, 21.2, 21.2, 48.5, 48.5, 94, 3, 30.3, 30.3, 30.3, 30.3, 30.3, 30.3, 84.9, 39.4, 66.7, 12.1, 21.2, 30.3, 48.5, 48.5, 75.8, 57.6, 21.2, 21.2, 48.5, 48.5, 48.5, 48.5, 3, 84.9, 84.9, 84.9, 75.8, 75.8, 21.2, 21.2, 48.5, 48.5, 48.5, 3, 84.9, 84.9, 84.9, 75.8, 75.8, 21.2, 21.2, 30.3, 30.3, 30.3, 75.8, 48.5, 48.5, 3, 84.9, 84.9, 21.2, 48.5, 48.5, 3, 84.9, 21.2, 21.2)
# Now we end at the receptors
links_pos_4 = c(3, 11.86, 3, 20.72, 29.59, 3, 3, 38.45, 38.45, 20.72, 29.59, 47.31, 38.45, 20.72, 20.72, 29.59, 56.17, 20.72, 29.59, 56.17, 20.72, 29.59, 56.17, 20.72, 56.17, 20.72, 29.59, 3, 11.86, 65.03, 3, 73.90, 20.72, 29.59, 82.76, 91.62, 100.48, 3, 109.34, 118.21, 73.90, 3, 38.45, 109.34, 82.76, 118.21, 73.90, 38.45, 127.07, 20.72, 29.59, 47.31, 109.34, 135.93, 82.76, 100.48, 118.21, 73.90, 144.79, 38.45, 20.72, 91.62, 20.72, 29.59, 56.17, 20.72, 29.59, 56.17, 20.72, 29.59, 73.90, 109.34, 73.90, 56.17, 20.72, 56.17, 20.72, 29.59, 3, 11.86, 20.72, 29.59, 3, 20.72, 29.59, 47.31, 144.79, 20.72, 20.72, 29.59, 56.17, 20.72, 29.59, 56.17, 20.72, 29.59, 56.17, 20.72, 56.17, 20.72, 29.59, 3, 3, 11.86, 65.03, 3, 153.66, 162.52, 73.90, 171.38, 20.72, 29.59, 82.76, 91.62, 100.48, 180.24, 189.10, 3, 162.52, 3, 109.34, 118.21, 73.90, 171.38, 3, 38.45, 162.52, 109.34, 82.76, 118.21, 73.90, 171.38, 38.45, 197.97, 162.52, 127.07, 206.83, 215.69, 180.24, 20.72, 29.59, 224.55, 162.52, 47.31, 47.31, 109.34, 135.93, 82.76, 100.48, 118.21, 73.90, 47.31, 144.79, 180.24, 38.45, 20.72, 91.62, 197.97, 162.52, 189.10, 171.38, 20.72, 29.59, 233.41, 197.97, 242.28, 162.52, 56.17, 206.83, 251.14, 215.69, 189.10, 260, 20.72, 29.59, 197.97, 242.28, 162.52, 56.17, 206.83, 251.14, 215.69, 189.10, 260, 20.72, 29.59, 73.90, 109.34, 73.90, 189.10, 197.97, 162.52, 56.17, 206.83, 215.69, 20.72, 197.97, 162.52, 56.17, 215.69, 20.72, 29.59)
# These links are for the receptors
links_labels_34 = c("LRP1", "LRP6", "", "ITGA3", "ITGB1", "", "", "VLDLR", "", "", "", "ERBB4", "", "", "", "", "NTRK1", "", "", "", "", "", "", "", "", "", "", "", "", "EPHB2", "", "EPHB6", "", "", "EPHA4", "EPHA5", "EPHA7", "", "EPHA10", "EPHB1", "", "", "", "", "", "", "", "", "EPHA8", "", "", "", "", "EPHA3", "", "", "", "", "ITGB5", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "NCSTN", "TNFRSF21", "", "PLXNA1", "", "", "", "", "", "UNC5A", "NRP1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "NGFR", "", "", "CD44", "CD9", "", "", "", "GPC1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "CAV1", "", "SLC45A3", "", "", "", "CD82", "", "", "PLXND1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_3, links_pos_3,
  links_pos_3, links_chromosomes_4, links_pos_4, links_pos_4,
  maxRadius = 0.5, width = "0.01em", color = 'cyan', labelSize = "0.75em", labels = links_labels_34, labelPadding = 3)

#links_chromosomes_5 = c('ligand', 'hDRG', 'hDRG') # Chromosomes on which the links should start
#links_chromosomes_6 = c('RA', 'RA', 'RA') # Chromosomes on which the links should end
#links_pos_5 = c(180.08, 187.46, 194.84, 202.22, 209.59, 216.97, 224.35, 231.73, 239.11, 216.97, 246.49, 253.86, 261.24, 268.62, 276)
#links_pos_6 = c(48, 55.5, 55.5, 55.5, 55.5, 55.5, 55.5, 55.5, 63, 63, 63, 63, 70.5, 70.5, 78)
#links_labels_56 = c("", "", '', '', '', '', '', '', "", '', '', '', "", '', "")

#tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_5, links_pos_5,
#  links_pos_5, links_chromosomes_6, links_pos_6, links_pos_6,
#  maxRadius = 0.5, width = "0.01em", color = 'brown', labelSize = "0.5em", labels = links_labels_56)

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra = c('Receptor') # Chromosomes on which the arcs should be displayed
arcs_begin_ra = c(0)
arcs_end_ra = c(263)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra, arcs_begin_ra, arcs_end_ra, minRadius = 1.25, maxRadius = 1.30, colors = 'forestgreen') 

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_hdrg = c('Ligand') # Chromosomes on which the arcs should be displayed
arcs_begin_hdrg = c(0)
arcs_end_hdrg = c(97)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_hdrg, arcs_begin_hdrg, arcs_end_hdrg, minRadius = 1.25, maxRadius = 1.30, colors = 'deeppink3')

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Human RA to Mouse DRG - All Cell Types Subset to Neurons', x = -1, y = -1.37, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Ligand', x = 0.85, y = -1.1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Receptor', x = -1.05, y = -1.1, size = "1em",
  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes, arcs_begin, arcs_end, minRadius = 0.85, maxRadius = 0.99, colors = 'forestgreen') 

BioCircos(tracklist, genome = myGenome, genomeFillColor = c('white', 'white'), chrPad = 0.05, displayGenomeBorder = FALSE, yChr =  FALSE,
  genomeTicksDisplay = FALSE,  genomeLabelTextSize = 0, genomeLabelDy = 0, width = 650, height = 650)
```

Below we make a legend for the fibroblast subtypes.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (3)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (5)', 'DKK3+ sublining fibroblasts (SC-F3) (3)', 'CD55+ lining fibroblasts (SC-F4) (11)'), pch=16, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes", at=0.24, cex=2)
```

Below we plot the legend for the cell types.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Cerebellum Neurons (3)', 'Cholinergic and Monoaminergic Neurons (6)', 'Di- and Mesencphalon Excitatory Neurons (7)', 'Di- and Mesencphalon Inhibitory Neurons (7)', 'Enteric Neurons (8)', 'Hindbrain Neurons (13)', 'Peripheral Sensory Neurofilament Neurons (6)', 'Peripheral Sensory Non-peptidergic Neurons (12)', 'Peripheral Sensory Peptidergic Neurons (11)', 'Sympathetic Cholinergic Neurons (6)', 'Sympathetic Noradrenergic Neurons (6)', 'Dentate Gyrus Granule Neurons (5)', 'Olfactory Inhibitory Neurons (1)', 'Spinal Cord Excitatory Neurons (1)', 'Spinal Cord Inhibitory Neurons (3)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('antiquewhite4', 'black', 'blue', 'blueviolet', 'brown', 'cadetblue1', 'gold', 'darkgrey', 'magenta', 'chartreuse3', 'cyan', 'orange', 'coral', 'darkgreen', 'firebrick'
            ))
mtext("mDRG Neuronal Cell Types", at=0.19, cex=1.1)
```



Below we create one plot with both mDRG to human RA and human RA to mDRG.

```{r, fig.height=10,fig.width=10}
library(BioCircos)

# Chromosomes on which the Receptor points should be displayed
points_chromosomes_ligand = c('Ligand', 'Ligand')
# Coordinates on which the Receptor points should be displayed (length.out = 19 because there are 19 unique genes in the receptor column
points_coordinates_ligand = seq(3, 92, length.out = 19)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique ligands
points_values_ligand = 1:length(points_coordinates_ligand)

# Size of the outside of the circle for ligands and receptors
# Since there are uniquely 19 ligands and 53 receptor genes, we make the slices of the circle 19/72*360 = 95 for the receptors and 360 - 95 for the ligands
myGenome = list("Ligand" = 95,
        "Receptor" = 265)

# Create circle for cell type identification to lie in
tracklist = BioCircosBackgroundTrack("myBackgroundTrack", minRadius = 0.83, maxRadius = 1.28,
  borderColors = "#AAAAAA", borderSize = 0.6, fillColors = "White") 

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("blue", "blue", "white", "white", "white", "blue", "white", "blue", "white", "blue", "blue", "blue", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.26, maxRadius = 1.26, shape = 'rect', size = 4)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("green", "white", "white", "white", "white", "white", "green", "white", "white", "green", "green", "green", "green", "green", "white", "white", "white", "white", "white", "white"), minRadius = 1.13, maxRadius = 1.13, shape = 'rect', size = 4)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("orange", "white", "orange", "white", "white", "white", "orange", "orange", "orange", "orange", "white", "orange", "white", "orange", "white", "white", "white", "white", "white", "white"), minRadius = 1.00, maxRadius = 1.00, shape = 'rect', size = 4)

tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_ligand, points_coordinates_ligand, 
  points_values_ligand, colors = c("red", "red", "white", "red", "red", "red", "red", "white", "red", "red", "red", "red", "red", "red", "red", "red", "red", "red", "red", "red"), minRadius = 0.86, maxRadius = 0.86, shape = 'rect', size = 4)

# Chromosomes on which the ligand should be displayed
points_chromosomes_receptor = c('Receptor', 'Receptor')
# Coordinates on which the ligand should be displayed length.out = 53
points_coordinates_receptor = seq(3, 262, length.out = 53)
# Values associated with each point, used as radial coordinate 
# We space them out according to the number of unique receptors
points_values_receptor = 1:length(points_chromosomes_receptor)

# Cerebellum neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("antiquewhite4", "antiquewhite4", "antiquewhite4", "antiquewhite4", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "antiquewhite4", "antiquewhite4", "white", "white", "white", "white", "white", "antiquewhite4", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.85, maxRadius = 0.85)

# Cholinergic and monoaminergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "black", "white", "black", "black", "black", "black", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "black", "black", "white", "white", "white", "white", "black", "white", "white", "white", "white", "white", "white", "white", "white", "black", "black", "black", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.877, maxRadius = 0.877)

# Di- and mesencphalon excitatory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("blue", "blue", "blue", "blue", "blue", "white", "blue", "white", "blue", "blue", "blue", "blue", "blue", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "blue", "white", "white", "white", "blue", "white", "white", "white", "blue", "white", "white", "white", "blue", "blue", "white", "white", "white", "white", "blue", "blue", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.903, maxRadius = 0.903)

# Di- and mesencphalon inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("blueviolet", "blueviolet", "blueviolet", "blueviolet", "white", "white", "blueviolet", "white", "white", "blueviolet", "blueviolet", "white", "blueviolet", "blueviolet", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "blueviolet", "white", "white", "white", "blueviolet", "blueviolet", "white", "white", "blueviolet", "blueviolet", "white", "white", "white", "white", "blueviolet", "blueviolet", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 0.93, maxRadius = 0.93)

# Enteric neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "brown", "brown", "brown", "white", "brown", "brown", "brown", "white", "white", "white", "brown", "brown", "brown", "white", "white", "white", "white", "white", "white", "white", "white", "brown", "brown", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "brown", "white", "white", "white", "brown", "white", "brown", "white", "brown", "brown", "brown", "white", "white", "white", "white", "white"), minRadius = 0.957, maxRadius = 0.957)

# Hindbrain neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("cadetblue1", "cadetblue1", "cadetblue1", "cadetblue1", "cadetblue1", "white", "cadetblue1", "white", "white", "white", "white", "cadetblue1", "white", "white", "cadetblue1", "white", "white", "cadetblue1", "cadetblue1", "white", "white", "white", "white", "white", "white", "cadetblue1", "white", "cadetblue1", "cadetblue1", "white", "white", "cadetblue1", "cadetblue1", "white", "cadetblue1", "cadetblue1", "cadetblue1", "white", "cadetblue1", "cadetblue1", "white", "cadetblue1", "white", "cadetblue1", "white", "white", "white", "white", "cadetblue1", "white", "white", "white", "white"), minRadius = 0.983, maxRadius = 0.983)

# Peripheral sensory neurofilament neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("gold", "white", "gold", "white", "gold", "white", "gold", "white", "gold", "gold", "gold", "white", "gold", "gold", "gold", "white", "white", "white", "gold", "gold", "gold", "gold", "white", "white", "white", "gold", "gold", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "gold", "gold", "white", "gold", "gold", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.016, maxRadius = 1.016)

# Peripheral sensory non-peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("darkgrey", "white", "white", "white", "darkgrey", "white", "white", "white", "darkgrey", "white", "white", "white", "white", "white", "white", "white", "white", "white", "darkgrey", "white", "white", "white", "white", "white", "white", "darkgrey", "darkgrey", "white", "white", "darkgrey", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "darkgrey", "white", "white", "darkgrey", "darkgrey", "darkgrey", "white", "darkgrey", "darkgrey", "darkgrey", "darkgrey"), minRadius = 1.037, maxRadius = 1.037)

# Peripheral sensory peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("magenta", "white", "white", "white", "magenta", "white", "white", "white", "magenta", "white", "white", "white", "white", "white", "white", "white", "white", "white", "magenta", "white", "magenta", "white", "white", "white", "white", "magenta", "magenta", "white", "white", "magenta", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "magenta", "white", "white", "magenta", "magenta", "magenta", "white", "white", "magenta", "magenta", "magenta"), minRadius = 1.063, maxRadius = 1.063)

# Sympathetic cholinergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "chartreuse3", "white", "white", "chartreuse3", "white", "white", "white", "chartreuse3", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "chartreuse3", "white", "white", "white", "chartreuse3", "chartreuse3", "chartreuse3", "white", "white", "white", "white", "white"), minRadius = 1.09, maxRadius = 1.09)

# Sympathetic nonadrenergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cyan", "white", "white", "white", "white", "cyan", "white", "white", "white", "white", "cyan", "cyan", "white", "white", "cyan", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "cyan", "white", "white", "white", "cyan", "white", "cyan", "white", "white", "white", "white", "white"), minRadius = 1.117, maxRadius = 1.117)

# Dendate gyrus granule neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "orange", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "orange", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "white", "orange", "orange", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.143, maxRadius = 1.143)

# Olfactory inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("coral", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "coral", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.17, maxRadius = 1.17)

# Spinal cord excitatory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "darkgreen", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.197, maxRadius = 1.197)

# Spinal cord inhibitory neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "firebrick", "firebrick", "white", "white", "firebrick", "white", "firebrick", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "firebrick", "white", "white", "white", "firebrick", "white", "white", "white", "white", "white", "white", "white", "white", "firebrick", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.223, maxRadius = 1.223)

# Peptidergic neurons:
tracklist = tracklist + BioCircosSNPTrack('Gene Type', points_chromosomes_receptor, points_coordinates_receptor, 
  points_values_receptor, colors = c("white", "white", "purple", "white", "white", "white", "purple", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white", "white"), minRadius = 1.25, maxRadius = 1.25)

#tracklist = tracklist + BioCircosTextTrack("testText", 'mDRG to Synovial Fibroblast', weight = "bold", size = '1em', x = -0.46, y = -1.25, displayGenomeBorder = TRUE)

links_chromosomes_1 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should start
links_chromosomes_2 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should end

# Coordinates to place start of links. The coordinates below correspond to the sequence given by seq(3, 92, length.out = 19) Their starting coordinate must be repeated for the number of connections between them and the number of ligands they are connected to.
links_pos_1 = c(3, 7.94, 3, 3, 12.89, 42.56, 42.56, 3, 7.94, 3, 7.94, 3, 3, 3, 3, 12.89, 3, 42.56, 3, 7.94, 3, 17.83, 22.78, 3, 27.72, 3, 27.72, 3, 22.78, 27.72, 3, 3, 3, 3, 3, 32.67, 7.94, 27.72, 3, 3, 3, 3, 27.72, 27.72, 3, 3, 3, 3, 32.67, 12.89, 27.72, 42.56, 42.56, 27.72, 3, 3, 3, 3, 7.94, 27.72, 17.83, 3, 3, 3, 7.94, 27.72, 3, 3, 3, 27.72, 27.72, 3, 3, 3, 12.89, 27.72, 42.56, 42.56, 27.72, 3, 3, 3, 7.94, 27.72, 17.83, 3, 3, 7.94, 3, 32.67, 27.72, 3, 3, 27.72, 3, 27.72, 3, 3, 12.89, 27.72, 3, 32.67, 27.72, 3, 3, 7.94, 27.72, 3, 17.83, 3, 3, 3, 3, 32.67, 7.94, 7.94, 27.72, 3, 3, 3, 3, 27.72, 3, 3, 3, 3, 32.67, 12.89, 12.89, 27.72, 42.56, 42.56, 3, 3, 3, 3, 7.94, 7.94, 17.83, 3, 3, 3, 3, 42.56, 17.83, 3, 3, 3, 3, 3, 3, 7.94, 32.67, 27.72, 27.72, 3, 3, 3, 3, 3, 3, 27.72, 27.72, 27.72, 3, 3, 3, 3, 3, 3, 12.89, 27.72, 37.61, 32.67, 27.72, 42.56, 27.72, 3, 3, 3, 3, 3, 3, 7.94, 27.72, 17.83, 37.61, 3, 3, 3, 27.72, 3, 3, 3, 27.72, 3, 3, 3, 27.72, 3, 3, 3, 3, 3, 3, 32.67, 27.72, 3, 3, 3, 27.72, 3, 3, 3, 32.67, 27.72, 3, 3, 3, 3, 3, 3, 42.56, 42.56, 3, 17.83, 3, 3, 3, 3, 3, 3, 3, 3, 3, 32.67, 32.67, 3, 3, 32.67, 32.67, 3, 47.5, 47.5, 52.44, 57.39, 57.39, 47.5, 52.44, 52.44, 52.44, 57.39, 57.39, 47.5, 52.44, 57.39, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 47.5, 57.39, 57.39, 47.5, 47.5, 62.33, 52.44, 62.33, 57.39, 57.39, 62.33, 62.33, 62.33, 47.5, 62.33, 62.33, 62.33, 52.44, 52.44, 62.33, 62.33, 62.33, 62.33, 52.44, 62.33, 57.39, 57.39, 47.5, 62.33, 62.33, 62.33, 62.33, 62.33, 62.33, 67.28, 52.44, 57.39, 62.33, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 57.39, 62.33, 62.33, 62.33, 47.5, 57.39, 47.5, 57.39, 57.39, 47.5, 47.5, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 67.28, 57.39, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 47.5, 57.39, 57.39, 42.56, 47.5, 47.5, 62.33, 52.44, 42.56, 42.56, 62.33, 72.22, 57.39, 57.39, 62.33, 62.33, 62.33, 77.17, 82.11, 42.56, 42.56, 47.5, 62.33, 62.33, 62.33, 72.22, 52.44, 52.44, 42.56, 62.33, 62.33, 62.33, 62.33, 72.22, 52.44, 42.56, 42.56, 62.33, 87.06, 87.06, 77.17, 57.39, 57.39, 42.56, 42.56, 92, 47.5, 62.33, 62.33, 62.33, 62.33, 62.33, 62.33, 87.06, 67.28, 77.17, 52.44, 57.39, 62.33, 42.56, 42.56, 82.11, 72.22, 57.39, 57.39, 42.56, 42.56, 42.56, 42.56, 47.5, 87.06, 87.06, 87.06, 82.11, 82.11, 57.39, 57.39, 42.56, 42.56, 42.56, 47.5, 87.06, 87.06, 87.06, 82.11, 82.11, 57.39, 57.39, 62.33, 62.33, 62.33, 82.11, 42.56, 42.56, 47.5, 87.06, 87.06, 57.39, 42.56, 42.56, 47.5, 87.06, 57.39, 57.39)

# Coordinates to place end of links. The coordinates below correspond to the sequence given by seq(3, 262, length.out = 53)
links_pos_2 = c(3, 7.98, 3, 3, 7.98, 12.96, 17.94, 3, 7.98, 22.92, 7.98, 27.90, 22.92, 27.90, 22.92, 7.98, 27.90, 17.94, 22.92, 7.98, 27.90, 32.88, 37.87, 42.85, 47.83, 42.85, 47.83, 42.85, 37.87, 47.83, 42.85, 22.92, 3, 42.85, 52.81, 57.59, 7.98, 47.83, 22.92, 3, 42.85, 52.81, 62.77, 47.83, 22.92, 3, 42.85, 52.81, 57.59, 7.98, 62.77, 12.96, 17.94, 47.83, 22.92, 3, 42.85, 52.81, 7.98, 62.77, 32.88, 67.75, 3, 52.81, 7.98, 47.83, 67.75, 3, 52.81, 62.77, 47.83, 67.75, 3, 52.81, 7.98, 62.77, 12.96, 17.94, 47.83, 67.75, 3, 52.81, 7.98, 62.77, 32.88, 22.92, 42.85, 72.73, 27.90, 77.71, 47.83, 22.92, 42.85, 82.69, 27.90, 47.83, 22.92, 42.85, 72.73, 82.69, 27.90, 77.71, 47.83, 22.92, 42.85, 72.73, 82.69, 27.90, 32.88, 22.92, 3, 87.67, 52.81, 57.59, 72.73, 7.98, 92.65, 22.92, 3, 87.67, 52.81, 92.65, 22.92, 3, 87.67, 52.81, 57.59, 72.73, 7.98, 92.65, 12.96, 17.94, 22.92, 3, 87.67, 52.81, 72.73, 7.98, 32.88, 3, 3, 3, 3, 12.96, 32.88, 22.92, 67.75, 3, 42.85, 97.63, 52.81, 72.73, 102.62, 92.65, 47.83, 22.92, 67.75, 3, 42.85, 97.63, 52.81, 62.77, 92.65, 47.83, 22.92, 67.75, 3, 42.85, 97.63, 52.81, 72.73, 62.77, 107.60, 102.62, 92.65, 12.96, 47.83, 22.92, 67.75, 3, 42.85, 97.63, 52.81, 72.73, 62.77, 32.88, 107.60, 22.92, 3, 42.85, 92.65, 22.92, 3, 42.85, 92.65, 22.92, 3, 42.85, 92.65, 22.92, 3, 42.85, 22.92, 3, 42.85, 102.62, 92.65, 22.92, 3, 42.85, 92.65, 22.92, 3, 42.85, 102.62, 92.65, 22.92, 3, 42.85, 42.85, 42.85, 42.85, 12.96, 17.94, 42.85, 32.88, 22.92, 112.58, 22.92, 112.58, 22.92, 112.58, 22.92, 112.58, 22.92, 77.71, 102.62, 22.92, 22.92, 77.71, 102.62, 22.92, 117.56, 122.54, 117.56, 127.52, 132.5, 117.56, 117.56, 137.48, 137.48, 127.52, 132.5, 142.46, 137.48, 127.52, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 147.44, 127.52, 132.5, 117.56, 122.54, 152.42, 117.56, 157.40, 127.52, 132.5, 162.38, 167.37, 172.35, 117.56, 177.33, 182.31, 157.40, 117.56, 137.48, 177.33, 162.38, 182.31, 157.40, 137.48, 187.29, 127.52, 132.5, 142.46, 177.33, 192.27, 162.38, 172.35, 182.31, 157.40, 197.25, 137.48, 127.52, 167.37, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 132.5, 157.40, 177.33, 157.40, 147.44, 127.52, 147.44, 127.52, 132.5, 117.56, 122.54, 127.52, 132.5, 117.56, 127.52, 132.5, 142.46, 197.25, 127.52, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 147.44, 127.52, 132.5, 117.56, 117.56, 122.54, 152.42, 117.56, 202.23, 207.21, 157.40, 212.19, 127.52, 132.5, 162.38, 167.37, 172.35, 217.17, 222.15, 117.56, 207.21, 117.56, 177.33, 182.31, 157.40, 212.19, 117.56, 137.48, 207.21, 177.33, 162.38, 182.31, 157.40, 212.19, 137.48, 227.13, 207.21, 187.29, 232.12, 237.10, 217.17, 127.52, 132.5, 242.08, 207.21, 142.46, 142.46, 177.33, 192.27, 162.38, 172.35, 182.31, 157.40, 142.46, 197.25, 217.17, 137.48, 127.52, 167.37, 227.13, 207.21, 222.15, 212.19, 127.52, 132.5, 247.06, 227.13, 252.04, 207.21, 147.44, 232.12, 257.02, 237.10, 222.15, 262, 127.52, 132.5, 227.13, 252.04, 207.21, 147.44, 232.12, 257.02, 237.10, 222.15, 262, 127.52, 132.5, 157.40, 177.33, 157.40, 222.15, 227.13, 207.21, 147.44, 232.12, 237.10, 127.52, 227.13, 207.21, 147.44, 237.10, 127.52, 132.5)


# The labels are given at the end of the link. 
links_labels_12 = c("FGF12", "LRRC4B", "", "", "", "CLSTN1", "CLSTN3", "", "", "FGF1", "", "NCAM1", "", "", "", "", "", "", "", "", "", "NFASC", "CXCL12", "FGF13", "EDIL3", "", "", "", "", "", "", "", "", "", "FGF9", "INHA", "", "", "", "", "", "", "LTBP3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF11", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "LFRN4", "", "TGFB1", "", "", "", "LTBP1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF14", "", "", "", "", "ADAM9", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF18", "", "", "TGFB3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "SHH", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "FGF7", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "LRP1", "LRP6", "", "ITGA3", "ITGB1", "", "", "VLDLR", "", "", "", "ERBB4", "", "", "", "", "NTRK1", "", "", "", "", "", "", "", "", "", "", "", "", "EPHB2", "", "EPHB6", "", "", "EPHA4", "EPHA5", "EPHA7", "", "EPHA10", "EPHB1", "", "", "", "", "", "", "", "", "EPHA8", "", "", "", "", "EPHA3", "", "", "", "", "ITGB5", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "NCSTN", "TNFRSF21", "", "PLXNA1", "", "", "", "", "", "UNC5A", "NRP1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "NGFR", "", "", "CD44", "CD9", "", "", "", "GPC1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "CAV1", "", "SLC45A3", "", "", "", "CD82", "", "", "PLXND1", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_1, links_pos_1,
  links_pos_1, links_chromosomes_2, links_pos_2, links_pos_2,
  maxRadius = 0.55, width = "0.01em", color = 'brown', labelSize = "0.75em", labels = links_labels_12)

links_chromosomes_3 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should start
links_chromosomes_4 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should end

# Now we start at the ligands
links_pos_3 = c(3, 7.98, 3, 3, 7.98, 12.96, 17.94, 3, 7.98, 22.92, 7.98, 27.90, 22.92, 27.90, 22.92, 7.98, 27.90, 17.94, 22.92, 7.98, 27.90, 32.88, 37.87, 42.85, 47.83, 42.85, 47.83, 42.85, 37.87, 47.83, 42.85, 22.92, 3, 42.85, 52.81, 57.59, 7.98, 47.83, 22.92, 3, 42.85, 52.81, 62.77, 47.83, 22.92, 3, 42.85, 52.81, 57.59, 7.98, 62.77, 12.96, 17.94, 47.83, 22.92, 3, 42.85, 52.81, 7.98, 62.77, 32.88, 67.75, 3, 52.81, 7.98, 47.83, 67.75, 3, 52.81, 62.77, 47.83, 67.75, 3, 52.81, 7.98, 62.77, 12.96, 17.94, 47.83, 67.75, 3, 52.81, 7.98, 62.77, 32.88, 22.92, 42.85, 72.73, 27.90, 77.71, 47.83, 22.92, 42.85, 82.69, 27.90, 47.83, 22.92, 42.85, 72.73, 82.69, 27.90, 77.71, 47.83, 22.92, 42.85, 72.73, 82.69, 27.90, 32.88, 22.92, 3, 87.67, 52.81, 57.59, 72.73, 7.98, 92.65, 22.92, 3, 87.67, 52.81, 92.65, 22.92, 3, 87.67, 52.81, 57.59, 72.73, 7.98, 92.65, 12.96, 17.94, 22.92, 3, 87.67, 52.81, 72.73, 7.98, 32.88, 3, 3, 3, 3, 12.96, 32.88, 22.92, 67.75, 3, 42.85, 97.63, 52.81, 72.73, 102.62, 92.65, 47.83, 22.92, 67.75, 3, 42.85, 97.63, 52.81, 62.77, 92.65, 47.83, 22.92, 67.75, 3, 42.85, 97.63, 52.81, 72.73, 62.77, 107.60, 102.62, 92.65, 12.96, 47.83, 22.92, 67.75, 3, 42.85, 97.63, 52.81, 72.73, 62.77, 32.88, 107.60, 22.92, 3, 42.85, 92.65, 22.92, 3, 42.85, 92.65, 22.92, 3, 42.85, 92.65, 22.92, 3, 42.85, 22.92, 3, 42.85, 102.62, 92.65, 22.92, 3, 42.85, 92.65, 22.92, 3, 42.85, 102.62, 92.65, 22.92, 3, 42.85, 42.85, 42.85, 42.85, 12.96, 17.94, 42.85, 32.88, 22.92, 112.58, 22.92, 112.58, 22.92, 112.58, 22.92, 112.58, 22.92, 77.71, 102.62, 22.92, 22.92, 77.71, 102.62, 22.92, 117.56, 122.54, 117.56, 127.52, 132.5, 117.56, 117.56, 137.48, 137.48, 127.52, 132.5, 142.46, 137.48, 127.52, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 147.44, 127.52, 132.5, 117.56, 122.54, 152.42, 117.56, 157.40, 127.52, 132.5, 162.38, 167.37, 172.35, 117.56, 177.33, 182.31, 157.40, 117.56, 137.48, 177.33, 162.38, 182.31, 157.40, 137.48, 187.29, 127.52, 132.5, 142.46, 177.33, 192.27, 162.38, 172.35, 182.31, 157.40, 197.25, 137.48, 127.52, 167.37, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 132.5, 157.40, 177.33, 157.40, 147.44, 127.52, 147.44, 127.52, 132.5, 117.56, 122.54, 127.52, 132.5, 117.56, 127.52, 132.5, 142.46, 197.25, 127.52, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 147.44, 127.52, 132.5, 117.56, 117.56, 122.54, 152.42, 117.56, 202.23, 207.21, 157.40, 212.19, 127.52, 132.5, 162.38, 167.37, 172.35, 217.17, 222.15, 117.56, 207.21, 117.56, 177.33, 182.31, 157.40, 212.19, 117.56, 137.48, 207.21, 177.33, 162.38, 182.31, 157.40, 212.19, 137.48, 227.13, 207.21, 187.29, 232.12, 237.10, 217.17, 127.52, 132.5, 242.08, 207.21, 142.46, 142.46, 177.33, 192.27, 162.38, 172.35, 182.31, 157.40, 142.46, 197.25, 217.17, 137.48, 127.52, 167.37, 227.13, 207.21, 222.15, 212.19, 127.52, 132.5, 247.06, 227.13, 252.04, 207.21, 147.44, 232.12, 257.02, 237.10, 222.15, 262, 127.52, 132.5, 227.13, 252.04, 207.21, 147.44, 232.12, 257.02, 237.10, 222.15, 262, 127.52, 132.5, 157.40, 177.33, 157.40, 222.15, 227.13, 207.21, 147.44, 232.12, 237.10, 127.52, 227.13, 207.21, 147.44, 237.10, 127.52, 132.5)
# Now we end at the receptors
links_pos_4 = c(3, 7.94, 3, 3, 12.89, 42.56, 42.56, 3, 7.94, 3, 7.94, 3, 3, 3, 3, 12.89, 3, 42.56, 3, 7.94, 3, 17.83, 22.78, 3, 27.72, 3, 27.72, 3, 22.78, 27.72, 3, 3, 3, 3, 3, 32.67, 7.94, 27.72, 3, 3, 3, 3, 27.72, 27.72, 3, 3, 3, 3, 32.67, 12.89, 27.72, 42.56, 42.56, 27.72, 3, 3, 3, 3, 7.94, 27.72, 17.83, 3, 3, 3, 7.94, 27.72, 3, 3, 3, 27.72, 27.72, 3, 3, 3, 12.89, 27.72, 42.56, 42.56, 27.72, 3, 3, 3, 7.94, 27.72, 17.83, 3, 3, 7.94, 3, 32.67, 27.72, 3, 3, 27.72, 3, 27.72, 3, 3, 12.89, 27.72, 3, 32.67, 27.72, 3, 3, 7.94, 27.72, 3, 17.83, 3, 3, 3, 3, 32.67, 7.94, 7.94, 27.72, 3, 3, 3, 3, 27.72, 3, 3, 3, 3, 32.67, 12.89, 12.89, 27.72, 42.56, 42.56, 3, 3, 3, 3, 7.94, 7.94, 17.83, 3, 3, 3, 3, 42.56, 17.83, 3, 3, 3, 3, 3, 3, 7.94, 32.67, 27.72, 27.72, 3, 3, 3, 3, 3, 3, 27.72, 27.72, 27.72, 3, 3, 3, 3, 3, 3, 12.89, 27.72, 37.61, 32.67, 27.72, 42.56, 27.72, 3, 3, 3, 3, 3, 3, 7.94, 27.72, 17.83, 37.61, 3, 3, 3, 27.72, 3, 3, 3, 27.72, 3, 3, 3, 27.72, 3, 3, 3, 3, 3, 3, 32.67, 27.72, 3, 3, 3, 27.72, 3, 3, 3, 32.67, 27.72, 3, 3, 3, 3, 3, 3, 42.56, 42.56, 3, 17.83, 3, 3, 3, 3, 3, 3, 3, 3, 3, 32.67, 32.67, 3, 3, 32.67, 32.67, 3, 47.5, 47.5, 52.44, 57.39, 57.39, 47.5, 52.44, 52.44, 52.44, 57.39, 57.39, 47.5, 52.44, 57.39, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 47.5, 57.39, 57.39, 47.5, 47.5, 62.33, 52.44, 62.33, 57.39, 57.39, 62.33, 62.33, 62.33, 47.5, 62.33, 62.33, 62.33, 52.44, 52.44, 62.33, 62.33, 62.33, 62.33, 52.44, 62.33, 57.39, 57.39, 47.5, 62.33, 62.33, 62.33, 62.33, 62.33, 62.33, 67.28, 52.44, 57.39, 62.33, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 57.39, 62.33, 62.33, 62.33, 47.5, 57.39, 47.5, 57.39, 57.39, 47.5, 47.5, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 67.28, 57.39, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 47.5, 57.39, 57.39, 42.56, 47.5, 47.5, 62.33, 52.44, 42.56, 42.56, 62.33, 72.22, 57.39, 57.39, 62.33, 62.33, 62.33, 77.17, 82.11, 42.56, 42.56, 47.5, 62.33, 62.33, 62.33, 72.22, 52.44, 52.44, 42.56, 62.33, 62.33, 62.33, 62.33, 72.22, 52.44, 42.56, 42.56, 62.33, 87.06, 87.06, 77.17, 57.39, 57.39, 42.56, 42.56, 92, 47.5, 62.33, 62.33, 62.33, 62.33, 62.33, 62.33, 87.06, 67.28, 77.17, 52.44, 57.39, 62.33, 42.56, 42.56, 82.11, 72.22, 57.39, 57.39, 42.56, 42.56, 42.56, 42.56, 47.5, 87.06, 87.06, 87.06, 82.11, 82.11, 57.39, 57.39, 42.56, 42.56, 42.56, 47.5, 87.06, 87.06, 87.06, 82.11, 82.11, 57.39, 57.39, 62.33, 62.33, 62.33, 82.11, 42.56, 42.56, 47.5, 87.06, 87.06, 57.39, 42.56, 42.56, 47.5, 87.06, 57.39, 57.39)
# These links are for the receptors
links_labels_34 = c('FGFR1', "PTPRS", "", "", "PTPRD", "APP", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "ANK3", "ACKR3", "", "ITGB5", "", "", "", "", "", "", "", "", "", "", "TGFBR3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "CDON", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "CTGF", "", "TFPI", "TIMP2", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "EFNA5", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "LTBP3", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "SEMA6D", "", "", "", "", "", "NTN4", "SEMA3E", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "HBEGF", "", "", "", "", "", "", "BTC", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_3, links_pos_3,
  links_pos_3, links_chromosomes_4, links_pos_4, links_pos_4,
  maxRadius = 0.55, width = "0.01em", color = 'brown', labelSize = "0.75em", labels = links_labels_34, labelPadding = 3)

links_chromosomes_5 = c('Receptor', 'Receptor', 'Receptor') # Chromosomes on which the links should start
links_chromosomes_6 = c('Ligand', 'Ligand', 'Ligand') # Chromosomes on which the links should end
links_pos_5 = c(117.56, 122.54, 117.56, 127.52, 132.5, 117.56, 117.56, 137.48, 137.48, 127.52, 132.5, 142.46, 137.48, 127.52, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 147.44, 127.52, 132.5, 117.56, 122.54, 152.42, 117.56, 157.40, 127.52, 132.5, 162.38, 167.37, 172.35, 117.56, 177.33, 182.31, 157.40, 117.56, 137.48, 177.33, 162.38, 182.31, 157.40, 137.48, 187.29, 127.52, 132.5, 142.46, 177.33, 192.27, 162.38, 172.35, 182.31, 157.40, 197.25, 137.48, 127.52, 167.37, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 132.5, 157.40, 177.33, 157.40, 147.44, 127.52, 147.44, 127.52, 132.5, 117.56, 122.54, 127.52, 132.5, 117.56, 127.52, 132.5, 142.46, 197.25, 127.52, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 132.5, 147.44, 127.52, 147.44, 127.52, 132.5, 117.56, 117.56, 122.54, 152.42, 117.56, 202.23, 207.21, 157.40, 212.19, 127.52, 132.5, 162.38, 167.37, 172.35, 217.17, 222.15, 117.56, 207.21, 117.56, 177.33, 182.31, 157.40, 212.19, 117.56, 137.48, 207.21, 177.33, 162.38, 182.31, 157.40, 212.19, 137.48, 227.13, 207.21, 187.29, 232.12, 237.10, 217.17, 127.52, 132.5, 242.08, 207.21, 142.46, 142.46, 177.33, 192.27, 162.38, 172.35, 182.31, 157.40, 142.46, 197.25, 217.17, 137.48, 127.52, 167.37, 227.13, 207.21, 222.15, 212.19, 127.52, 132.5, 247.06, 227.13, 252.04, 207.21, 147.44, 232.12, 257.02, 237.10, 222.15, 262, 127.52, 132.5, 227.13, 252.04, 207.21, 147.44, 232.12, 257.02, 237.10, 222.15, 262, 127.52, 132.5, 157.40, 177.33, 157.40, 222.15, 227.13, 207.21, 147.44, 232.12, 237.10, 127.52, 227.13, 207.21, 147.44, 237.10, 127.52, 132.5)
links_pos_6 = c(47.5, 47.5, 52.44, 57.39, 57.39, 47.5, 52.44, 52.44, 52.44, 57.39, 57.39, 47.5, 52.44, 57.39, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 47.5, 57.39, 57.39, 47.5, 47.5, 62.33, 52.44, 62.33, 57.39, 57.39, 62.33, 62.33, 62.33, 47.5, 62.33, 62.33, 62.33, 52.44, 52.44, 62.33, 62.33, 62.33, 62.33, 52.44, 62.33, 57.39, 57.39, 47.5, 62.33, 62.33, 62.33, 62.33, 62.33, 62.33, 67.28, 52.44, 57.39, 62.33, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 57.39, 62.33, 62.33, 62.33, 47.5, 57.39, 47.5, 57.39, 57.39, 47.5, 47.5, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 67.28, 57.39, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 57.39, 47.5, 57.39, 47.5, 57.39, 57.39, 42.56, 47.5, 47.5, 62.33, 52.44, 42.56, 42.56, 62.33, 72.22, 57.39, 57.39, 62.33, 62.33, 62.33, 77.17, 82.11, 42.56, 42.56, 47.5, 62.33, 62.33, 62.33, 72.22, 52.44, 52.44, 42.56, 62.33, 62.33, 62.33, 62.33, 72.22, 52.44, 42.56, 42.56, 62.33, 87.06, 87.06, 77.17, 57.39, 57.39, 42.56, 42.56, 92, 47.5, 62.33, 62.33, 62.33, 62.33, 62.33, 62.33, 87.06, 67.28, 77.17, 52.44, 57.39, 62.33, 42.56, 42.56, 82.11, 72.22, 57.39, 57.39, 42.56, 42.56, 42.56, 42.56, 47.5, 87.06, 87.06, 87.06, 82.11, 82.11, 57.39, 57.39, 42.56, 42.56, 42.56, 47.5, 87.06, 87.06, 87.06, 82.11, 82.11, 57.39, 57.39, 62.33, 62.33, 62.33, 82.11, 42.56, 42.56, 47.5, 87.06, 87.06, 57.39, 42.56, 42.56, 47.5, 87.06, 57.39, 57.39)
links_labels_56 = c("", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", "", "", '', '', '', '', '', '', "", '', '', '', "", '', "", "", "", "", "", "", '', "")

tracklist = tracklist + BioCircosLinkTrack('myLinkTrack', links_chromosomes_5, links_pos_5,
  links_pos_5, links_chromosomes_6, links_pos_6, links_pos_6,
  maxRadius = 0.55, width = "0.01em", color = 'cyan', labelSize = "0.5em", labels = links_labels_56)

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra = c('Ligand') # Chromosomes on which the arcs should be displayed
arcs_begin_ra = c(0)
arcs_end_ra = c(43)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra, arcs_begin_ra, arcs_end_ra, minRadius = 1.3, maxRadius = 1.35, colors = 'forestgreen')

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_ra2 = c('Ligand') # Chromosomes on which the arcs should be displayed
arcs_begin_ra2 = c(39)
arcs_end_ra2 = c(95)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_ra2, arcs_begin_ra2, arcs_end_ra2, minRadius = 1.35, maxRadius = 1.4, colors = 'deeppink3')

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_hdrg = c('Receptor') # Chromosomes on which the arcs should be displayed
arcs_begin_hdrg = c(0)
arcs_end_hdrg = c(114)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_hdrg, arcs_begin_hdrg, arcs_end_hdrg, minRadius = 1.35, maxRadius = 1.4, colors = 'deeppink3')

# Create inside arc for RA scRNA gene identification
arcs_chromosomes_hdrg2 = c('Receptor') # Chromosomes on which the arcs should be displayed
arcs_begin_hdrg2 = c(115)
arcs_end_hdrg2 = c(265)

tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes_hdrg2, arcs_begin_hdrg2, arcs_end_hdrg2, minRadius = 1.3, maxRadius = 1.35, colors = 'forestgreen')

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Mouse DRG to Human RA - Specific Neurons', x = -0.6, y = -1.30, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Receptor', x #= 0.75, y = -1, size = "1em",
##  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosTextTrack('MytextTrack', 'Ligand', x = #-0.95, y = -1, size = "1em",
#  weight = "bold", opacity = 1, color = "#000000")

#tracklist = tracklist + BioCircosArcTrack('myArcTrack', arcs_chromosomes, arcs_begin, arcs_end, minRadius = 0.85, maxRadius = 0.99, colors = 'forestgreen') 

BioCircos(tracklist, genome = myGenome, genomeFillColor = c('white', 'white'), chrPad = 0.05, displayGenomeBorder = FALSE, yChr =  FALSE,
  genomeTicksDisplay = FALSE,  genomeLabelTextSize = 0, genomeLabelDy = 0, width = 650, height = 650)
```

Below we make a legend for the ligand and receptor tracts.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Receptor', 'Ligand'), bty='n', lwd = 8, cex = 1.2, col = c("forestgreen", "deeppink3"), lty = c(1, 1), pch = c(NA, NA))
 #   col = c('forestgreen', 'deeppink3'))
#mtext("mDRG Neuronal Cell Types", at=0.19, cex=1.1)
```

Below we make a legend for the fibroblast subtypes.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (6)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (6)', 'DKK3+ sublining fibroblasts (SC-F3) (7)', 'CD55+ lining fibroblasts (SC-F4) (17)'), pch=15, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes", at=0.24, cex=2)
```

Below we split the legend in half - one for the ligands and one for the recpetors.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (3)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (1)', 'DKK3+ sublining fibroblasts (SC-F3) (4)', 'CD55+ lining fibroblasts (SC-F4) (7)'), pch=15, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes For Pain-Associated Receptors", at=0.24, cex=1.6)
```

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (3)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (5)', 'DKK3+ sublining fibroblasts (SC-F3) (4)', 'CD55+ lining fibroblasts (SC-F4) (11)'), pch=15, pt.cex=3, cex=1.5, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Subtypes For Pain-Associated Ligands", at=0.24, cex=1.6)
```

Below we plot the legend for the cell types.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Cerebellum Neurons (7)', 'Cholinergic and Monoaminergic Neurons (11)', 'Di- and Mesencphalon Excitatory Neurons (17)', 'Di- and Mesencphalon Inhibitory Neurons (16)', 'Enteric Neurons (17)', 'Hindbrain Neurons (23)', 'Peripheral Sensory Neurofilament Neurons (20)', 'Peripheral Sensory Non-peptidergic Neurons (15)', 'Peripheral Sensory Peptidergic Neurons (15)', 'Sympathetic Cholinergic Neurons (8)', 'Sympathetic Noradrenergic Neurons (9)', 'Dentate Gyrus Granule Neurons (8)', 'Olfactory Inhibitory Neurons (2)', 'Spinal Cord Excitatory Neurons (1)', 'Spinal Cord Inhibitory Neurons (7)', 'Peptidergic Neurons (2)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('antiquewhite4', 'black', 'blue', 'blueviolet', 'brown', 'cadetblue1', 'gold', 'darkgrey', 'magenta', 'chartreuse3', 'cyan', 'orange', 'coral', 'darkgreen', 'firebrick', 'purple'))
mtext("Neuronal Cell Types", at=0.19, cex=1.1)
```

Below we split the legend in half - one for the ligands and one for the recpetors.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Cerebellum Neurons (4)', 'Cholinergic and Monoaminergic Neurons (5)', 'Di- and Mesencphalon Excitatory Neurons (10)', 'Di- and Mesencphalon Inhibitory Neurons (9)', 'Enteric Neurons (9)', 'Hindbrain Neurons (10)', 'Peripheral Sensory Neurofilament Neurons (14)', 'Peripheral Sensory Non-peptidergic Neurons (4)', 'Peripheral Sensory Peptidergic Neurons (5)', 'Sympathetic Cholinergic Neurons (1)', 'Sympathetic Noradrenergic Neurons (3)', 'Dentate Gyrus Granule Neurons (3)', 'Olfactory Inhibitory Neurons (1)', 'Spinal Cord Inhibitory Neurons (4)', 'Peptidergic Neurons (2)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('antiquewhite4', 'black', 'blue', 'blueviolet', 'brown', 'cadetblue1', 'gold', 'darkgrey', 'magenta', 'chartreuse3', 'cyan', 'orange', 'coral', 'firebrick', 'purple'
            ))
mtext("Neuronal Cell Types For Ligands", at=0.19, cex=1.1)
```

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Cerebellum Neurons (3)', 'Cholinergic and Monoaminergic Neurons (6)', 'Di- and Mesencphalon Excitatory Neurons (7)', 'Di- and Mesencphalon Inhibitory Neurons (7)', 'Enteric Neurons (8)', 'Hindbrain Neurons (13)', 'Peripheral Sensory Neurofilament Neurons (6)', 'Peripheral Sensory Non-peptidergic Neurons (11)', 'Peripheral Sensory Peptidergic Neurons (10)', 'Sympathetic Cholinergic Neurons (7)', 'Sympathetic Noradrenergic Neurons (6)', 'Dentate Gyrus Granule Neurons (5)', 'Olfactory Inhibitory Neurons (1)', 'Spinal Cord Excitatory Neurons (1)', 'Spinal Cord Inhibitory Neurons (3)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('antiquewhite4', 'black', 'blue', 'blueviolet', 'brown', 'cadetblue1', 'gold', 'darkgrey', 'magenta', 'chartreuse3', 'cyan', 'orange', 'coral', 'darkgreen', 'firebrick', 'purple'))
mtext("Neuronal Cell Types For Receptors", at=0.19, cex=1.1)
```

Below we create plots for the number of connections for mDRG to RA and RA to mDRG.

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Mouse Nervous System to Synovial Fibroblast (242 connections)', 'Synovial Fibroblast to Mouse Nervous System (202 connections)'), bty='n', lwd = 8, cex = 1.2, col = c('brown', 'cyan'), lty = c(1, 1, 1, 1), pch = c(NA, NA, NA, NA))
 #   col = c('brown', 'cyan'))
#mtext("mDRG Neuronal Cell Types", at=0.19, cex=1.1)
```

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (27 Connections)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (51 Connections)', 'DKK3+ sublining fibroblasts (SC-F3) (23 Connections)', 'CD55+ lining fibroblasts (SC-F4) (101 Connections)'), pch=15, pt.cex=3, cex=1.2, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Connections Synovial Fibroblast to Mouse Nervous System", at=0.34, cex=1.3)
```

```{r}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('CD34+ sublining fibroblasts (SC-F1) (51 Connections)', 'HLA-DRAhi sublining fibroblasts (SC-F2) (48 Connections)', 'DKK3+ sublining fibroblasts (SC-F3) (64 Connections)', 'CD55+ lining fibroblasts (SC-F4) (79 Connections)'), pch=15, pt.cex=3, cex=1.2, bty='n',
    col = c('blue', 'green', 'orange', 'red'))
mtext("Synovial Fibroblast Connections Mouse Nervous System to Synovial Fibroblast", at=0.34, cex=1.3)
```

Below we create the connections legend for the cell types for ligands.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Cerebellum Neurons (9 Connections)', 'Cholinergic and Monoaminergic Neurons (13 Connections)', 'Di- and Mesencphalon Excitatory Neurons (30 Connections)', 'Di- and Mesencphalon Inhibitory Neurons (24 Connections)', 'Enteric Neurons (24 Connections)', 'Hindbrain Neurons (30 Connections)', 'Peripheral Sensory Neurofilament Neurons (42)', 'Peripheral Sensory Non-peptidergic Neurons (15 Connections)', 'Peripheral Sensory Peptidergic Neurons (17 Connections)', 'Sympathetic Cholinergic Neurons (8 Connections)', 'Sympathetic Noradrenergic Neurons (8 Connections)', 'Dentate Gyrus Granule Neurons (9 Connections)', 'Olfactory Inhibitory Neurons (1 Connection)', 'Spinal Cord Inhibitory Neurons (7 Connections)', 'Peptidergic Neurons (2 Connections)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('antiquewhite4', 'black', 'blue', 'blueviolet', 'brown', 'cadetblue1', 'gold', 'darkgrey', 'magenta', 'chartreuse3', 'cyan', 'orange', 'coral', 'firebrick', 'purple'
            ))
mtext("Neuronal Cell Type Connections Mouse Nervous System to Synovial Fibroblast", at=0.4, cex=1.1)
```

Below we create the connections legend for the cell types for receptors.

```{r, fig.height = 2}
plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
legend("topleft", legend =c('Cerebellum Neurons (14 Connections)', 'Cholinergic and Monoaminergic Neurons (13 Connections)', 'Di- and Mesencphalon Excitatory Neurons (19 Connections)', 'Di- and Mesencphalon Inhibitory Neurons (13 Connections)', 'Enteric Neurons (15 Connections)', 'Hindbrain Neurons (31 Connections)', 'Peripheral Sensory Neurofilament Neurons (12 Connections)', 'Peripheral Sensory Non-peptidergic Neurons (21 Connections)', 'Peripheral Sensory Peptidergic Neurons (20 Connections)', 'Sympathetic Cholinergic Neurons (12 Connections)', 'Sympathetic Noradrenergic Neurons (15 Connections)', 'Dentate Gyrus Granule Neurons (8 Connections)', 'Olfactory Inhibitory Neurons (2 Connections)', 'Spinal Cord Excitatory Neurons (2 Connections)', 'Spinal Cord Inhibitory Neurons (5 Connections)'), pch=16, pt.cex=1, cex=0.95, bty='n',
    col = c('antiquewhite4', 'black', 'blue', 'blueviolet', 'brown', 'cadetblue1', 'gold', 'darkgrey', 'magenta', 'chartreuse3', 'cyan', 'orange', 'coral', 'darkgreen', 'firebrick', 'purple'))
mtext("Neuronal Cell Type Connections Synovial Fibroblast to Mouse Nervous System", at=0.4, cex=1.1)
```

